<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AG Chiocchetti" />


<title>The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">femNATCD_MethSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/achiocch/femNATCD_MethSeq">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</h1>
<h3 class="subtitle">Preprocessing</h3>
<h4 class="author">AG Chiocchetti</h4>
<h4 class="date">29 Dezember 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>femNATCD_MethSeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree723a1aa6b066fd0af5ab10215e9f45b2b859b783targetblank723a1aaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/723a1aa6b066fd0af5ab10215e9f45b2b859b783" target="_blank">723a1aa</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree723a1aa6b066fd0af5ab10215e9f45b2b859b783targetblank723a1aaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/723a1aa6b066fd0af5ab10215e9f45b2b859b783" target="_blank">723a1aa</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/Epicounts.csv
    Ignored:    data/Epimeta.csv
    Ignored:    data/Epitpm.csv
    Ignored:    data/KangUnivers.txt
    Ignored:    data/Kang_DataPreprocessing.RData
    Ignored:    data/Kang_dataset_genesMod_version2.txt
    Ignored:    data/PatMeta.csv
    Ignored:    data/ProcessedData.RData
    Ignored:    data/RTrawdata/
    Ignored:    data/SNPCommonFilt.csv
    Ignored:    data/femNAT_PC20.txt
    Ignored:    output/BrainMod_Enrichemnt.pdf
    Ignored:    output/Brain_Module_Heatmap.pdf
    Ignored:    output/DMR_Results.csv
    Ignored:    output/GOres.xlsx
    Ignored:    output/LME_GOplot.pdf
    Ignored:    output/LME_Results.csv
    Ignored:    output/LME_Results_Sig.csv
    Ignored:    output/LME_tophit.svg
    Ignored:    output/ProcessedData.RData
    Ignored:    output/RNAvsMETplots.pdf
    Ignored:    output/Regions_GOplot.pdf
    Ignored:    output/ResultsgroupComp.txt
    Ignored:    output/SEM_summary_groupEpi_M15.txt
    Ignored:    output/SEM_summary_groupEpi_M2.txt
    Ignored:    output/SEM_summary_groupEpi_M_all.txt
    Ignored:    output/SEM_summary_groupEpi_TopHit.txt
    Ignored:    output/SEM_summary_groupEpi_all.txt
    Ignored:    output/SEMplot_Epi_M15.html
    Ignored:    output/SEMplot_Epi_M15.png
    Ignored:    output/SEMplot_Epi_M15_files/
    Ignored:    output/SEMplot_Epi_M2.html
    Ignored:    output/SEMplot_Epi_M2.png
    Ignored:    output/SEMplot_Epi_M2_files/
    Ignored:    output/SEMplot_Epi_M_all.html
    Ignored:    output/SEMplot_Epi_M_all.png
    Ignored:    output/SEMplot_Epi_M_all_files/
    Ignored:    output/SEMplot_Epi_TopHit.html
    Ignored:    output/SEMplot_Epi_TopHit.png
    Ignored:    output/SEMplot_Epi_TopHit_files/
    Ignored:    output/SEMplot_Epi_all.html
    Ignored:    output/SEMplot_Epi_all.png
    Ignored:    output/SEMplot_Epi_all_files/
    Ignored:    output/barplots.pdf
    Ignored:    output/circos_DMR_tags.svg
    Ignored:    output/circos_LME_tags.svg
    Ignored:    output/clinFact.RData
    Ignored:    output/dds_filt_analyzed.RData
    Ignored:    output/designh0.RData
    Ignored:    output/designh1.RData
    Ignored:    output/envFact.RData
    Ignored:    output/functional_Enrichemnt.pdf
    Ignored:    output/gostres.pdf
    Ignored:    output/modelFact.RData
    Ignored:    output/resdmr.RData
    Ignored:    output/resultsdmr_table.RData
    Ignored:    output/table1_filtered.Rmd
    Ignored:    output/table1_filtered.docx
    Ignored:    output/table1_unfiltered.Rmd
    Ignored:    output/table1_unfiltered.docx
    Ignored:    setup_built.R

Unstaged changes:
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/01_Preprocessing.Rmd</code>) and HTML (<code>docs/01_Preprocessing.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/70cd649ce41081b3d0bade1bd0cc176931c783f4/docs/01_Preprocessing.html" target="_blank">70cd649</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2a53a8725668ff1a29d6f35437a6c587012809ac/docs/01_Preprocessing.html" target="_blank">2a53a87</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/e3a9ae36c28b2ba178d5307fea867742cebed422/docs/01_Preprocessing.html" target="_blank">e3a9ae3</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f6bbdc0c4b54e5c101704cc3c7cb9c80408f92b2/docs/01_Preprocessing.html" target="_blank">f6bbdc0</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/710904a781215272de2c506a1e4c29f7fc802778/docs/01_Preprocessing.html" target="_blank">710904a</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/16112c31a3720918231ac96f98ae578113c75dc9/docs/01_Preprocessing.html" target="_blank">16112c3</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, “docs/”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/cde8384951ac43972b93866ec2512bb1ad4cc315/docs/01_Preprocessing.html" target="_blank">cde8384</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/d761be40b9985b450720b472375b01adc12b63d6/docs/01_Preprocessing.html" target="_blank">d761be4</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/b452d2f02a4bb8e5c4dd961ce50f42a703696053/docs/01_Preprocessing.html" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/1a9f36fde268d7bd9b5615ff46b35309442db585/analysis/01_Preprocessing.Rmd" target="_blank">1a9f36f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
reviewed analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2734c4e51aa341b50f59acf0df2900ab91d06654/docs/01_Preprocessing.html" target="_blank">2734c4e</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/a8478239ce3ca497da23250c261c259b1d1f9088/analysis/01_Preprocessing.Rmd" target="_blank">a847823</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/a8478239ce3ca497da23250c261c259b1d1f9088/docs/01_Preprocessing.html" target="_blank">a847823</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/9cc52f7bc59be3ada26bac3a9458fb6dbb154395/docs/01_Preprocessing.html" target="_blank">9cc52f7</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/158d0b43dc3337bef41cfd99c32e3322750945f4/analysis/01_Preprocessing.Rmd" target="_blank">158d0b4</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/158d0b43dc3337bef41cfd99c32e3322750945f4/docs/01_Preprocessing.html" target="_blank">158d0b4</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/0f262d1b920469dca52216115355a20663827e5e/docs/01_Preprocessing.html" target="_blank">0f262d1</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/5167b90e01239a0adae727c7561e97a2ae16a750/docs/01_Preprocessing.html" target="_blank">5167b90</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b5b9b9f50851ff5671827908b5f1ef47db1eb892/analysis/01_Preprocessing.Rmd" target="_blank">b5b9b9f</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
maual commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/b5b9b9f50851ff5671827908b5f1ef47db1eb892/docs/01_Preprocessing.html" target="_blank">b5b9b9f</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
maual commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/05aac7fc7613833ecedbf0ff9dd4fd18088608fc/docs/01_Preprocessing.html" target="_blank">05aac7f</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f5c52650307ec764d17e5f70981ad2cab2dc514b/docs/01_Preprocessing.html" target="_blank">f5c5265</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/17f1eec7fe797b93bcf724547cf86ce90c52e37f/docs/01_Preprocessing.html" target="_blank">17f1eec</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/91de221841561ea6b43fb67c9fd06d813bdb04b4/docs/01_Preprocessing.html" target="_blank">91de221</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/acd87d4a9901519f038d295fb45a30dbf07a5e1a/analysis/01_Preprocessing.Rmd" target="_blank">acd87d4</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b6c6b33c63a1e7fcceeb14961390eedf6a45dc19/analysis/01_Preprocessing.Rmd" target="_blank">b6c6b33</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
updated GO function, and model def
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/4ea1bba17a570a8d27e65e5b4153833c4d999e77/docs/01_Preprocessing.html" target="_blank">4ea1bba</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/6c21638381f6c8e162516fa9f9278f8e82503ab4/analysis/01_Preprocessing.Rmd" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/6c21638381f6c8e162516fa9f9278f8e82503ab4/docs/01_Preprocessing.html" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>source(paste0(Home,&quot;/code/custom_functions.R&quot;))</code></pre>
<pre><code>Lade nötiges Paket: RCircos</code></pre>
<pre><code>Lade nötiges Paket: scales</code></pre>
<pre><code>
Attache Paket: &#39;scales&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    discard</code></pre>
<pre><code>The following object is masked from &#39;package:readr&#39;:

    col_factor</code></pre>
<div id="initializing-rawdata" class="section level1">
<h1>Initializing rawdata</h1>
<pre class="r"><code>## load Data  ####
EpiData = fread(paste0(Home,&quot;/data/Epicounts.csv&quot;), header = T, data.table=F)
EpiData = EpiData %&gt;% column_to_rownames(&quot;ID&quot;)

EpiMeta = fread(paste0(Home,&quot;/data/Epimeta.csv&quot;), header = T, data.table=F)
EpiMeta = EpiMeta %&gt;% column_to_rownames(&quot;ID&quot;)

Patdata = fread(paste0(Home,&quot;/data/PatMeta.csv&quot;), header = T, data.table=F)
Patdata = Patdata %&gt;% column_to_rownames(&quot;V1&quot;)

SNPdata = fread(paste0(Home,&quot;/data/SNPCommonFilt.csv&quot;), header = T, data.table=F)
SNPdata = SNPdata %&gt;% column_to_rownames(&quot;V1&quot;)

PopPCAdata = fread(paste0(Home,&quot;/data/femNAT_PC20.txt&quot;), header = T, data.table=F)
PopPCAdata$ID = paste0(&quot;ID_&quot;, gsub(&quot;-&quot;, &quot;_&quot;, PopPCAdata$FID))
PopPCAdata = PopPCAdata %&gt;% column_to_rownames(&quot;ID&quot;)
PopPCAdata = PopPCAdata[rownames(Patdata), grepl(&quot;PC_&quot;,colnames(PopPCAdata))]

Patdata = cbind(Patdata, PopPCAdata)
Patdata$group = factor(Patdata$group, c(0,1), c(&quot;CTRL&quot;, &quot;CD&quot;))
Patdata$group = relevel(Patdata$group, &quot;CTRL&quot;)

Patdata$Age = Patdata$Alter

Patdata$site = factor(substr(Patdata$ID_femNAT, 1,2), 
                      c(&quot;01&quot;,&quot;02&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;), c(&quot;FRA&quot;, &quot;AAC&quot;, &quot;BCN&quot;, &quot;BLB&quot;, &quot;SZG&quot;))

Patdata$Pubstat = as.factor(gsub(&quot; &quot;, &quot;&quot;, Patdata$Pubstat))

Patdata$contraceptives = yesnofac(Patdata$contraceptives)
Patdata$Ever_smoked = factor(Patdata$Ever_smoked, c(0, 1,2), c(NA, &quot;no&quot;, &quot;yes&quot;))


allowedvalues=as.character(c(0,1,8, NA))



for (i in 1:ncol(Patdata)) {
  if(sum(Patdata[,i] %in% allowedvalues)==nrow(Patdata)){
    Patdata[,i] = yesnofac(Patdata[,i])
    print(colnames(Patdata)[i])
  }
  
}</code></pre>
<pre><code>[1] &quot;smoke&quot;
[1] &quot;drugs&quot;
[1] &quot;aggress&quot;
[1] &quot;stress&quot;
[1] &quot;mpsych_4_12&quot;
[1] &quot;mpsych_gt12&quot;
[1] &quot;ppsych_le12&quot;
[1] &quot;inconsist&quot;
[1] &quot;single&quot;
[1] &quot;disharm&quot;
[1] &quot;isolation&quot;
[1] &quot;psych_fam&quot;
[1] &quot;Dep_life&quot;
[1] &quot;anp_life&quot;
[1] &quot;CD_life&quot;
[1] &quot;ADHD_life&quot;
[1] &quot;TIC_life&quot;
[1] &quot;ODD_life&quot;
[1] &quot;DMDD_life&quot;
[1] &quot;Alk_life&quot;
[1] &quot;SUD_life&quot;
[1] &quot;Anx_life&quot;
[1] &quot;OCD_life&quot;
[1] &quot;PTSD_life&quot;
[1] &quot;Elim_life&quot;
[1] &quot;Eat_life&quot;
[1] &quot;Oth_life&quot;
[1] &quot;Bip_life&quot;
[1] &quot;Scz_life&quot;
[1] &quot;medication&quot;</code></pre>
<pre class="r"><code>SNPdatafilt = SNPdata[complete.cases(SNPdata),]

display_tab(head(EpiMeta))</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:240px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
CHR
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
start
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
stop
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
gene
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
feature
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
cpg
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
tf_binding
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
raw_gene
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
distance
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
samples_per_target
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mean
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
std
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
chr1+_10482_10487
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10482
</td>
<td style="text-align:left;">
10487
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
TAF7_(SQ-8)//ZBTB33
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
20.569
</td>
<td style="text-align:left;">
13.877
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10495_10500
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10495
</td>
<td style="text-align:left;">
10500
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
TAF7_(SQ-8)//ZBTB33
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
98
</td>
<td style="text-align:left;">
10.608
</td>
<td style="text-align:left;">
9.030
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10587_10592
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10587
</td>
<td style="text-align:left;">
10592
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
85
</td>
<td style="text-align:left;">
6.176
</td>
<td style="text-align:left;">
6.170
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10639_10644
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10639
</td>
<td style="text-align:left;">
10644
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
0.941
</td>
<td style="text-align:left;">
1.241
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10668_10673
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10668
</td>
<td style="text-align:left;">
10673
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
51
</td>
<td style="text-align:left;">
0.922
</td>
<td style="text-align:left;">
1.272
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10697_10702
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10697
</td>
<td style="text-align:left;">
10702
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
DDX11L1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
65
</td>
<td style="text-align:left;">
1.029
</td>
<td style="text-align:left;">
1.085
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>display_tab(head(EpiData))</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:240px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0023
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0028
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0029
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0056
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0074
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0075
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0145
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0173
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0185
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0209
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0221
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0248
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0249
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0260
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0284
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0286
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0289
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0292
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0293
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0294
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0295
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0311
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0322
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0325
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0343
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0362
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0365
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0379
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0382
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0383
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0385
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0386
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0400
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0431
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0433
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0439
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0500
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_01_0515
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0009
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0016
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0021
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0022
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0032
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0038
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0086
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0090
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0137
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0140
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0184
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0189
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0192
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0197
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0201
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0211
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0233
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0244
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0269
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0275
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0302
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0304
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0333
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0354
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0394
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0438
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0497
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_02_0535
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0006
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0007
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0008
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0017
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0022
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0025
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0029
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0032
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0034
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0037
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0041
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0044
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_10_0045
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0001
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0002
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0003
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0006
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0007
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0010
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0013
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0017
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0018
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0046
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0056
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_11_0058
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0003
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0008
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0011
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0014
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0017
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0021
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0023
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0025
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0026
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0028
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_12_0029
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
chr1+_10482_10487
</td>
<td style="text-align:left;">
32
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
29
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
37
</td>
<td style="text-align:left;">
39
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
26
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
27
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
38
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
45
</td>
<td style="text-align:left;">
41
</td>
<td style="text-align:left;">
52
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
26
</td>
<td style="text-align:left;">
48
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
62
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
32
</td>
<td style="text-align:left;">
41
</td>
<td style="text-align:left;">
42
</td>
<td style="text-align:left;">
43
</td>
<td style="text-align:left;">
27
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
45
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
40
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
29
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
43
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
43
</td>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
37
</td>
<td style="text-align:left;">
40
</td>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
54
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10495_10500
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
40
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
42
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
26
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
37
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10587_10592
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
13
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
19
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
20
</td>
<td style="text-align:left;">
18
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10639_10644
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10668_10673
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr>
<td style="text-align:left;">
chr1+_10697_10702
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
</tr>
</tbody>
</table>
</div>
<pre class="r"><code>display_tab(head(Patdata))</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:240px; overflow-x: scroll; width:900px; ">
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; ">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
lfd.Nr
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ID_femNAT
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
DateIncome
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
X260_280
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ng_per_ul
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ul
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
contraceptives
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
group
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Diagnose
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Alter
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Pubstat
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Timediff_ExtrPurification
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Reliability
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
LCL
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Prosp_Distance
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
cigday_1
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Ever_smoked
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
age_months
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
scrsympt
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
suppsympt
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
tsympt
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
smoke
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
drugs
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
aggress
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
stress
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mpsych_4_12
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mpsych_gt12
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ppsych_le12
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
inconsist
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
single
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
disharm
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
isolation
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mother_school
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mother_prof
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
father_school
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
father_prof
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
psych_fam
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
n_trauma
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Dep_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
anp_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
CD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ADHD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
TIC_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
ODD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
DMDD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Alk_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
SUD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Anx_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
OCD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PTSD_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Elim_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Eat_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Oth_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Bip_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Scz_life
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
medication
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
reads_per_sample
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
targets_per_sample
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_1
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_2
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_3
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_4
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_5
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_6
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_7
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_8
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_9
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_10
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_11
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_12
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_13
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_14
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_15
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_16
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_17
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_18
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_19
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
PC_20
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
Age
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
site
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ID_01_0023
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
01-0023
</td>
<td style="text-align:left;">
2014-08-28
</td>
<td style="text-align:left;">
1.853
</td>
<td style="text-align:left;">
1135
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CTRL
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
17.2
</td>
<td style="text-align:left;">
Latepubertal
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.462
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
205
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
10313348
</td>
<td style="text-align:left;">
1087812
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
-0.001
</td>
<td style="text-align:left;">
-0.007
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.005
</td>
<td style="text-align:left;">
-0.003
</td>
<td style="text-align:left;">
0.011
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
0.007
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
-0.016
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
-0.008
</td>
<td style="text-align:left;">
0.020
</td>
<td style="text-align:left;">
17.2
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
<tr>
<td style="text-align:left;">
ID_01_0028
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
01-0028
</td>
<td style="text-align:left;">
2014-05-09
</td>
<td style="text-align:left;">
1.813
</td>
<td style="text-align:left;">
363
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CTRL
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
15.2
</td>
<td style="text-align:left;">
Latepubertal
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.501
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
182
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
11522586
</td>
<td style="text-align:left;">
1091132
</td>
<td style="text-align:left;">
-0.011
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.006
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
-0.007
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.007
</td>
<td style="text-align:left;">
-0.003
</td>
<td style="text-align:left;">
-0.012
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
-0.012
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
-0.001
</td>
<td style="text-align:left;">
0.027
</td>
<td style="text-align:left;">
15.2
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
<tr>
<td style="text-align:left;">
ID_01_0029
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
01-0029
</td>
<td style="text-align:left;">
2014-05-19
</td>
<td style="text-align:left;">
1.831
</td>
<td style="text-align:left;">
298
</td>
<td style="text-align:left;">
17
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CTRL
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
17.5
</td>
<td style="text-align:left;">
Postpubertal
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.219
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
209
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
10266110
</td>
<td style="text-align:left;">
1069619
</td>
<td style="text-align:left;">
-0.011
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
0.026
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
-0.005
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
0.016
</td>
<td style="text-align:left;">
-0.001
</td>
<td style="text-align:left;">
0.015
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.020
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
0.005
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
17.5
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
<tr>
<td style="text-align:left;">
ID_01_0056
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
01-0056
</td>
<td style="text-align:left;">
2014-06-12
</td>
<td style="text-align:left;">
1.850
</td>
<td style="text-align:left;">
555
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CTRL
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
16.9
</td>
<td style="text-align:left;">
Postpubertal
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
duplicate
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.254
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
10098940
</td>
<td style="text-align:left;">
1076017
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
-0.008
</td>
<td style="text-align:left;">
-0.007
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
-0.003
</td>
<td style="text-align:left;">
-0.003
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.007
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
-0.005
</td>
<td style="text-align:left;">
0.024
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.011
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
-0.024
</td>
<td style="text-align:left;">
0.009
</td>
<td style="text-align:left;">
16.9
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
<tr>
<td style="text-align:left;">
ID_01_0074
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
01-0074
</td>
<td style="text-align:left;">
2014-08-28
</td>
<td style="text-align:left;">
1.842
</td>
<td style="text-align:left;">
1220
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CD
</td>
<td style="text-align:left;">
case
</td>
<td style="text-align:left;">
15.5
</td>
<td style="text-align:left;">
Latepubertal
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.509
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
186
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
9715101
</td>
<td style="text-align:left;">
1058814
</td>
<td style="text-align:left;">
-0.010
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
-0.004
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
0.011
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
-0.001
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
-0.027
</td>
<td style="text-align:left;">
-0.004
</td>
<td style="text-align:left;">
-0.049
</td>
<td style="text-align:left;">
0.036
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.015
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.028
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
15.5
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
<tr>
<td style="text-align:left;">
ID_01_0075
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
01-0075
</td>
<td style="text-align:left;">
2014-09-08
</td>
<td style="text-align:left;">
1.862
</td>
<td style="text-align:left;">
508
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
CTRL
</td>
<td style="text-align:left;">
control
</td>
<td style="text-align:left;">
16.0
</td>
<td style="text-align:left;">
Latepubertal
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
0.474
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
190
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
no
</td>
<td style="text-align:left;">
5664046
</td>
<td style="text-align:left;">
823351
</td>
<td style="text-align:left;">
0.067
</td>
<td style="text-align:left;">
-0.025
</td>
<td style="text-align:left;">
-0.004
</td>
<td style="text-align:left;">
0.028
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
-0.004
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
-0.002
</td>
<td style="text-align:left;">
0.020
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
-0.014
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
-0.015
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
-0.048
</td>
<td style="text-align:left;">
-0.004
</td>
<td style="text-align:left;">
0.026
</td>
<td style="text-align:left;">
-0.009
</td>
<td style="text-align:left;">
16.0
</td>
<td style="text-align:left;">
FRA
</td>
</tr>
</tbody>
</table>
</div>
<ul>
<li><p>Total number of samples: 102</p></li>
<li><p>Total number of Tags: 2 271 932</p></li>
<li><p>Tags in TFbinding sites: 415 218</p></li>
<li><p>Tags in CpGs:</p>
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
<p>Var1</p>
</th>
<th style="text-align:left;">
<p>Freq</p>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>cpg</p>
</td>
<td style="text-align:left;">
<p>265576</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>NoCpg</p>
</td>
<td style="text-align:left;">
<p>2006356</p>
</td>
</tr>
</tbody>
</table></li>
<li><p>Tags per feature</p>
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
<p>Var1</p>
</th>
<th style="text-align:left;">
<p>Freq</p>
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<p>downstream</p>
</td>
<td style="text-align:left;">
<p>37759</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>exonic</p>
</td>
<td style="text-align:left;">
<p>110147</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>intergenic</p>
</td>
<td style="text-align:left;">
<p>986333</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>intronic</p>
</td>
<td style="text-align:left;">
<p>887762</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ncR0_exonic</p>
</td>
<td style="text-align:left;">
<p>13324</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ncR0_intronic</p>
</td>
<td style="text-align:left;">
<p>61552</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ncR0_splicing</p>
</td>
<td style="text-align:left;">
<p>4</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ncR0_UTR3</p>
</td>
<td style="text-align:left;">
<p>415</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>ncR0_UTR5</p>
</td>
<td style="text-align:left;">
<p>670</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>splicing</p>
</td>
<td style="text-align:left;">
<p>1952</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>upstream</p>
</td>
<td style="text-align:left;">
<p>104665</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>UTR3</p>
</td>
<td style="text-align:left;">
<p>26178</p>
</td>
</tr>
<tr>
<td style="text-align:left;">
<p>UTR5</p>
</td>
<td style="text-align:left;">
<p>41171</p>
</td>
</tr>
</tbody>
</table></li>
</ul>
<div id="sample-descriptives" class="section level2">
<h2>Sample descriptives</h2>
<pre class="r"><code>index = sample(1:nrow(EpiData), 10000)

Sampleddata = EpiData[index,]

boxplot_counts = function(plotsubset, maintitle, colorcode) {
 dat=log2(plotsubset+1)
   par(mar=c(5,5,5,8))
  
  a = boxplot(dat, main = maintitle, col = Dark8[as.factor(Patdata[,colorcode])], names=NA,
              ylab = &quot;log2 transformed&quot;, xlab=&quot;samples&quot;)
  
  legend(x=length(a$n)*1.05, y=max(dat), 
         xpd=T,
         legend = levels(Patdata[,colorcode]), 
         bty=&quot;n&quot;,pch = 16, 
         col = Dark8[1:length(unique(Patdata[,colorcode]))])
}


barplot_counts = function(DF, maintitle, colorcode) {
  dat=log2(DF[,&quot;reads_per_sample&quot;])
  par(mar=c(5,5,5,8))
  a = barplot(dat, main = maintitle, col = Dark8[as.factor(DF[,colorcode])], names=&quot;&quot;,
              ylab = &quot;log2 transformed&quot;,xlab=&quot;samples&quot;)
  legend(x=max(a)*1.05,y=max(dat), xpd=T, bty=&quot;n&quot;,
         legend = levels(DF[,colorcode]), pch = 16, 
         col = Dark8[1:length(unique(DF[,colorcode]))])
}


boxplot_counts(Sampleddata, &quot;raw counts&quot;, &quot;group&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/plot_counts-1.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>boxplot_counts(Sampleddata, &quot;raw counts&quot;, &quot;site&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/plot_counts-2.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot_counts(Patdata, &quot;total reads&quot;, &quot;group&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/plot_counts-3.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot_counts(Patdata, &quot;total reads&quot;, &quot;site&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/plot_counts-4.png" width="4500" style="display: block; margin: auto;" /></p>
<ul>
<li><p>Average Reads per Sample 10 059 904</p></li>
<li><p>Average Targets detected per sample 1 009 668</p></li>
<li><p>N Targets detected in at least 50% of samples 904 845</p></li>
</ul>
</div>
<div id="sample-stats" class="section level2">
<h2>Sample stats</h2>
<div id="population-structure" class="section level3">
<h3>Population structure</h3>
<pre class="r"><code>summary(aov(PC_1~site, data = Patdata))</code></pre>
<pre><code>            Df   Sum Sq   Mean Sq F value Pr(&gt;F)
site         4 0.000436 0.0001089   0.402  0.807
Residuals   94 0.025459 0.0002708               
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_2~site, data = Patdata))</code></pre>
<pre><code>            Df   Sum Sq   Mean Sq F value  Pr(&gt;F)   
site         4 0.003664 0.0009161   3.721 0.00741 **
Residuals   94 0.023139 0.0002462                   
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_3~site, data = Patdata))</code></pre>
<pre><code>            Df  Sum Sq   Mean Sq F value Pr(&gt;F)  
site         4 0.00420 0.0010500   2.954 0.0239 *
Residuals   94 0.03341 0.0003554                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_4~site, data = Patdata))</code></pre>
<pre><code>            Df  Sum Sq  Mean Sq F value   Pr(&gt;F)    
site         4 0.00654 0.001635   6.124 0.000201 ***
Residuals   94 0.02509 0.000267                     
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_1~site, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_1 ~ site, data = Patdata)

$site
                 diff         lwr         upr     p adj
AAC-FRA -0.0040129005 -0.01547863 0.007452831 0.8664634
BCN-FRA -0.0008664120 -0.01607299 0.014340167 0.9998558
BLB-FRA -0.0003408281 -0.01554741 0.014865751 0.9999965
SZG-FRA -0.0055795906 -0.02189406 0.010734882 0.8758893
BCN-AAC  0.0031464885 -0.01264726 0.018940241 0.9811739
BLB-AAC  0.0036720723 -0.01212168 0.019465825 0.9668597
SZG-AAC -0.0015666902 -0.01842980 0.015296423 0.9990011
BLB-BCN  0.0005255838 -0.01816184 0.019213004 0.9999914
SZG-BCN -0.0047131787 -0.02431271 0.014886353 0.9625894
SZG-BLB -0.0052387625 -0.02483829 0.014360769 0.9456298</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_2~site, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_2 ~ site, data = Patdata)

$site
                diff           lwr        upr     p adj
AAC-FRA  0.002094151 -0.0088367334 0.01302504 0.9837234
BCN-FRA  0.016825136  0.0023279051 0.03132237 0.0144702
BLB-FRA  0.009168805 -0.0053284265 0.02366604 0.4035041
SZG-FRA  0.012851776 -0.0027016683 0.02840522 0.1545913
BCN-AAC  0.014730985 -0.0003260293 0.02978800 0.0582884
BLB-AAC  0.007074653 -0.0079823610 0.02213167 0.6876306
SZG-AAC  0.010757625 -0.0053188673 0.02683412 0.3454238
BLB-BCN -0.007656332 -0.0254720314 0.01015937 0.7540415
SZG-BCN -0.003973360 -0.0226586238 0.01471190 0.9760529
SZG-BLB  0.003682971 -0.0150022922 0.02236823 0.9819062</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_3~site, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_3 ~ site, data = Patdata)

$site
                diff           lwr         upr     p adj
AAC-FRA -0.001737926 -0.0148728028 0.011396951 0.9960246
BCN-FRA -0.013906462 -0.0313267673 0.003513844 0.1812633
BLB-FRA  0.008036310 -0.0093839954 0.025456616 0.7021660
SZG-FRA  0.010114316 -0.0085751669 0.028803798 0.5617034
BCN-AAC -0.012168536 -0.0302614939 0.005924422 0.3402626
BLB-AAC  0.009774236 -0.0083187220 0.027867194 0.5634084
SZG-AAC  0.011852241 -0.0074657512 0.031170234 0.4349907
BLB-BCN  0.021942772  0.0005348955 0.043350648 0.0417121
SZG-BCN  0.024020778  0.0015680073 0.046473548 0.0297960
SZG-BLB  0.002078006 -0.0203747646 0.024530776 0.9990161</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_4~site, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_4 ~ site, data = Patdata)

$site
                diff           lwr           upr     p adj
AAC-FRA -0.001024591 -0.0124078626  0.0103586798 0.9991183
BCN-FRA -0.020580024 -0.0356772391 -0.0054828094 0.0024043
BLB-FRA -0.017184914 -0.0322821291 -0.0020876994 0.0173663
SZG-FRA  0.001992382 -0.0142047589  0.0181895220 0.9970040
BCN-AAC -0.019555433 -0.0352355982 -0.0038752675 0.0069194
BLB-AAC -0.016160323 -0.0318404882 -0.0004801575 0.0400165
SZG-AAC  0.003016973 -0.0137248623  0.0197588082 0.9870587
BLB-BCN  0.003395110 -0.0151579118  0.0219481318 0.9862898
SZG-BCN  0.022572406  0.0031138323  0.0420309793 0.0145363
SZG-BLB  0.019177296 -0.0002812777  0.0386358693 0.0554122</code></pre>
<pre class="r"><code>summary(aov(PC_1~group, data = Patdata))</code></pre>
<pre><code>            Df   Sum Sq   Mean Sq F value Pr(&gt;F)
group        1 0.000058 0.0000577   0.217  0.643
Residuals   97 0.025837 0.0002664               
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_2~group, data = Patdata))</code></pre>
<pre><code>            Df  Sum Sq   Mean Sq F value Pr(&gt;F)  
group        1 0.00154 0.0015401   5.913 0.0169 *
Residuals   97 0.02526 0.0002604                 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_3~group, data = Patdata))</code></pre>
<pre><code>            Df  Sum Sq   Mean Sq F value Pr(&gt;F)
group        1 0.00011 0.0001141   0.295  0.588
Residuals   97 0.03750 0.0003866               
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>summary(aov(PC_4~group, data = Patdata))</code></pre>
<pre><code>            Df  Sum Sq   Mean Sq F value Pr(&gt;F)
group        1 0.00000 0.0000022   0.007  0.935
Residuals   97 0.03163 0.0003261               
3 observations deleted due to missingness</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_1~group, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_1 ~ group, data = Patdata)

$group
               diff          lwr         upr     p adj
CD-CTRL 0.001526944 -0.004984366 0.008038254 0.6426666</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_2~group, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_2 ~ group, data = Patdata)

$group
               diff         lwr        upr     p adj
CD-CTRL 0.007888848 0.001450216 0.01432748 0.0168606</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_3~group, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_3 ~ group, data = Patdata)

$group
                diff          lwr         upr     p adj
CD-CTRL -0.002147168 -0.009991311 0.005696975 0.5881862</code></pre>
<pre class="r"><code>TukeyHSD(aov(PC_4~group, data = Patdata))</code></pre>
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = PC_4 ~ group, data = Patdata)

$group
                diff          lwr         upr     p adj
CD-CTRL 0.0002963217 -0.006908285 0.007500928 0.9351087</code></pre>
<pre class="r"><code># Population structure plot

pairs(~PC_1+PC_2+PC_3+PC_4, data=Patdata, main=&quot;Genetic population structure&quot;,
      col=Dark8[as.factor(Patdata$site)], pch=as.numeric(Patdata$group)+15)

legend(0.72,0.03, cex = 0.7, legend = c(levels(Patdata$site)), title=&quot;site&quot;,xpd=T,
       pch = 16, col=Dark8[1:length(unique(Patdata$site))], ncol = nlevels(Patdata$site), bty = &quot;n&quot;)

legend(0.30,0.03, cex = 0.7, legend = c(levels(Patdata$group)), title=&quot;group&quot;,xpd=T,
       pch = 15+c(1:nlevels(Patdata$group)), col=&quot;gray&quot;, ncol = nlevels(Patdata$group), bty = &quot;n&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/plot_population-1.png" width="4500" style="display: block; margin: auto;" /></p>
</div>
<div id="covariates" class="section level3">
<h3>Covariates</h3>
<pre class="r"><code># Patient specific covariates ####

clinFact = c(&quot;site&quot;, &quot;Age&quot;, &quot;Pubstat&quot;)
modelFact = c(&quot;Age&quot;)

Patdata$int_dis = with(Patdata, (Dep_life==&quot;yes&quot; | 
                                   Anx_life==&quot;yes&quot; |
                                   OCD_life==&quot;yes&quot; |
                                   #PTSD_life==&quot;yes&quot; |
                                   #Eat_life==&quot;yes&quot; |
                                   DMDD_life==&quot;yes&quot;)*1)
Patdata$int_dis = yesnofac(Patdata$int_dis)
clinFact = c(clinFact, &quot;int_dis&quot;)
modelFact = c(modelFact,&quot;int_dis&quot;)

# Patdata$ext_dis = with(Patdata, (ADHD_life==&quot;yes&quot; | 
#                                   ODD_life==&quot;yes&quot; |
#                                   anp_life==&quot;yes&quot;) * 1)
# Patdata$ext_dis = yesnofac(Patdata$ext_dis)
# clinFact  = c(clinFact, &quot;ext_dis&quot;)

# Patdata$add_dis = with(Patdata, (Alk_life==&quot;yes&quot; | 
#                                    SUD_life==&quot;yes&quot;) * 1)
# 
# Patdata$add_dis = yesnofac(Patdata$add_dis)
# clinFact  = c(clinFact, &quot;add_dis&quot;)

clinFact = c(clinFact, &quot;medication&quot;, &quot;contraceptives&quot;, &quot;cigday_1&quot;)
modelFact = c(modelFact,&quot;medication&quot;, &quot;contraceptives&quot;, &quot;cigday_1&quot;)


# Environmental Factors ####
envFact = c()

#maternal smoking,
Patdata$Matsmk = with(Patdata, (smoke==&quot;yes&quot;)*1)
Patdata$Matsmk = factor(Patdata$Matsmk,levels = c(0,1), labels = c(&quot;no&quot;, &quot;yes&quot;))
envFact = c(envFact, &quot;Matsmk&quot;)

#violence exposure 
Patdata$Matagg = with(Patdata, (aggress==&quot;yes&quot;)*1)
Patdata$Matagg = factor(Patdata$Matagg,levels = c(0,1), labels = c(&quot;no&quot;, &quot;yes&quot;))
envFact = c(envFact, &quot;Matagg&quot;)


#adverse family situation
Patdata$FamScore = with(Patdata, rowSums(cbind(#single==&quot;yes&quot;, 
  disharm==&quot;yes&quot;, 
  isolation==&quot;yes&quot;), 
  na.rm=T))
envFact = c(envFact, &quot;FamScore&quot;)

#parental education status 
Patdata$EduPar = with(Patdata, 
                      apply(cbind(apply(cbind(mother_school, mother_prof), 1, max, na.rm=T), 
                                  apply(cbind(father_school, father_prof), 1, max, na.rm=T)), 
                            1, mean, na.rm=T))</code></pre>
<pre><code>Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück

Warning in FUN(newX[, i], ...): kein nicht-fehlendes Argument für max; gebe -Inf
zurück</code></pre>
<pre class="r"><code>Patdata$EduPar[is.infinite(Patdata$EduPar)]=NA
envFact = c(envFact, &quot;EduPar&quot;)

# number of trauma
envFact = c(envFact, &quot;n_trauma&quot;)

corplotfacs=c(&quot;group&quot;, unique(c(modelFact, clinFact)), &quot;PC_1&quot;, &quot;PC_2&quot;, &quot;PC_3&quot;, &quot;PC_4&quot;)
corplotfacs = corplotfacs[corplotfacs != &quot;site&quot;]

cormat = cor(apply(Patdata[,corplotfacs] %&gt;% mutate_all(as.numeric), 2, minmax_scaling),
             use = &quot;pairwise.complete.obs&quot;)

corrplot(cormat, order = &quot;hclust&quot;, main=&quot;clustered and scaled correlations&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/cov_factors-1.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save(clinFact, file=paste0(Home,&quot;/output/clinFact.RData&quot;))
save(modelFact, file=paste0(Home,&quot;/output/modelFact.RData&quot;))
save(envFact, file=paste0(Home,&quot;/output/envFact.RData&quot;))</code></pre>
<pre class="r"><code># test differences between site  ####

## add IQ and Anx here 

tablevariates = c(&quot;site&quot;,&quot;Age&quot;,&quot;tsympt&quot;, &quot;Pubstat&quot;, &quot;cigday_1&quot;,&quot;contraceptives&quot;,
                  &quot;PC_1&quot;, &quot;PC_2&quot;, &quot;PC_3&quot;, &quot;PC_4&quot;)

genomicvariates = c(&quot;X260_280&quot;,&quot;ng_per_ul&quot;, &quot;Timediff_ExtrPurification&quot;,&quot;reads_per_sample&quot; ) 

res = table_sumstat(Patdata, 
                    columns=unique(c(tablevariates, clinFact,modelFact, envFact, genomicvariates) ), 
                    groupfactor = &quot;group&quot;)

res</code></pre>
<pre><code>
--------Summary descriptives table by &#39;group&#39;---------

________________________________________________________________________ 
                                 CTRL               CD         p.overall 
                                 N=51              N=51                  
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
site:                                                            0.223   
    FRA                       24 (47.1%)        14 (27.5%)               
    AAC                       10 (19.6%)        18 (35.3%)               
    BCN                       5 (9.80%)          8 (15.7%)               
    BLB                       6 (11.8%)          6 (11.8%)               
    SZG                       6 (11.8%)          5 (9.80%)               
Age                          16.1 (1.58)        15.8 (1.48)      0.295   
tsympt                       0.00 (0.00)        4.75 (2.50)     &lt;0.001   
Pubstat:                                                         0.155   
    Latepubertal              36 (70.6%)        43 (84.3%)               
    Postpubertal              15 (29.4%)         8 (15.7%)               
cigday_1                     0.51 (2.06)        6.06 (6.50)     &lt;0.001   
contraceptives:                                                  0.019   
    no                        41 (80.4%)        29 (56.9%)               
    yes                       10 (19.6%)        22 (43.1%)               
PC_1                         0.00 (0.02)        0.00 (0.02)      0.642   
PC_2                         -0.01 (0.01)       0.00 (0.02)      0.019   
PC_3                         0.00 (0.01)        0.00 (0.03)      0.592   
PC_4                         0.00 (0.02)        0.00 (0.02)      0.935   
int_dis:                                                        &lt;0.001   
    no                        46 (90.2%)        20 (39.2%)               
    yes                       5 (9.80%)         31 (60.8%)               
medication:                                                      0.001   
    no                        48 (94.1%)        33 (64.7%)               
    yes                       3 (5.88%)         18 (35.3%)               
Matsmk:                                                          0.030   
    no                        40 (81.6%)        24 (58.5%)               
    yes                       9 (18.4%)         17 (41.5%)               
Matagg:                                                          0.001   
    no                        48 (98.0%)        32 (72.7%)               
    yes                       1 (2.04%)         12 (27.3%)               
FamScore                     0.20 (0.49)        0.76 (0.81)     &lt;0.001   
EduPar                       7.97 (2.80)        6.05 (2.60)      0.002   
n_trauma                     0.84 (1.10)        2.20 (1.81)     &lt;0.001   
X260_280                     1.84 (0.02)        1.84 (0.02)      0.976   
ng_per_ul                     641 (250)          724 (296)       0.133   
Timediff_ExtrPurification    4.39 (3.00)        4.10 (3.48)      0.648   
reads_per_sample          10843361 (5589726) 9276447 (4314540)   0.116   
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<pre class="r"><code>setwd(tempdir())
export2word(res, file = paste0(Home,&quot;/output/table1_unfiltered.docx&quot;))
setwd(Home)</code></pre>
</div>
</div>
</div>
<div id="qc-filter-and-normalize" class="section level1">
<h1>QC Filter and Normalize</h1>
<div id="filter-snps" class="section level2">
<h2>filter SNPs</h2>
<pre class="r"><code>form0=paste0(&quot;~0+&quot;,paste0(modelFact, collapse=&quot;+&quot;))
designh0 = as.formula(form0)
designh1 = as.formula(paste(form0,&quot; + group&quot;))


save(designh0, file=paste0(Home,&quot;/output/designh0.RData&quot;))
save(designh1, file=paste0(Home,&quot;/output/designh1.RData&quot;))

ddsMat &lt;- DESeqDataSetFromMatrix(countData = EpiData,
                                 colData = Patdata,
                                 rowData = EpiMeta,
                                 design = designh1)</code></pre>
<pre><code>  the design formula contains one or more numeric variables with integer values,
  specifying a model with increasing fold change for higher values.
  did you mean for this to be a factor? if so, first convert
  this variable to a factor using the factor() function</code></pre>
<pre><code>  the design formula contains one or more numeric variables that have mean or
  standard deviation larger than 5 (an arbitrary threshold to trigger this message).
  it is generally a good idea to center and scale numeric variables in the design
  to improve GLM convergence.</code></pre>
<pre class="r"><code># drop overlapping with common SNPs ####

gs = with(rowData(ddsMat), GRanges(chrtochar(CHR), IRanges(start, stop)))
values(gs) = EpiMeta[,c(&quot;raw_gene&quot;,&quot;gene&quot;, &quot;distance&quot;, &quot;feature&quot;, &quot;cpg&quot;, &quot;tf_binding&quot;, &quot;samples_per_target&quot;, &quot;mean&quot;,&quot;std&quot;)]
rowRanges(ddsMat) = gs
genome(ddsMat) = &quot;hg19&quot;

GRSNP = with(SNPdatafilt, GRanges(chrtochar(CHR), IRanges(start, stop)))
genome(GRSNP) = &quot;hg19&quot;

hits = findOverlaps(GRSNP, ddsMat)


ddsMat = ddsMat[-c(hits@to),]


# drop non autosomal and X locus
autosomal_X = chrtonum(seqnames(rowRanges(ddsMat)))&lt;=23
ddsMat = ddsMat[autosomal_X,]

#Focus on 
#CpG-island, intergenic region, 2kb upstream of transcriptional start, intronic, exonic, 2kb # # downstream of transcription stop, 3’UTR, 5’UTR, splicing site

#Check annotated features 
table(rowRanges(ddsMat)$feature)</code></pre>
<pre><code>
   downstream        exonic    intergenic      intronic   ncR0_exonic 
        37061        108405        964508        871683         13107 
ncR0_intronic ncR0_splicing     ncR0_UTR3     ncR0_UTR5      splicing 
        60449             4           402           657          1929 
     upstream          UTR3          UTR5 
       102954         25722         40548 </code></pre>
<pre class="r"><code>idxkeep = rowRanges(ddsMat)$feature %in% c(&quot;downstream&quot;, &quot;exonic&quot;, &quot;intronic&quot;, &quot;splicing&quot;, &quot;upstream&quot;, &quot;UTR3&quot;, &quot;UTR5&quot;)

# drop intergenic 
# rowRanges(ddsMat)$distance_min = get_min_distance(rowRanges(ddsMat)$distance)

Nintergenig=sum(idxkeep)
ddsMat = ddsMat[idxkeep,]</code></pre>
<p>Tags excluded based on SNP location: 37 753 Tags excluded Sex or M chrom location: 7 892 Tags excluded intergenic: 1 188 302</p>
</div>
<div id="filter-reads" class="section level2">
<h2>filter reads</h2>
<pre class="r"><code># define keep Rows based on reads ####

ddsMat = estimateSizeFactors(ddsMat)

Ntot = length(ddsMat)
reads = as.data.frame(counts(ddsMat, normalized=T))

SDs = apply(reads, 1, sd)
keepvar = SDs&gt;0

# count number of identified reads &gt; 1 per group
countspergroup &lt;- data.frame(cases=rowSums(reads[,ddsMat$group ==&quot;CD&quot;]&gt;=5),
                             controls=rowSums(reads[,ddsMat$group ==&quot;CTRL&quot;]&gt;=5))

# keep if tag was detected 5 times in one group in at leat 25 samples 
keepidx = apply(countspergroup, 1, max) &gt;= 25

ddsMat &lt;- ddsMat[keepvar&amp;keepidx,]

Nfilt = length(ddsMat)</code></pre>
<p>Tags removed due to low reads: 972 200 Total tags included after filtering: 216 102</p>
</div>
<div id="final-tag-characteristics" class="section level2">
<h2>final tag characteristics</h2>
<pre class="r"><code>barplot(table(droplevels(as.data.frame(rowRanges(ddsMat))$seqnames)), main=&quot;tags per chromosome&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/tagchar-1.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>reads = counts(ddsMat) # not normalized here to see batch effects
logreads=log2(reads+1)

patsum = colSums(reads)

barplot(log2(patsum), main = &quot;total reads per sample&quot;, ylab = &quot;log2(sum(reads)&quot;, xlab = &quot;Samples&quot;, col=Dark8[as.numeric(ddsMat$group)], ylim = c(0, max(log2(patsum))*1.2))
legend(&quot;topleft&quot;, legend=levels(ddsMat$group), pch=15, col = Dark8[1:nlevels(ddsMat$group)])</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/tagchar-2.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>barplot(log2(patsum), main = &quot;total reads per sample&quot;, ylab = &quot;log2(sum(reads)&quot;, xlab = &quot;Samples&quot;, col=Dark8[as.numeric(ddsMat$site)], ylim = c(0, max(log2(patsum))*1.2))
legend(&quot;topleft&quot;, legend=levels(ddsMat$site), pch=15, col = Dark8[1:nlevels(ddsMat$site)],ncol = 2)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/tagchar-3.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dds = ddsMat
dds = estimateSizeFactors(dds)

cpm = counts(dds)
log2_cpm = log2(cpm+1)

idx = sample(1:nrow(log2_cpm), 1000)
log2_cpm_scaled = apply(log2_cpm, 2, minmax_scaling)
plot(density(log2_cpm_scaled[idx,], bw = 0.02), main = &quot;minmaxscaled 1000 random tags, min-max-scaled&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/tagchar-4.png" width="4500" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-samples" class="section level2">
<h2>filter samples</h2>
<pre class="r"><code># sample based filter ####

# calculate distance plot heatmaps based on top 20000 tags by variance####

variances = apply(log2_cpm,1, var)
set = order(variances, decreasing = T)[1:2000]

cpm.sel.trans = t(log2_cpm[set,])

distance = dist(cpm.sel.trans)
sampleDistMatrix &lt;- as.matrix(distance)

rownames(sampleDistMatrix) &lt;- rownames(Patdata)
colnames(sampleDistMatrix) &lt;- rownames(Patdata)
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)

groupcol = Dark8[1:nlevels(dds$group)]
names(groupcol) = levels(dds$group)

sitecol = Dark8[1:nlevels(dds$site)]
names(sitecol) = levels(dds$site)

ccptcol = brewer.pal(3,&quot;Set1&quot;)[1:nlevels(dds$contraceptives)]
names(ccptcol) = levels(dds$contraceptives)

ann_colors = list(
  group = groupcol, 
  contraceptives = ccptcol,
  site = sitecol)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = distance,
         clustering_distance_cols = distance,
         clustering_method = &quot;ward.D2&quot;,
         border_color = NA, 
         annotation_row = Patdata[,c(&quot;Age&quot;, &quot;site&quot;, &quot;X260_280&quot;, &quot;Timediff_ExtrPurification&quot;)],
         annotation_col = Patdata[,c(&quot;group&quot;, &quot;scrsympt&quot;,clinFact[-c(1,2)])],
         col = colors, 
         annotation_colors = ann_colors,
         main = &quot;Distances not normalized log2 counts&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/filter_samples-1.png" width="7500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#PCA filter 

gpca &lt;- glmpca(t(cpm.sel.trans), L = 2)
gpca.dat &lt;- gpca$factors
gpca.dat$site &lt;- dds$site
gpca.dat$group &lt;- dds$group
gpca.dat$ID_femNAT &lt;- dds$ID_femNAT
rownames(gpca.dat) = colnames(dds)

mds = as.data.frame(colData(dds)) %&gt;% cbind(cmdscale(distance))

# filter samples outside of 4 sd ####

idx1 = abs(gpca.dat$dim1)&gt;4*sd(gpca.dat$dim1)
idx2 = abs(gpca.dat$dim2)&gt;4*sd(gpca.dat$dim2)

exclGPCA = rownames(gpca.dat)[idx1|idx2]

idx3 = abs(mds$&quot;1&quot;)&gt;4*sd(mds$&quot;1&quot;)
idx4 = abs(mds$&quot;2&quot;)&gt;4*sd(mds$&quot;2&quot;)
exclMDS = rownames(mds)[idx3|idx4]

excl = unique(c(exclGPCA, exclMDS))

gpca.dat$drop = factor(rownames(gpca.dat) %in% excl, c(T,F), c(&quot;excluded&quot;, &quot;keep&quot;))
mds$drop  = factor(rownames(mds) %in% excl, c(T,F), c(&quot;excluded&quot;, &quot;keep&quot;))

ggplot(gpca.dat, aes(x = dim1, y = dim2, color = group, shape = drop)) +
  geom_point(size = 2)  + ggtitle(&quot;glmpca - Generalized PCA&quot;) </code></pre>
<p><img src="figure/01_Preprocessing.Rmd/filter_samples2-1.png" width="3500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(mds, aes(x = `1`, y = `2`, color = group, shape = drop)) +
  geom_point(size = 2)  + ggtitle(&quot;MDS with log2 counts&quot;) </code></pre>
<p><img src="figure/01_Preprocessing.Rmd/filter_samples2-2.png" width="3500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>keepSamples = row.names(mds)[! row.names(mds) %in% excl]

ddsMat = ddsMat[,keepSamples]

ddsMat = estimateSizeFactors(ddsMat)

distance=dist(log2(t(counts(ddsMat)+1)))

# hierarchical clustering ####

HC = hclust(distance, method = &quot;ward.D2&quot;)
plot(HC, main = &quot;clustered samples log2(counts + 1)&quot;, cex = 0.4)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/filter_samples2-3.png" width="3500" style="display: block; margin: auto;" /></p>
<pre class="r"><code># visual inspection no obvious cluster detectable
n = 2 # check outlier cluster
clusters = cutree(HC,k = n)
tab=table(clusters)

# drop if a cluster hols only 5 percent of the cohort
keeper=as.numeric(names(tab)[tab&gt;0.05*length(clusters)])

keepSamples = names(clusters[clusters %in% keeper])
dds_filt = ddsMat[,keepSamples]
dds_filt = estimateSizeFactors(dds_filt)</code></pre>
</div>
<div id="replot-cleaned-samples" class="section level2">
<h2>replot cleaned samples</h2>
<pre class="r"><code># replot with normalized and cleaned data
cpm = counts(dds_filt, normalized = T)
log2_cpm = log2(cpm+1)

distance = dist(t(log2_cpm))
sampleDistMatrix &lt;- as.matrix(distance)

rownames(sampleDistMatrix) &lt;- rownames(colData(dds_filt))
colnames(sampleDistMatrix) &lt;- rownames(colData(dds_filt))
colors &lt;- colorRampPalette( rev(brewer.pal(9, &quot;Blues&quot;)) )(255)


ann_colors = list(
  group = groupcol, 
  contraceptives = ccptcol,
  site = sitecol)

pheatmap(sampleDistMatrix,
         clustering_distance_rows = distance,
         clustering_distance_cols = distance,
         clustering_method = &quot;ward.D2&quot;,
         border_color = NA, 
         annotation_row = Patdata[,c(&quot;Age&quot;, &quot;site&quot;, &quot;X260_280&quot;, &quot;Timediff_ExtrPurification&quot;)],
         annotation_col = Patdata[,c(&quot;group&quot;, &quot;scrsympt&quot;,clinFact[-c(1,2)])],
         col = colors, 
         annotation_colors = ann_colors,
         main = &quot;Distances normalized log2 counts, after filtering&quot;)</code></pre>
<p><img src="figure/01_Preprocessing.Rmd/rplot_cleaned-1.png" width="7500" style="display: block; margin: auto;" /></p>
<p>Samples Excluded: 01-0515, 02-0211, 02-0354</p>
</div>
</div>
<div id="surrogate-variable-analysis" class="section level1">
<h1>Surrogate variable analysis</h1>
<pre class="r"><code># only tags with enough reads
idx1  &lt;- rowMeans(cpm) &gt; 1
sd = apply(log2_cpm, 1, sd)

# only tags with some variance 
idx2 = sd&gt; 0.5

cpm_subset = cpm[idx1 &amp; idx2,]

mod  &lt;- model.matrix(designh1, colData(dds_filt))
mod0 &lt;- model.matrix(designh0, colData(dds_filt))

nsv = num.sv(cpm_subset, mod, method = &quot;be&quot;)

svaset &lt;- svaseq(cpm_subset, mod, mod0, n.sv = nsv)</code></pre>
<pre><code>Number of significant surrogate variables is:  32 
Iteration (out of 5 ):1  2  3  4  5  </code></pre>
<pre class="r"><code>Patdata = cbind(colData(dds_filt), as.data.frame(svaset$sv))

colData(dds_filt) &lt;- Patdata

save(dds_filt, file= paste0(Home,&quot;/output/ProcessedData.RData&quot;))</code></pre>
<p>Total Number of Surrogate Variables extracted: 32</p>
</div>
<div id="sample-statistics-after-cleaning" class="section level1">
<h1>Sample statistics after cleaning</h1>
<pre class="r"><code>Patreads = colSums(cpm)

dds_filt$reads_per_sample_cleaned = Patreads

clPatdata = as.data.frame(colData(dds_filt))

res = table_sumstat(clPatdata, 
                    columns=unique(c(tablevariates, clinFact,modelFact, envFact, genomicvariates)), 
                    groupfactor = &quot;group&quot;)</code></pre>
<pre><code>Warning in chisq.test(xx, correct = FALSE): Chi-squared approximation may be
incorrect

Warning in chisq.test(xx, correct = FALSE): Chi-squared approximation may be
incorrect</code></pre>
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
<pre class="r"><code>res</code></pre>
<pre><code>
--------Summary descriptives table by &#39;group&#39;---------

________________________________________________________________________ 
                                 CTRL               CD         p.overall 
                                 N=50              N=49                  
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ 
site:                                                            0.164   
    FRA                       24 (48.0%)        13 (26.5%)               
    AAC                       9 (18.0%)         17 (34.7%)               
    BCN                       5 (10.0%)          8 (16.3%)               
    BLB                       6 (12.0%)          6 (12.2%)               
    SZG                       6 (12.0%)          5 (10.2%)               
Age                          16.1 (1.60)        15.8 (1.49)      0.404   
tsympt                       0.00 (0.00)        4.76 (2.50)     &lt;0.001   
Pubstat:                                                         0.170   
    Latepubertal              35 (70.0%)        41 (83.7%)               
    Postpubertal              15 (30.0%)         8 (16.3%)               
cigday_1                     0.52 (2.08)        6.14 (6.57)     &lt;0.001   
contraceptives:                                                  0.025   
    no                        40 (80.0%)        28 (57.1%)               
    yes                       10 (20.0%)        21 (42.9%)               
PC_1                         0.00 (0.02)        0.00 (0.02)      0.575   
PC_2                         -0.01 (0.01)       0.00 (0.02)      0.010   
PC_3                         0.00 (0.01)        0.00 (0.03)      0.667   
PC_4                         0.00 (0.02)        0.00 (0.02)      0.828   
int_dis:                                                        &lt;0.001   
    no                        45 (90.0%)        20 (40.8%)               
    yes                       5 (10.0%)         29 (59.2%)               
medication:                                                      0.001   
    no                        47 (94.0%)        32 (65.3%)               
    yes                       3 (6.00%)         17 (34.7%)               
Matsmk:                                                          0.071   
    no                        39 (81.2%)        24 (61.5%)               
    yes                       9 (18.8%)         15 (38.5%)               
Matagg:                                                          0.001   
    no                        47 (97.9%)        30 (71.4%)               
    yes                       1 (2.08%)         12 (28.6%)               
FamScore                     0.20 (0.49)        0.76 (0.80)     &lt;0.001   
EduPar                       8.11 (2.66)        5.94 (2.62)     &lt;0.001   
n_trauma                     0.86 (1.11)        2.16 (1.83)     &lt;0.001   
X260_280                     1.84 (0.02)        1.84 (0.02)      0.748   
ng_per_ul                     642 (252)          731 (299)       0.113   
Timediff_ExtrPurification    4.34 (3.01)        4.02 (3.38)      0.620   
reads_per_sample          11017072 (5505663) 9475335 (4256539)   0.122   
¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯¯ </code></pre>
<pre class="r"><code>setwd(tempdir())
export2word(res, file = paste0(Home,&quot;/output/table1_filtered.docx&quot;))
setwd(Home)</code></pre>
<ul>
<li>Total number of samples: 99</li>
<li>Total number of Tags: 216 102</li>
<li>Tags in TFbinding sites: 114 671</li>
<li>Tags in CpGs:
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Var1
</th>
<th style="text-align:left;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cpg
</td>
<td style="text-align:left;">
66346
</td>
</tr>
<tr>
<td style="text-align:left;">
NoCpg
</td>
<td style="text-align:left;">
149756
</td>
</tr>
</tbody>
</table></li>
<li>Tags per feature
<table class=" lightable-classic" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
Var1
</th>
<th style="text-align:left;">
Freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
downstream
</td>
<td style="text-align:left;">
7177
</td>
</tr>
<tr>
<td style="text-align:left;">
exonic
</td>
<td style="text-align:left;">
19002
</td>
</tr>
<tr>
<td style="text-align:left;">
intronic
</td>
<td style="text-align:left;">
133566
</td>
</tr>
<tr>
<td style="text-align:left;">
splicing
</td>
<td style="text-align:left;">
624
</td>
</tr>
<tr>
<td style="text-align:left;">
upstream
</td>
<td style="text-align:left;">
37979
</td>
</tr>
<tr>
<td style="text-align:left;">
UTR3
</td>
<td style="text-align:left;">
4885
</td>
</tr>
<tr>
<td style="text-align:left;">
UTR5
</td>
<td style="text-align:left;">
12869
</td>
</tr>
</tbody>
</table></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.1.1                RCircos_1.2.1              
 [3] compareGroups_4.4.6         cluster_2.1.0              
 [5] kableExtra_1.3.1            knitr_1.30                 
 [7] glmpca_0.2.0                sva_3.38.0                 
 [9] BiocParallel_1.24.1         genefilter_1.72.0          
[11] mgcv_1.8-33                 nlme_3.1-151               
[13] pheatmap_1.0.12             vsn_3.58.0                 
[15] DESeq2_1.30.0               SummarizedExperiment_1.20.0
[17] Biobase_2.50.0              MatrixGenerics_1.2.0       
[19] matrixStats_0.57.0          GenomicRanges_1.42.0       
[21] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[23] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[25] forcats_0.5.0               stringr_1.4.0              
[27] dplyr_1.0.2                 purrr_0.3.4                
[29] readr_1.4.0                 tidyr_1.1.2                
[31] tibble_3.0.4                tidyverse_1.3.0            
[33] corrplot_0.84               ggplot2_3.3.3              
[35] gplots_3.1.1                RColorBrewer_1.1-2         
[37] data.table_1.13.6           workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] uuid_0.1-4             readxl_1.3.1           backports_1.2.0       
  [4] systemfonts_0.3.2      splines_4.0.3          digest_0.6.27         
  [7] htmltools_0.5.1.1      fansi_0.4.1            magrittr_2.0.1        
 [10] Rsolnp_1.16            memoise_2.0.0          limma_3.46.0          
 [13] annotate_1.68.0        modelr_0.1.8           officer_0.3.16        
 [16] colorspace_2.0-0       blob_1.2.1             rvest_0.3.6           
 [19] haven_2.3.1            xfun_0.20              crayon_1.3.4          
 [22] RCurl_1.98-1.2         jsonlite_1.7.2         survival_3.2-7        
 [25] glue_1.4.2             gtable_0.3.0           zlibbioc_1.36.0       
 [28] XVector_0.30.0         webshot_0.5.2          DelayedArray_0.16.0   
 [31] DBI_1.1.1              edgeR_3.32.1           Rcpp_1.0.5            
 [34] viridisLite_0.3.0      xtable_1.8-4           bit_4.0.4             
 [37] preprocessCore_1.52.1  truncnorm_1.0-8        httr_1.4.2            
 [40] ellipsis_0.3.1         mice_3.12.0            farver_2.0.3          
 [43] pkgconfig_2.0.3        XML_3.99-0.5           dbplyr_2.0.0          
 [46] locfit_1.5-9.4         labeling_0.4.2         tidyselect_1.1.0      
 [49] rlang_0.4.10           later_1.1.0.1          AnnotationDbi_1.52.0  
 [52] munsell_0.5.0          cellranger_1.1.0       tools_4.0.3           
 [55] cachem_1.0.1           cli_2.2.0              generics_0.1.0        
 [58] RSQLite_2.2.2          broom_0.7.3            evaluate_0.14         
 [61] fastmap_1.1.0          yaml_2.2.1             bit64_4.0.5           
 [64] fs_1.5.0               zip_2.1.1              caTools_1.18.1        
 [67] whisker_0.4            xml2_1.3.2             compiler_4.0.3        
 [70] rstudioapi_0.13        affyio_1.60.0          reprex_1.0.0          
 [73] geneplotter_1.68.0     stringi_1.5.3          HardyWeinberg_1.7.1   
 [76] highr_0.8              ps_1.5.0               gdtools_0.2.3         
 [79] lattice_0.20-41        Matrix_1.2-18          vctrs_0.3.6           
 [82] pillar_1.4.7           lifecycle_0.2.0        BiocManager_1.30.10   
 [85] flextable_0.6.2        bitops_1.0-6           httpuv_1.5.5          
 [88] R6_2.5.0               affy_1.68.0            promises_1.1.1        
 [91] KernSmooth_2.23-18     writexl_1.3.1          MASS_7.3-53           
 [94] gtools_3.8.2           assertthat_0.2.1       chron_2.3-56          
 [97] rprojroot_2.0.2        withr_2.4.1            GenomeInfoDbData_1.2.4
[100] hms_1.0.0              grid_4.0.3             rmarkdown_2.6         
[103] git2r_0.28.0           base64enc_0.1-3        lubridate_1.7.9.2     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
