<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AG Chiocchetti" />


<title>The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">femNATCD_MethSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/achiocch/femNATCD_MethSeq">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</h1>
<h3 class="subtitle">Differential Methylation Analysis</h3>
<h4 class="author">AG Chiocchetti</h4>
<h4 class="date">29 Dezember 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-04-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>femNATCD_MethSeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree4dee231f75db4afbace8606bf75213f739dd24adtargetblank4dee231a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/4dee231f75db4afbace8606bf75213f739dd24ad" target="_blank">4dee231</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree4dee231f75db4afbace8606bf75213f739dd24adtargetblank4dee231a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/4dee231f75db4afbace8606bf75213f739dd24ad" target="_blank">4dee231</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/Epicounts.csv
    Ignored:    data/Epimeta.csv
    Ignored:    data/Epitpm.csv
    Ignored:    data/KangUnivers.txt
    Ignored:    data/Kang_DataPreprocessing.RData
    Ignored:    data/Kang_dataset_genesMod_version2.txt
    Ignored:    data/PatMeta.csv
    Ignored:    data/ProcessedData.RData
    Ignored:    data/SNPCommonFilt.csv
    Ignored:    output/4A76FA10
    Ignored:    output/DMR_Results.csv
    Ignored:    output/GOres.xlsx
    Ignored:    output/LME_Results.csv
    Ignored:    output/LME_Results_Sig.csv
    Ignored:    output/LME_Results_Sig_mod.csv
    Ignored:    output/ProcessedData.RData
    Ignored:    output/SEM_summary_groupEpi_M15.txt
    Ignored:    output/SEM_summary_groupEpi_M2.txt
    Ignored:    output/SEM_summary_groupEpi_M_all.txt
    Ignored:    output/SEM_summary_groupEpi_TopHit.txt
    Ignored:    output/SEM_summary_groupEpi_all.txt
    Ignored:    output/SEMplot_Epi_M15.html
    Ignored:    output/SEMplot_Epi_M15.png
    Ignored:    output/SEMplot_Epi_M15_files/
    Ignored:    output/SEMplot_Epi_M2.html
    Ignored:    output/SEMplot_Epi_M2.png
    Ignored:    output/SEMplot_Epi_M2_files/
    Ignored:    output/SEMplot_Epi_M_all.html
    Ignored:    output/SEMplot_Epi_M_all.png
    Ignored:    output/SEMplot_Epi_M_all_files/
    Ignored:    output/SEMplot_Epi_TopHit.html
    Ignored:    output/SEMplot_Epi_TopHit.png
    Ignored:    output/SEMplot_Epi_TopHit_files/
    Ignored:    output/SEMplot_Epi_all.html
    Ignored:    output/SEMplot_Epi_all.png
    Ignored:    output/SEMplot_Epi_all_files/
    Ignored:    output/circos_DMR_tags.svg
    Ignored:    output/circos_LME_tags.svg
    Ignored:    output/clinFact.RData
    Ignored:    output/dds_filt_analyzed.RData
    Ignored:    output/designh0.RData
    Ignored:    output/designh1.RData
    Ignored:    output/envFact.RData
    Ignored:    output/gostres.pdf
    Ignored:    output/modelFact.RData
    Ignored:    output/resdmr.RData
    Ignored:    output/resultsdmr_table.RData
    Ignored:    output/table1_filtered.Rmd
    Ignored:    output/table1_filtered.docx
    Ignored:    output/table1_unfiltered.Rmd
    Ignored:    output/table1_unfiltered.docx
    Ignored:    setup_built.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/02_2_Main_DMR.Rmd</code>) and HTML (<code>docs/02_2_Main_DMR.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/91de221841561ea6b43fb67c9fd06d813bdb04b4/docs/02_2_Main_DMR.html" target="_blank">91de221</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/4ea1bba17a570a8d27e65e5b4153833c4d999e77/docs/02_2_Main_DMR.html" target="_blank">4ea1bba</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/6c21638381f6c8e162516fa9f9278f8e82503ab4/analysis/02_2_Main_DMR.Rmd" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/6c21638381f6c8e162516fa9f9278f8e82503ab4/docs/02_2_Main_DMR.html" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>## load Data  ####
load(paste0(Home,&quot;/output/dds_filt_analyzed.RData&quot;))</code></pre>
<div id="bumphunting" class="section level1">
<h1>bumphunting</h1>
<pre class="r"><code>ranges=as.data.frame(rowRanges(dds_filt))
log_cpm = log2(counts(dds_filt, normalized=T)+1)

BootN = 250
GapSize = 500 


if (reanalyze){
    cl &lt;- clusterMaker(ranges$seqnames, ranges$start, maxGap = GapSize)
    tab=table(cl)
    designmatrix = model.matrix(design(dds_filt), colData(dds_filt))
    resdmr = bumphunter(log_cpm, 
                        design=designmatrix,
                        pos=ranges$start,
                        chr=ranges$seqnames,
                        coef=ncol(designmatrix), cluster=cl, cutoff=0.5, nullMethod = &quot;bootstrap&quot;, B=BootN)
    
    bumphunter:::foreachCleanup()
    
    save(resdmr, file=paste0(Home,&quot;/output/resdmr.RData&quot;)) 
} else {
  if(file.exists(paste0(Home,&quot;/output/resdmr.RData&quot;))){
    load(paste0(Home,&quot;/output/resdmr.RData&quot;))
  } else {
    cl &lt;- clusterMaker(ranges$seqnames, ranges$start, maxGap = GapSize)
    tab=table(cl)
    designmatrix = model.matrix(design(dds_filt), colData(dds_filt))
    resdmr = bumphunter(log_cpm, 
                        design=designmatrix,
                        pos=ranges$start,
                        chr=ranges$seqnames,
                        coef=ncol(designmatrix), cluster=cl, cutoff=0.5, nullMethod = &quot;bootstrap&quot;, B=BootN)
    
  }
}</code></pre>
<pre><code>[bumphunterEngine] Parallelizing using 6 workers/cores (backend: doParallelSNOW, version: 1.0.16).</code></pre>
<pre><code>[bumphunterEngine] Computing coefficients.</code></pre>
<pre><code>[bumphunterEngine] Performing 250 bootstraps.</code></pre>
<pre><code>[bumphunterEngine] Computing marginal bootstrap p-values.</code></pre>
<pre><code>[bumphunterEngine] cutoff: 0.5</code></pre>
<pre><code>[bumphunterEngine] Finding regions.</code></pre>
<pre><code>[bumphunterEngine] Found 8108 bumps.</code></pre>
<pre><code>[bumphunterEngine] Computing regions for each bootstrap.</code></pre>
<pre><code>[bumphunterEngine] Estimating p-values and FWER.</code></pre>
<pre class="r"><code>subject = annotateTranscripts(txdb = TxDb.Hsapiens.UCSC.hg19.knownGene, 
                              by=&quot;gene&quot;)</code></pre>
<pre><code>No annotationPackage supplied. Trying org.Hs.eg.db.</code></pre>
<pre><code>Lade nötiges Paket: org.Hs.eg.db</code></pre>
<pre><code></code></pre>
<pre><code>Getting TSS and TSE.</code></pre>
<pre><code>  403 genes were dropped because they have exons located on both strands
  of the same reference sequence or on more than one reference sequence,
  so cannot be represented by a single genomic range.
  Use &#39;single.strand.genes.only=FALSE&#39; to get all the genes in a
  GRangesList object, or use suppressMessages() to suppress this message.</code></pre>
<pre><code>Getting CSS and CSE.</code></pre>
<pre><code>Getting exons.</code></pre>
<pre><code>Annotating genes.</code></pre>
<pre class="r"><code>resultsdmr_table = resdmr$table

subject = subject[subject@seqnames %in% unique(resultsdmr_table$chr),]

chck = matchGenes(resultsdmr_table, subject, type =&quot;any&quot;, promoterDist = 2000)</code></pre>
<pre><code>.................................................................................</code></pre>
<pre class="r"><code>resultsdmr_table = cbind(resultsdmr_table, chck)

display_tab_simple(head(resultsdmr_table, n=10))</code></pre>
<table class=" lightable-classic" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
chr
</th>
<th style="text-align:left;">
start
</th>
<th style="text-align:left;">
end
</th>
<th style="text-align:left;">
value
</th>
<th style="text-align:left;">
area
</th>
<th style="text-align:left;">
cluster
</th>
<th style="text-align:left;">
indexStart
</th>
<th style="text-align:left;">
indexEnd
</th>
<th style="text-align:left;">
L
</th>
<th style="text-align:left;">
clusterL
</th>
<th style="text-align:left;">
p.value
</th>
<th style="text-align:left;">
fwer
</th>
<th style="text-align:left;">
p.valueArea
</th>
<th style="text-align:left;">
fwerArea
</th>
<th style="text-align:left;">
name
</th>
<th style="text-align:left;">
annotation
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
region
</th>
<th style="text-align:left;">
distance
</th>
<th style="text-align:left;">
subregion
</th>
<th style="text-align:left;">
insideDistance
</th>
<th style="text-align:left;">
exonnumber
</th>
<th style="text-align:left;">
nexons
</th>
<th style="text-align:left;">
UTR
</th>
<th style="text-align:left;">
strand
</th>
<th style="text-align:left;">
geneL
</th>
<th style="text-align:left;">
codingL
</th>
<th style="text-align:left;">
Geneid
</th>
<th style="text-align:left;">
subjectHits
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
3194
</td>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:left;">
24561792
</td>
<td style="text-align:left;">
24561792
</td>
<td style="text-align:left;">
-1.365
</td>
<td style="text-align:left;">
1.365
</td>
<td style="text-align:left;">
16970
</td>
<td style="text-align:left;">
35466
</td>
<td style="text-align:left;">
35466
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.268
</td>
<td style="text-align:left;">
0.004
</td>
<td style="text-align:left;">
0.980
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
close to 3’
</td>
<td style="text-align:left;">
close to 3’
</td>
<td style="text-align:left;">
1134
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
73
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
100616448
</td>
<td style="text-align:left;">
1791
</td>
</tr>
<tr>
<td style="text-align:left;">
6414
</td>
<td style="text-align:left;">
chr16
</td>
<td style="text-align:left;">
87636490
</td>
<td style="text-align:left;">
87636843
</td>
<td style="text-align:left;">
-0.647
</td>
<td style="text-align:left;">
1.941
</td>
<td style="text-align:left;">
75725
</td>
<td style="text-align:left;">
158341
</td>
<td style="text-align:left;">
158343
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.420
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.464
</td>
<td style="text-align:left;">
JPH3
</td>
<td style="text-align:left;">
NM_001271604 NM_001271605 NM_020655 NP_001258533 NP_001258534 NP_065706 NR_073379
</td>
<td style="text-align:left;">
overlaps exon upstream
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
1049
</td>
<td style="text-align:left;">
overlaps exon upstream
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
overlaps 5’ UTR
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
96320
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
57338
</td>
<td style="text-align:left;">
15750
</td>
</tr>
<tr>
<td style="text-align:left;">
6516
</td>
<td style="text-align:left;">
chr17
</td>
<td style="text-align:left;">
5403095
</td>
<td style="text-align:left;">
5403469
</td>
<td style="text-align:left;">
-0.640
</td>
<td style="text-align:left;">
1.919
</td>
<td style="text-align:left;">
76781
</td>
<td style="text-align:left;">
160823
</td>
<td style="text-align:left;">
160825
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.444
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.488
</td>
<td style="text-align:left;">
LOC728392
</td>
<td style="text-align:left;">
NM_001162371 NP_001155843
</td>
<td style="text-align:left;">
overlaps two exons
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
850
</td>
<td style="text-align:left;">
overlaps two exons
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
3’UTR
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
1572
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
728392
</td>
<td style="text-align:left;">
18663
</td>
</tr>
<tr>
<td style="text-align:left;">
5076
</td>
<td style="text-align:left;">
chr10
</td>
<td style="text-align:left;">
135088739
</td>
<td style="text-align:left;">
135088739
</td>
<td style="text-align:left;">
-1.289
</td>
<td style="text-align:left;">
1.289
</td>
<td style="text-align:left;">
52539
</td>
<td style="text-align:left;">
109119
</td>
<td style="text-align:left;">
109119
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.472
</td>
<td style="text-align:left;">
0.006
</td>
<td style="text-align:left;">
0.996
</td>
<td style="text-align:left;">
ADAM8
</td>
<td style="text-align:left;">
NM_001109 NM_001164489 NM_001164490 NP_001100 NP_001157961 NP_001157962 XM_011539117 XM_017015465 XM_017015466 XP_011537419 XP_016870954 XP_016870955
</td>
<td style="text-align:left;">
inside intron
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
1668
</td>
<td style="text-align:left;">
inside intron
</td>
<td style="text-align:left;">
248
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
inside transcription region
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
14487
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
101
</td>
<td style="text-align:left;">
2152
</td>
</tr>
<tr>
<td style="text-align:left;">
2559
</td>
<td style="text-align:left;">
chr1
</td>
<td style="text-align:left;">
68151243
</td>
<td style="text-align:left;">
68151858
</td>
<td style="text-align:left;">
-0.537
</td>
<td style="text-align:left;">
1.610
</td>
<td style="text-align:left;">
4738
</td>
<td style="text-align:left;">
10048
</td>
<td style="text-align:left;">
10050
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
0.000
</td>
<td style="text-align:left;">
0.756
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.856
</td>
<td style="text-align:left;">
GADD45A
</td>
<td style="text-align:left;">
NM_001199741 NM_001199742 NM_001924 NP_001186670 NP_001186671 NP_001915
</td>
<td style="text-align:left;">
covers exon(s)
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
383
</td>
<td style="text-align:left;">
covers exon(s)
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
inside transcription region
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
3161
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1647
</td>
<td style="text-align:left;">
5589
</td>
</tr>
<tr>
<td style="text-align:left;">
3396
</td>
<td style="text-align:left;">
chr3
</td>
<td style="text-align:left;">
195943548
</td>
<td style="text-align:left;">
195943548
</td>
<td style="text-align:left;">
-1.192
</td>
<td style="text-align:left;">
1.192
</td>
<td style="text-align:left;">
21455
</td>
<td style="text-align:left;">
44421
</td>
<td style="text-align:left;">
44421
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.808
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
1.000
</td>
<td style="text-align:left;">
SLC51A
</td>
<td style="text-align:left;">
NM_152672 NP_689885
</td>
<td style="text-align:left;">
inside exon
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
165
</td>
<td style="text-align:left;">
inside exon
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
5’ UTR
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
16918
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
200931
</td>
<td style="text-align:left;">
6115
</td>
</tr>
<tr>
<td style="text-align:left;">
5489
</td>
<td style="text-align:left;">
chr12
</td>
<td style="text-align:left;">
50348045
</td>
<td style="text-align:left;">
50348422
</td>
<td style="text-align:left;">
-0.778
</td>
<td style="text-align:left;">
1.555
</td>
<td style="text-align:left;">
60025
</td>
<td style="text-align:left;">
125369
</td>
<td style="text-align:left;">
125370
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.812
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.896
</td>
<td style="text-align:left;">
AQP2
</td>
<td style="text-align:left;">
NM_000486 NP_000477
</td>
<td style="text-align:left;">
overlaps two exons
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
3521
</td>
<td style="text-align:left;">
overlaps two exons
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
inside transcription region
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
8140
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
359
</td>
<td style="text-align:left;">
10625
</td>
</tr>
<tr>
<td style="text-align:left;">
7597
</td>
<td style="text-align:left;">
chr21
</td>
<td style="text-align:left;">
34696824
</td>
<td style="text-align:left;">
34696854
</td>
<td style="text-align:left;">
-0.776
</td>
<td style="text-align:left;">
1.552
</td>
<td style="text-align:left;">
93519
</td>
<td style="text-align:left;">
199373
</td>
<td style="text-align:left;">
199374
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.812
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.896
</td>
<td style="text-align:left;">
IFNAR1
</td>
<td style="text-align:left;">
NM_000629 NM_001384498 NM_001384499 NM_001384500 NM_001384501 NM_001384502 NM_001384503 NM_001384504 NP_000620 NP_001371427 NP_001371428 NP_001371429 NP_001371430 NP_001371431 NP_001371432 NP_001371433 XM_011529552 XP_011527854
</td>
<td style="text-align:left;">
inside exon
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
90
</td>
<td style="text-align:left;">
inside exon
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
5’ UTR
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
35394
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
3454
</td>
<td style="text-align:left;">
10411
</td>
</tr>
<tr>
<td style="text-align:left;">
5983
</td>
<td style="text-align:left;">
chr15
</td>
<td style="text-align:left;">
31652488
</td>
<td style="text-align:left;">
31652665
</td>
<td style="text-align:left;">
-0.768
</td>
<td style="text-align:left;">
1.536
</td>
<td style="text-align:left;">
68659
</td>
<td style="text-align:left;">
143276
</td>
<td style="text-align:left;">
143277
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.824
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.904
</td>
<td style="text-align:left;">
KLF13
</td>
<td style="text-align:left;">
NM_001302461 NM_015995 NP_001289390 NP_057079 NR_033741
</td>
<td style="text-align:left;">
inside intron
</td>
<td style="text-align:left;">
inside
</td>
<td style="text-align:left;">
33405
</td>
<td style="text-align:left;">
inside intron
</td>
<td style="text-align:left;">
11547
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
inside transcription region
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
51019
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
51621
</td>
<td style="text-align:left;">
13653
</td>
</tr>
<tr>
<td style="text-align:left;">
5730
</td>
<td style="text-align:left;">
chr13
</td>
<td style="text-align:left;">
88324376
</td>
<td style="text-align:left;">
88324376
</td>
<td style="text-align:left;">
-1.187
</td>
<td style="text-align:left;">
1.187
</td>
<td style="text-align:left;">
64598
</td>
<td style="text-align:left;">
134778
</td>
<td style="text-align:left;">
134778
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
0.001
</td>
<td style="text-align:left;">
0.828
</td>
<td style="text-align:left;">
0.010
</td>
<td style="text-align:left;">
1.000
</td>
<td style="text-align:left;">
SLITRK5
</td>
<td style="text-align:left;">
NM_001384609 NM_001384610 NM_015567 NP_001371538 NP_001371539 NP_056382
</td>
<td style="text-align:left;">
promoter
</td>
<td style="text-align:left;">
promoter
</td>
<td style="text-align:left;">
494
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;">
7000
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
26050
</td>
<td style="text-align:left;">
8052
</td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>Nsig_region = sum(resultsdmr_table$fwer&lt;=0.2)

Sighits = resultsdmr_table[resultsdmr_table$fwer&lt;=0.2,]

save(resultsdmr_table, file=paste0(Home,&quot;/output/resultsdmr_table.RData&quot;))
save(resdmr, file=paste0(Home,&quot;/output/resdmr.RData&quot;))

res_dmr_filtered = resultsdmr_table %&gt;% dplyr::select(-annotation)
write.csv(resultsdmr_table, file = paste0(Home,&quot;/output/DMR_Results.csv&quot;), row.names = T)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] org.Hs.eg.db_3.12.0                    
 [2] RCircos_1.2.1                          
 [3] compareGroups_4.4.6                    
 [4] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [5] GenomicFeatures_1.42.1                 
 [6] AnnotationDbi_1.52.0                   
 [7] bumphunter_1.32.0                      
 [8] locfit_1.5-9.4                         
 [9] doParallel_1.0.16                      
[10] iterators_1.0.13                       
[11] foreach_1.5.1                          
[12] kableExtra_1.3.1                       
[13] forcats_0.5.0                          
[14] stringr_1.4.0                          
[15] dplyr_1.0.2                            
[16] purrr_0.3.4                            
[17] readr_1.4.0                            
[18] tidyr_1.1.2                            
[19] tibble_3.0.4                           
[20] ggplot2_3.3.3                          
[21] tidyverse_1.3.0                        
[22] DESeq2_1.30.0                          
[23] SummarizedExperiment_1.20.0            
[24] Biobase_2.50.0                         
[25] MatrixGenerics_1.2.0                   
[26] matrixStats_0.57.0                     
[27] GenomicRanges_1.42.0                   
[28] GenomeInfoDb_1.26.2                    
[29] IRanges_2.24.1                         
[30] S4Vectors_0.28.1                       
[31] BiocGenerics_0.36.0                    
[32] RColorBrewer_1.1-2                     
[33] knitr_1.30                             
[34] workflowr_1.6.2                        

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             uuid_0.1-4               backports_1.2.0         
  [4] systemfonts_0.3.2        BiocFileCache_1.14.0     splines_4.0.3           
  [7] BiocParallel_1.24.1      digest_0.6.27            htmltools_0.5.1.1       
 [10] fansi_0.4.1              magrittr_2.0.1           Rsolnp_1.16             
 [13] memoise_2.0.0            Biostrings_2.58.0        annotate_1.68.0         
 [16] modelr_0.1.8             officer_0.3.16           askpass_1.1             
 [19] prettyunits_1.1.1        colorspace_2.0-0         blob_1.2.1              
 [22] rvest_0.3.6              rappdirs_0.3.1           haven_2.3.1             
 [25] xfun_0.20                crayon_1.3.4             RCurl_1.98-1.2          
 [28] jsonlite_1.7.2           genefilter_1.72.0        survival_3.2-7          
 [31] glue_1.4.2               gtable_0.3.0             zlibbioc_1.36.0         
 [34] XVector_0.30.0           webshot_0.5.2            DelayedArray_0.16.0     
 [37] scales_1.1.1             DBI_1.1.1                rngtools_1.5            
 [40] Rcpp_1.0.5               viridisLite_0.3.0        xtable_1.8-4            
 [43] progress_1.2.2           bit_4.0.4                truncnorm_1.0-8         
 [46] httr_1.4.2               ellipsis_0.3.1           mice_3.12.0             
 [49] pkgconfig_2.0.3          XML_3.99-0.5             dbplyr_2.0.0            
 [52] tidyselect_1.1.0         rlang_0.4.10             later_1.1.0.1           
 [55] munsell_0.5.0            cellranger_1.1.0         tools_4.0.3             
 [58] cachem_1.0.1             cli_2.2.0                generics_0.1.0          
 [61] RSQLite_2.2.2            broom_0.7.3              evaluate_0.14           
 [64] fastmap_1.1.0            yaml_2.2.1               bit64_4.0.5             
 [67] fs_1.5.0                 zip_2.1.1                doRNG_1.8.2             
 [70] whisker_0.4              xml2_1.3.2               biomaRt_2.46.2          
 [73] compiler_4.0.3           rstudioapi_0.13          curl_4.3                
 [76] reprex_1.0.0             geneplotter_1.68.0       stringi_1.5.3           
 [79] HardyWeinberg_1.7.1      highr_0.8                ps_1.5.0                
 [82] gdtools_0.2.3            lattice_0.20-41          Matrix_1.2-18           
 [85] vctrs_0.3.6              pillar_1.4.7             lifecycle_0.2.0         
 [88] data.table_1.13.6        bitops_1.0-6             flextable_0.6.2         
 [91] httpuv_1.5.5             rtracklayer_1.49.5       R6_2.5.0                
 [94] promises_1.1.1           writexl_1.3.1            codetools_0.2-18        
 [97] assertthat_0.2.1         chron_2.3-56             openssl_1.4.3           
[100] rprojroot_2.0.2          withr_2.4.1              GenomicAlignments_1.26.0
[103] Rsamtools_2.6.0          GenomeInfoDbData_1.2.4   hms_1.0.0               
[106] grid_4.0.3               rmarkdown_2.6            git2r_0.28.0            
[109] base64enc_0.1-3          lubridate_1.7.9.2       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
