<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AG Chiocchetti" />


<title>The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.3/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/vis-4.20.1/vis.css" rel="stylesheet" />
<script src="site_libs/vis-4.20.1/vis.min.js"></script>
<script src="site_libs/visNetwork-binding-2.0.9/visNetwork.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">femNATCD_MethSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/achiocch/femNATCD_MethSeq">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</h1>
<h3 class="subtitle">Post-hoc Analyses</h3>
<h4 class="author">AG Chiocchetti</h4>
<h4 class="date">29 Dezember 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>femNATCD_MethSeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtreed3629d5f21b4307bf4b789f9e72542404fe10912targetblankd3629d5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/d3629d5f21b4307bf4b789f9e72542404fe10912" target="_blank">d3629d5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtreed3629d5f21b4307bf4b789f9e72542404fe10912targetblankd3629d5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/d3629d5f21b4307bf4b789f9e72542404fe10912" target="_blank">d3629d5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/Epicounts.csv
    Ignored:    data/Epimeta.csv
    Ignored:    data/Epitpm.csv
    Ignored:    data/KangUnivers.txt
    Ignored:    data/Kang_DataPreprocessing.RData
    Ignored:    data/Kang_dataset_genesMod_version2.txt
    Ignored:    data/PatMeta.csv
    Ignored:    data/ProcessedData.RData
    Ignored:    data/RTrawdata/
    Ignored:    data/SNPCommonFilt.csv
    Ignored:    data/femNAT_PC20.txt
    Ignored:    output/4A76FA10
    Ignored:    output/BrainMod_Enrichemnt.pdf
    Ignored:    output/Brain_Module_Heatmap.pdf
    Ignored:    output/DMR_Results.csv
    Ignored:    output/GOres.xlsx
    Ignored:    output/LME_GOplot.pdf
    Ignored:    output/LME_GOplot.svg
    Ignored:    output/LME_Results.csv
    Ignored:    output/LME_Results_Sig.csv
    Ignored:    output/LME_Results_Sig_mod.csv
    Ignored:    output/LME_tophit.svg
    Ignored:    output/ProcessedData.RData
    Ignored:    output/RNAvsMETplots.pdf
    Ignored:    output/Regions_GOplot.pdf
    Ignored:    output/Regions_GOplot.svg
    Ignored:    output/ResultsgroupComp.txt
    Ignored:    output/SEM_summary_groupEpi_M15.txt
    Ignored:    output/SEM_summary_groupEpi_M2.txt
    Ignored:    output/SEM_summary_groupEpi_M_all.txt
    Ignored:    output/SEM_summary_groupEpi_TopHit.txt
    Ignored:    output/SEM_summary_groupEpi_all.txt
    Ignored:    output/SEMplot_Epi_M15.html
    Ignored:    output/SEMplot_Epi_M15.png
    Ignored:    output/SEMplot_Epi_M15_files/
    Ignored:    output/SEMplot_Epi_M2.html
    Ignored:    output/SEMplot_Epi_M2.png
    Ignored:    output/SEMplot_Epi_M2_files/
    Ignored:    output/SEMplot_Epi_M_all.html
    Ignored:    output/SEMplot_Epi_M_all.png
    Ignored:    output/SEMplot_Epi_M_all_files/
    Ignored:    output/SEMplot_Epi_TopHit.html
    Ignored:    output/SEMplot_Epi_TopHit.png
    Ignored:    output/SEMplot_Epi_TopHit_files/
    Ignored:    output/SEMplot_Epi_all.html
    Ignored:    output/SEMplot_Epi_all.png
    Ignored:    output/SEMplot_Epi_all_files/
    Ignored:    output/barplots.pdf
    Ignored:    output/circos_DMR_tags.svg
    Ignored:    output/circos_LME_tags.svg
    Ignored:    output/clinFact.RData
    Ignored:    output/dds_filt_analyzed.RData
    Ignored:    output/designh0.RData
    Ignored:    output/designh1.RData
    Ignored:    output/envFact.RData
    Ignored:    output/functional_Enrichemnt.pdf
    Ignored:    output/gostres.pdf
    Ignored:    output/modelFact.RData
    Ignored:    output/resdmr.RData
    Ignored:    output/resultsdmr_table.RData
    Ignored:    output/table1_filtered.Rmd
    Ignored:    output/table1_filtered.docx
    Ignored:    output/table1_unfiltered.Rmd
    Ignored:    output/table1_unfiltered.docx
    Ignored:    setup_built.R

Unstaged changes:
    Deleted:    analysis/RNAvsMETplots.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd</code>) and HTML (<code>docs/03_01_Posthoc_analyses_Gene_Enrichment.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/d3629d5f21b4307bf4b789f9e72542404fe10912/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">d3629d5</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/d3629d5f21b4307bf4b789f9e72542404fe10912/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">d3629d5</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/d761be40b9985b450720b472375b01adc12b63d6/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">d761be4</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b452d2f02a4bb8e5c4dd961ce50f42a703696053/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/b452d2f02a4bb8e5c4dd961ce50f42a703696053/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/1a9f36fde268d7bd9b5615ff46b35309442db585/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">1a9f36f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
reviewed analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2734c4e51aa341b50f59acf0df2900ab91d06654/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">2734c4e</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/a8478239ce3ca497da23250c261c259b1d1f9088/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">a847823</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/9cc52f7bc59be3ada26bac3a9458fb6dbb154395/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">9cc52f7</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/158d0b43dc3337bef41cfd99c32e3322750945f4/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">158d0b4</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/0f262d1b920469dca52216115355a20663827e5e/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">0f262d1</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/5167b90e01239a0adae727c7561e97a2ae16a750/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">5167b90</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/05aac7fc7613833ecedbf0ff9dd4fd18088608fc/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">05aac7f</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/5f070a5e95490f217b935168430e427c7617b2c3/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">5f070a5</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-23
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f5c52650307ec764d17e5f70981ad2cab2dc514b/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">f5c5265</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/dc9e069fc26e9867005f94c140185f870dca8306/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">dc9e069</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/dc9e069fc26e9867005f94c140185f870dca8306/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">dc9e069</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/17f1eec7fe797b93bcf724547cf86ce90c52e37f/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">17f1eec</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/4dee231f75db4afbace8606bf75213f739dd24ad/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">4dee231</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-10
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/91de221841561ea6b43fb67c9fd06d813bdb04b4/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">91de221</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b6c6b33c63a1e7fcceeb14961390eedf6a45dc19/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">b6c6b33</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
updated GO function, and model def
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/4ea1bba17a570a8d27e65e5b4153833c4d999e77/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">4ea1bba</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/6c21638381f6c8e162516fa9f9278f8e82503ab4/analysis/03_01_Posthoc_analyses_Gene_Enrichment.Rmd" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/6c21638381f6c8e162516fa9f9278f8e82503ab4/docs/03_01_Posthoc_analyses_Gene_Enrichment.html" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="sensitivity-main-hit" class="section level2">
<h2>Sensitivity main hit</h2>
<p>For the most significant tag of interest (5’ of the SLITRK5 gene), we tested if the group effect is stable if correcting for Ethnicity (PC1-PC4) or CD associated environmental risk factors.</p>
<pre class="r"><code>tophit=which.min(results_Deseq$padj)
methdata=log2_cpm[tophit,]
Probdat=as.data.frame(colData(dds_filt))
Probdat$topHit=methdata[rownames(Probdat)]


model0=as.character(design(dds_filt))[2]
model0=as.formula(gsub(&quot;0&quot;, &quot;topHit ~ 0 + (1|site)&quot;, model0))

lmres=lmer(model0, data=Probdat)</code></pre>
<pre><code>boundary (singular) fit: see ?isSingular</code></pre>
<pre class="r"><code>lmrescoeff = as.data.frame(coefficients(summary(lmres)))
lmrescoeff$pval = dt(lmrescoeff$&quot;t value&quot;, df=length(Probdat)-1)


totestpar=c(&quot;PC_1&quot;, &quot;PC_2&quot;, &quot;PC_3&quot;, &quot;PC_4&quot;, envFact)

ressens=data.frame(matrix(nrow = length(totestpar)+1, ncol=c(3)))
colnames(ressens) = c(&quot;beta&quot;, &quot;se&quot;, &quot;p.value&quot;)
rownames(ressens) = c(&quot;original&quot;, totestpar)

ressens[&quot;original&quot;,] = lmrescoeff[&quot;groupCD&quot;, c(&quot;Estimate&quot;, &quot;Std. Error&quot;, &quot;pval&quot;)]

for( parm in totestpar){
  modelpar=as.character(design(dds_filt))[2]
  modelpar=as.formula(gsub(&quot;0&quot;, paste0(&quot;topHit ~ 0 + (1|site) +&quot;,parm), modelpar))
  lmres=lmer(modelpar, data=Probdat)
  lmrescoeff = as.data.frame(coefficients(summary(lmres)))
  lmrescoeff$pval = dt(lmrescoeff$&quot;t value&quot;, df=length(Probdat)-1)
  ressens[parm,] = lmrescoeff[&quot;groupCD&quot;, c(&quot;Estimate&quot;, &quot;Std. Error&quot;, &quot;pval&quot;)] 
}</code></pre>
<pre><code>boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular
boundary (singular) fit: see ?isSingular</code></pre>
<pre class="r"><code>a = barplot(height = ressens$beta, 
            ylim=rev(range(c(0,ressens$beta-ressens$se)))*1.3, 
            names.arg = rownames(ressens), col=Set3, border = NA, las=3, 
            ylab=&quot;beta[se]&quot;, main=&quot;Effect sensitvity analysis&quot;)

arrows(a,ressens$beta, a, ressens$beta+ressens$se, angle = 90, length = 0.1)
arrows(a,ressens$beta, a, ressens$beta-ressens$se, angle = 90, length = 0.1)

text(a, min(ressens$beta-ressens$se)*1.15, 
     formatC(ressens$p.value), cex=0.6, srt=90)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/disect_SLTIRK5-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>All models are corrected for: <br> site, Age, Pubstat, int_dis, medication, contraceptives, cigday_1, <br> site is included as random effect. <br></p>
<p>original: model defined as 0 + (1 | site) + Age + int_dis + medication + contraceptives + cigday_1 + V8 + group <br></p>
<p>all other models represent the original model + the variable of interest <br></p>
</div>
<div id="genomic-location-enrichment" class="section level2">
<h2>Genomic location enrichment</h2>
<p>Significant loci with a p-value &lt;= 0.01 and a absolute log2 fold-change lager 0.5 were tested for enrichment in annotated genomic feature using fisher exact test.</p>
<pre class="r"><code>Ranges=rowData(dds_filt)

TotTagsofInterest=sum(Ranges$WaldPvalue_groupCD&lt;=thresholdp &amp; abs(Ranges$groupCD)&gt;thresholdLFC)

Resall=data.frame()
index = Ranges$WaldPvalue_groupCD&lt;=thresholdp&amp; abs(Ranges$groupCD)&gt;thresholdLFC
for (feat in unique(Ranges$feature)){
  tmp=table(Ranges$feature == feat, signif=index)
  resfish=fisher.test(tmp)
  res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
  Resall = rbind(Resall, res)
}
tmp=table(Ranges$tf_binding!=&quot;&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resall = rbind(Resall, res)
tmp=table(Ranges$cpg==&quot;cpg&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resall = rbind(Resall, res)
colnames(Resall)=c(&quot;OR&quot;, &quot;CI95L&quot;, &quot;CI95U&quot;, &quot;P&quot;)
rownames(Resall)=c(unique(Ranges$feature), &quot;TF-binding&quot;, &quot;CpG-island&quot;)
Resall$Beta = log(Resall$OR)
Resall$SE = (log(Resall$OR)-log(Resall$CI95L))/1.96
Resall$Padj=p.adjust(Resall$P, method = &quot;bonferroni&quot;)

Resdown=data.frame()
index = Ranges$WaldPvalue_groupCD&lt;=thresholdp &amp; Ranges$groupCD&lt;thresholdLFC
for (feat in unique(Ranges$feature)){
  tmp=table(Ranges$feature == feat, signif=index)
  resfish=fisher.test(tmp)
  res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
  Resdown = rbind(Resdown, res)
}
tmp=table(Ranges$tf_binding!=&quot;&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resdown = rbind(Resdown, res)
tmp=table(Ranges$cpg==&quot;cpg&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resdown = rbind(Resdown, res)
colnames(Resdown)=c(&quot;OR&quot;, &quot;CI95L&quot;, &quot;CI95U&quot;, &quot;P&quot;)
rownames(Resdown)=c(unique(Ranges$feature), &quot;TF-binding&quot;, &quot;CpG-island&quot;)
Resdown$Beta = log(Resdown$OR)
Resdown$SE = (log(Resdown$OR)-log(Resdown$CI95L))/1.96
Resdown$Padj=p.adjust(Resdown$P, method = &quot;bonferroni&quot;)

Resup=data.frame()
index = Ranges$WaldPvalue_groupCD&lt;=thresholdp &amp; Ranges$groupCD&gt;thresholdLFC
for (feat in unique(Ranges$feature)){
  tmp=table(Ranges$feature == feat, signif=index)
  resfish=fisher.test(tmp)
  res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
  Resup = rbind(Resup, res)
}
tmp=table(Ranges$tf_binding!=&quot;&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resup = rbind(Resup, res)
tmp=table(Ranges$cpg==&quot;cpg&quot;, signif=index)
resfish=fisher.test(tmp)
res = c(resfish$estimate, unlist(resfish$conf.int), resfish$p.value)
Resup = rbind(Resup, res)
colnames(Resup)=c(&quot;OR&quot;, &quot;CI95L&quot;, &quot;CI95U&quot;, &quot;P&quot;)
rownames(Resup)=c(unique(Ranges$feature), &quot;TF-binding&quot;, &quot;CpG-island&quot;)
Resup$Beta = log(Resup$OR)
Resup$SE = (log(Resup$OR)-log(Resup$CI95L))/1.96
Resup$Padj=p.adjust(Resup$P, method = &quot;bonferroni&quot;)

multiORplot(Resall, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;All diff. methylated loci&quot;)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/genomiclocation-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>multiORplot(Resup, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;hypomethylated loci&quot;)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/genomiclocation-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>multiORplot(Resdown, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;Hypermethylated loci&quot;)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/genomiclocation-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(paste0(Home, &quot;/output/functional_Enrichemnt.pdf&quot;))
multiORplot(Resall, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;All diff. methylated loci&quot;)
multiORplot(Resup, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;hypomethylated loci&quot;)
multiORplot(Resdown, Pval = &quot;P&quot;, Padj = &quot;Padj&quot;, beta=&quot;Beta&quot;,SE = &quot;SE&quot;, pheno=&quot;Hypermethylated loci&quot;)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="go-term-enrichment" class="section level2">
<h2>GO-term Enrichment</h2>
<p>Significant loci and differentially methylated regions with a p-value &lt;= 0.01 and an absolute log2 fold-change lager 0.5 were tested for enrichment among GO-terms Molecular Function, Cellular Compartment and Biological Processes, KEGG pathways, Transcription factor Binding sites, Human Protein Atlas Tissue Expression, Human Phenotypes.</p>
<pre class="r"><code>getGOresults = function(geneset, genereference){
  resgo = gost(geneset, organism = &quot;hsapiens&quot;,
               correction_method = &quot;g_SCS&quot;,
               domain_scope = &quot;custom&quot;,
               sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;),
               custom_bg = genereference)
  if(length(resgo) != 0){
    return(resgo)
  } else {
    print(&quot;no significant results&quot;)
    return(NULL)
  }  
}

gene_univers = getuniquegenes(as.data.frame(rowRanges(dds_filt))$gene)


idx = (results_Deseq$pvalue &lt;= thresholdp &amp; 
         (abs(results_Deseq$log2FoldChange) &gt; thresholdLFC))

genes_reg = getuniquegenes(as.data.frame(rowRanges(dds_filt))$gene[idx])


dmr_genes = unique(resultsdmr_table$name[resultsdmr_table$p.value&lt;=thresholdp &amp; 
                   abs(resultsdmr_table$value)&gt;=thresholdLFC])


Genes_of_interset = list(&quot;01_dmregions&quot; = dmr_genes,  
                         &quot;02_dmtag&quot; = genes_reg
                         )

gostres = getGOresults(Genes_of_interset, gene_univers)

gostplot(gostres, capped = TRUE, interactive = T)</code></pre>
<div id="htmlwidget-44c81c46de429b626eeb" style="width:864px;height:1440px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-44c81c46de429b626eeb">{"x":{"data":[{"x":[138.095046488634],"y":[1.85315377698654],"text":"GO:0048731 (4421) <br> system development <br> 1.402e-02","key":["GO:0048731"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":19.5572105100368,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData0905215c","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[138.095046488634,138.557894784493,90.2123840900491,112.964396407373,90.6551085469579,112.960371639583,108.565325212815,138.606191997974,137.185448968076,137.306192001778,93.6535605505677,107.873065152921,137.966253919351,108.106501684746,89.7656348653501,78.6572757647286,114.461610025283,114.457585257493,78.6492262291485,138.565944320073,108.621671961876,169.270897790596,110.577709107855,138.38482976952,137.837461350069,152.531888551652,139.314551129028,167.411455071579,139.266253915547,89.004953753025,137.346439679679,139.278328218918,137.841486117859,167.532198105281,139.282352986708,90.6873066892785,131.224767870967,91.6773995656383,111.282043471119,123.678328264566,151.099071218383,166.4696594087,167.536222873071,90.6913314570686,165.612383869413,117.154179676846,108.2876162353],"y":[15.7891725212773,15.2650297433841,14.9584302171471,13.621410256286,10.8244007359635,9.52437935564416,8.87637286592102,8.77729201893277,7.82401527939161,6.69364965787958,6.02021546014978,5.85910030827547,5.70697606682804,5.19341033572917,5.04112871444826,4.86246382147518,4.67331264390008,4.64263892644462,4.60132329905935,4.45231739009699,3.96902998622611,3.85454805924345,3.64803343931882,3.54288745834155,3.52547317926018,3.06639450334527,2.74138797567311,2.70469638610715,2.68084806938811,2.5606348738613,2.39806782998628,2.37802306508785,2.23404026332143,2.02730434578675,1.98302713076904,1.95799238637558,1.88335827213225,1.75420797873719,1.74581898605794,1.6584402860475,1.61437948984051,1.60191789265926,1.58200621390344,1.5738520079254,1.5316457118958,1.40187409607766,1.3285865707126],"text":["GO:0048731 (4421) <br> system development <br> 1.625e-16","GO:0048856 (5338) <br> anatomical structure development <br> 5.432e-16","GO:0007275 (4921) <br> multicellular organism development <br> 1.100e-15","GO:0032502 (5754) <br> developmental process <br> 2.391e-14","GO:0007399 (2263) <br> nervous system development <br> 1.498e-11","GO:0032501 (6571) <br> multicellular organismal process <br> 2.990e-10","GO:0030154 (3780) <br> cell differentiation <br> 1.329e-09","GO:0048869 (3844) <br> cellular developmental process <br> 1.670e-09","GO:0048468 (1900) <br> cell development <br> 1.500e-08","GO:0048513 (3211) <br> animal organ development <br> 2.025e-07","GO:0009653 (2520) <br> anatomical structure morphogenesis <br> 9.545e-07","GO:0022008 (1548) <br> neurogenesis <br> 1.383e-06","GO:0048699 (1439) <br> generation of neurons <br> 1.963e-06","GO:0022610 (1334) <br> biological adhesion <br> 6.406e-06","GO:0007155 (1328) <br> cell adhesion <br> 9.096e-06","GO:0000904 (700) <br> cell morphogenesis involved in differentiation <br> 1.373e-05","GO:0032990 (660) <br> cell part morphogenesis <br> 2.122e-05","GO:0032989 (742) <br> cellular component morphogenesis <br> 2.277e-05","GO:0000902 (971) <br> cell morphogenesis <br> 2.504e-05","GO:0048858 (644) <br> cell projection morphogenesis <br> 3.529e-05","GO:0030182 (1313) <br> neuron differentiation <br> 1.074e-04","GO:0120039 (640) <br> plasma membrane bounded cell projection morphogenesis <br> 1.398e-04","GO:0031175 (935) <br> neuron projection development <br> 2.249e-04","GO:0048812 (627) <br> neuron projection morphogenesis <br> 2.865e-04","GO:0048666 (1065) <br> neuron development <br> 2.982e-04","GO:0065007 (10586) <br> biological regulation <br> 8.582e-04","GO:0050804 (403) <br> modulation of chemical synaptic transmission <br> 1.814e-03","GO:0099177 (404) <br> regulation of trans-synaptic signaling <br> 1.974e-03","GO:0050789 (10009) <br> regulation of biological process <br> 2.085e-03","GO:0006928 (1927) <br> movement of cell or subcellular component <br> 2.750e-03","GO:0048523 (4455) <br> negative regulation of cellular process <br> 3.999e-03","GO:0050793 (2275) <br> regulation of developmental process <br> 4.188e-03","GO:0048667 (566) <br> cell morphogenesis involved in neuron differentiation <br> 5.834e-03","GO:0099536 (681) <br> synaptic signaling <br> 9.391e-03","GO:0050794 (9510) <br> regulation of cellular process <br> 1.040e-02","GO:0007409 (448) <br> axonogenesis <br> 1.102e-02","GO:0045595 (1476) <br> regulation of cell differentiation <br> 1.308e-02","GO:0008150 (14695) <br> biological_process <br> 1.761e-02","GO:0031589 (332) <br> cell-substrate adhesion <br> 1.795e-02","GO:0040011 (1681) <br> locomotion <br> 2.196e-02","GO:0061564 (482) <br> axon development <br> 2.430e-02","GO:0098742 (251) <br> cell-cell adhesion via plasma-membrane adhesion molecules <br> 2.501e-02","GO:0099537 (662) <br> trans-synaptic signaling <br> 2.618e-02","GO:0007411 (262) <br> axon guidance <br> 2.668e-02","GO:0097485 (263) <br> neuron projection guidance <br> 2.940e-02","GO:0034330 (657) <br> cell junction organization <br> 3.964e-02","GO:0023051 (3244) <br> regulation of signaling <br> 4.693e-02"],"key":["GO:0048731","GO:0048856","GO:0007275","GO:0032502","GO:0007399","GO:0032501","GO:0030154","GO:0048869","GO:0048468","GO:0048513","GO:0009653","GO:0022008","GO:0048699","GO:0022610","GO:0007155","GO:0000904","GO:0032990","GO:0032989","GO:0000902","GO:0048858","GO:0030182","GO:0120039","GO:0031175","GO:0048812","GO:0048666","GO:0065007","GO:0050804","GO:0099177","GO:0050789","GO:0006928","GO:0048523","GO:0050793","GO:0048667","GO:0099536","GO:0050794","GO:0007409","GO:0045595","GO:0008150","GO:0031589","GO:0040011","GO:0061564","GO:0098742","GO:0099537","GO:0007411","GO:0097485","GO:0034330","GO:0023051"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[19.5572105100368,20.0672993551912,19.8491606094274,20.2659960771741,17.5933876869613,20.6117921783681,19.1203983965255,19.1678108903827,17.0328594997736,18.6518283671846,17.9272182411365,16.3442001517204,16.089479745425,15.8194657814909,15.8032095794893,13.2135709086509,12.9389573628929,13.4778569158187,14.6145960436799,12.8219577200101,15.7621477423026,12.7920118416962,14.4622363939182,12.6926673454302,14.9785116483559,21.7991364635162,10.1894908444215,10.2062445192508,21.6636724478295,17.078974121741,19.5782646242804,17.6099892104808,12.1796579594183,13.0861506703163,21.5391665463785,10.869914370257,16.1786389777086,22.5722677242784,8.70599188712214,16.6256796290843,11.3042482162819,3.77952755905512,12.9532791480363,5.70877976524072,5.79263328325251,12.9173507617318,18.6816282498582],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData0905215c","name":"GO:BP","legendgroup":"GO:BP","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[68.0551468973956,57.5203128050762,56.4132897346185,70.9696367265279,59.5773629469085,65.2976530673464,55.8778931223607,70.3577548839476,61.5216980124761,58.7642041824269],"y":[7.80681014933879,7.15351042968926,6.08744092080892,5.52390622960661,4.2292755070979,3.18810583814936,2.0003135848526,1.88938236815936,1.53412715861802,1.35298934129863],"text":["GO:0071944 (4931) <br> cell periphery <br> 1.560e-08","GO:0005886 (4511) <br> plasma membrane <br> 7.022e-08","GO:0005575 (15295) <br> cellular_component <br> 8.176e-07","GO:0110165 (15176) <br> cellular anatomical entity <br> 2.993e-06","GO:0030054 (1923) <br> cell junction <br> 5.898e-05","GO:0045202 (1234) <br> synapse <br> 6.485e-04","GO:0000785 (1098) <br> chromatin <br> 9.993e-03","GO:0098984 (339) <br> neuron to neuron synapse <br> 1.290e-02","GO:0032279 (316) <br> asymmetric synapse <br> 2.923e-02","GO:0014069 (310) <br> postsynaptic density <br> 4.436e-02"],"key":["GO:0071944","GO:0005886","GO:0005575","GO:0110165","GO:0030054","GO:0045202","GO:0000785","GO:0098984","GO:0032279","GO:0014069"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[19.854640878224,19.6125341366952,22.6644367627277,22.6464828059672,17.0721932941884,15.5353064218788,15.0961216170688,8.88645139918929,8.24986098503999,8.05975739183683],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData0905215c","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[3.59791046486707,9.4904568642761,9.55888124186487],"y":[7.34554033340611,3.06875002931791,1.81650967156365],"text":["GO:0003674 (15380) <br> molecular_function <br> 4.513e-08","GO:0005488 (14350) <br> binding <br> 8.536e-04","GO:0005515 (12728) <br> protein binding <br> 1.526e-02"],"key":["GO:0003674","GO:0005488","GO:0005515"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[22.6771653543307,22.5173364922675,22.2375048082395],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData0905215c","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,51.4265503994146],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[2,51.4265503994146],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[55.4511860479298,73.3970364046588],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[55.4511860479298,73.3970364046588],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[77.421672053174,200.004024635649],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[77.421672053174,200.004024635649],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x2","yaxis":"y2","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":34.6245339262839,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP"],"tickvals":[26.7132751997073,64.4241112262943,138.712848344411],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0.509276249462564,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0120785480374521,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"01_dmregions","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"},{"text":"02_dmtag","x":0.5,"y":0.490723750537436,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0.509276249462564,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":0.490723750537436},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":0.490723750537436,"ysizemode":"pixel"}],"xaxis2":{"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP"],"tickvals":[26.7132751997073,64.4241112262943,138.712848344411],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"domain":[0,1],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y2","title":"","hoverformat":".2f"},"yaxis2":{"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"domain":[0,0.490723750537436],"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"1ad453332ffb":{"colour":{},"size":{},"alpha":{},"key":{},"x":{},"y":{},"text":{},"type":"scatter"},"1ad42a7a6ee3":{"x":{},"xend":{},"y":{},"yend":{}},"1ad4c5a2faf":{"x":{},"xend":{},"y":{},"yend":{}},"1ad4353534ea":{"x":{},"xend":{},"y":{},"yend":{}},"1ad4252d6552":{"x":{},"y":{}},"1ad463c9b74":{"yintercept":{}}},"cur_data":"1ad453332ffb","visdat":{"1ad453332ffb":["function (y) ","x"],"1ad42a7a6ee3":["function (y) ","x"],"1ad4c5a2faf":["function (y) ","x"],"1ad4353534ea":["function (y) ","x"],"1ad4252d6552":["function (y) ","x"],"1ad463c9b74":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData0905215c"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>p = gostplot(gostres, capped = TRUE, interactive = F)

toptab = gostres$result

pp = publish_gostplot(p, filename = paste0(Home,&quot;/output/gostres.pdf&quot;))</code></pre>
<pre><code>The image is saved to C:/Users/chiocchetti/Projects/femNATCD_MethSeq/output/gostres.pdf</code></pre>
<pre class="r"><code>write.xlsx2(toptab, file = paste0(Home,&quot;/output/GOres.xlsx&quot;), sheetName = &quot;GO_enrichment&quot;)</code></pre>
</div>
<div id="kang-brain-module-enrichments" class="section level2">
<h2>Kang Brain Module Enrichments</h2>
<p>Gene sets identified to be deferentially methylated with a p-value &lt;= 0.01 and an absolute log2 fold-change larger 0.5 were tested for enrichment among gene-modules coregulated during Brain expression.</p>
<pre class="r"><code># define Reference Universe 

KangUnivers&lt;- read.table(paste0(Home,&quot;/data/KangUnivers.txt&quot;), sep=&quot;\t&quot;, header=T)
colnames(KangUnivers)&lt;-c(&quot;EntrezId&quot;,&quot;Symbol&quot;)

Kang_genes&lt;-read.table(paste0(Home,&quot;/data/Kang_dataset_genesMod_version2.txt&quot;),sep=&quot;\t&quot;,header=TRUE)

#3)Generate Gene universe to be used for single gene lists
tmp=merge(KangUnivers,Kang_genes,by.y=&quot;EntrezGene&quot;,by.x=&quot;EntrezId&quot;,all=TRUE) #18826
KangUni_Final&lt;-tmp[duplicated(tmp$EntrezId)==FALSE,] #18675


# Local analysis gene universe
Annotation_list&lt;-data.frame(Symbol = gene_univers)

# match modules 
Annotation_list$Module = Kang_genes$Module[match(Annotation_list$Symbol,Kang_genes$symbol)]

# check if overlapping in gene universes
Annotation_list$univers = Annotation_list$Symbol %in% KangUni_Final$Symbol

# drop duplicates 
Annotation_list = Annotation_list[duplicated(Annotation_list$Symbol)==FALSE,]

# selct only genes that have been detected on both datasets
Annotation_list = Annotation_list[Annotation_list$univers==T,] 

# final reference 
UniversalGeneset=Annotation_list$Symbol

# define Gene lists to test 
# sort and order Modules to be tested

Modules=unique(Annotation_list$Module)
Modules = Modules[! Modules %in% c(NA, &quot;&quot;)]
Modules = Modules[order(as.numeric(gsub(&quot;M&quot;,&quot;&quot;,Modules)))]

GL_all=list()

for(i in Modules){
  GL_all[[i]]=Annotation_list$Symbol[Annotation_list$Module%in%i]
}
GL_all[[&quot;M_all&quot;]]=Kang_genes$symbol[Kang_genes$Module %in% Modules]


GOI1 = Genes_of_interset

Resultsall=list()
for(j in names(GOI1)){
  Res = data.frame()
  for(i in names(GL_all)){
    Modulegene=GL_all[[i]]
    Factorgene=GOI1[[j]]
    Testframe&lt;-fisher.test(table(factor(UniversalGeneset %in% Factorgene,levels=c(&quot;TRUE&quot;,&quot;FALSE&quot;)),
                                 factor(UniversalGeneset %in% Modulegene,levels=c(&quot;TRUE&quot;,&quot;FALSE&quot;))))
    beta=log(Testframe$estimate)
    Res[i, &quot;beta&quot;] =beta
    Res[i, &quot;SE&quot;]=abs(beta-log(Testframe$conf.int[1]))/1.96
    Res[i, &quot;Pval&quot;]=Testframe$p.value
    Res[i, &quot;OR&quot;]=(Testframe$estimate)
    Res[i, &quot;ORL&quot;]=(Testframe$conf.int[1])
    Res[i, &quot;ORU&quot;]=(Testframe$conf.int[2])
  }
  Res$Padj = p.adjust(Res$Pval, method = &quot;bonferroni&quot;)
  Resultsall[[j]] = Res
  
}
par(mfrow = c(2,1))
for (i in names(Resultsall)){ 
  multiORplot(datatoplot = Resultsall[[i]], pheno=i)
}</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/brainModules-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mfrow = c(1,1))
pdf(paste0(Home, &quot;/output/BrainMod_Enrichemnt.pdf&quot;))
for (i in names(Resultsall)){
  multiORplot(datatoplot = Resultsall[[i]], pheno=i)
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>Modsig = c()

for(r in names(Resultsall)){
  a=rownames(Resultsall[[r]])[Resultsall[[r]]$Padj&lt;=0.05]
  Modsig = c(Modsig,a)
}</code></pre>
<pre class="r"><code># show brains and expression
Modsig2=unique(Modsig[Modsig!=&quot;M_all&quot;])

load(paste0(Home,&quot;/data/Kang_DataPreprocessing.RData&quot;)) #Load the Kang expression data of all genes 
datExprPlot=matriz #Expression data of Kang loaded as Rdata object DataPreprocessing.RData


Genes = GL_all[names(GL_all)!=&quot;M_all&quot;]

Genes_expression&lt;-list()

pcatest&lt;-list()
for (i in names(Genes)){
  Genes_expression[[i]]&lt;-matriz[,which(colnames(matriz) %in% Genes[[i]])]
  pcatest[[i]]=prcomp(t(as.matrix(Genes_expression[[i]])),retx=TRUE)
}

# PCA test
PCA&lt;-data.frame(pcatest[[1]]$rotation)
PCA$donor_name&lt;-rownames(PCA)
PC1&lt;-data.frame(PCA[,c(1,ncol(PCA))])

#Combining the age with expression data
list &lt;- strsplit(sampleInfo$age, &quot; &quot;)
library(&quot;plyr&quot;)</code></pre>
<pre><code>------------------------------------------------------------------------------</code></pre>
<pre><code>You have loaded plyr after dplyr - this is likely to cause problems.
If you need functions from both plyr and dplyr, please load plyr first, then dplyr:
library(plyr); library(dplyr)</code></pre>
<pre><code>------------------------------------------------------------------------------</code></pre>
<pre><code>
Attache Paket: &#39;plyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    desc</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    rename</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    compact</code></pre>
<pre class="r"><code>df &lt;- ldply(list)
colnames(df) &lt;- c(&quot;Age&quot;, &quot;time&quot;)

sampleInfo&lt;-cbind(sampleInfo[,1:9],df)
sampleInfo$Age&lt;-as.numeric(sampleInfo$Age)

sampleInfo$period&lt;-ifelse(sampleInfo$time==&quot;pcw&quot;,sampleInfo$Age*7,ifelse(sampleInfo$time==&quot;yrs&quot;,sampleInfo$Age*365+270,ifelse(sampleInfo$time==&quot;mos&quot;,sampleInfo$Age*30+270,NA)))

#We need it just for the donor names

PCA_matrix&lt;-merge.with.order(PC1,sampleInfo,by.y=&quot;SampleID&quot;,by.x=&quot;donor_name&quot;,keep_order=1)

#Select which have phenotype info present 
matriz2&lt;-matriz[which(rownames(matriz) %in% PCA_matrix$donor_name),]
FactorGenes_expression&lt;-list()
#Factors here mean modules
for (i in names(Genes)){
  FactorGenes_expression[[i]]&lt;-matriz2[,which(colnames(matriz2) %in% Genes[[i]])]
}


FactorseGE&lt;-list()
for (i in names(Genes)){
  FactorseGE[[i]]&lt;-FactorGenes_expression[[i]]
}

allModgenes=NULL
colors=vector()
for ( i in names(Genes)){
  allModgenes=cbind(allModgenes,FactorseGE[[i]])
  colors=c(colors, rep(i, ncol(FactorseGE[[i]])))
}


lengths=unlist(lapply(FactorGenes_expression, ncol), use.names = F)

MEorig=moduleEigengenes(allModgenes, colors)

PCA_matrixfreeze=PCA_matrix

index=!PCA_matrix$structure_acronym %in% c(&quot;URL&quot;, &quot;DTH&quot;, &quot;CGE&quot;,&quot;LGE&quot;, &quot;MGE&quot;,  &quot;Ocx&quot;, &quot;PCx&quot;, &quot;M1C-S1C&quot;,&quot;DIE&quot;, &quot;TCx&quot;, &quot;CB&quot;)
PCA_matrix=PCA_matrix[index,]
ME = MEorig$eigengenes[index,]
matsel = matriz2[index,]

colnames(ME) = gsub(&quot;ME&quot;, &quot;&quot;, colnames(ME))

timepoints=seq(56,15000, length.out=1000)
matrix(c(&quot;CB&quot;, &quot;THA&quot;, &quot;CBC&quot;, &quot;MD&quot;), ncol=2 ) -&gt; cnm


brainheatmap=function(Module){
  MEmod=ME[,Module]
  toplot=data.frame(matrix(NA, nrow=length(table(PCA_matrix$structure_acronym)), ncol=998))
  rownames(toplot)=unique(PCA_matrix$structure_acronym)
  target &lt;- c(&quot;OFC&quot;, &quot;DFC&quot;, &quot;VFC&quot;, &quot;MFC&quot;,&quot;M1C&quot;,&quot;S1C&quot;,&quot;IPC&quot;,&quot;A1C&quot;,&quot;STC&quot;,&quot;ITC&quot;,&quot;V1C&quot;,&quot;HIP&quot;,&quot;AMY&quot;,&quot;STR&quot;,&quot;MD&quot;,&quot;CBC&quot;)
  toplot&lt;-toplot[c(6,2,8,5,11,12,10,9,7,4,14,3,1,13,16,15),]
  
  
  for ( i in unique(PCA_matrix$structure_acronym)){
    index=PCA_matrix$structure_acronym==i
    LOESS=loess(MEmod[index]~PCA_matrix$period[index])
    toplot[i,]=predict(LOESS,newdata = round(exp(seq(log(56),log(15000), length.out=998)),2))
    colnames(toplot)[c(1,77,282,392,640,803,996)]&lt;-
      c(&quot;1pcw&quot;,&quot;21pcw&quot;,&quot;Birth&quot;,&quot;1.3years&quot;,&quot;5.4years&quot;,&quot;13.6years&quot;,&quot;40.7years&quot;)
  }
  
  
  
  cols=viridis(100)
  labvec &lt;- c(rep(NA, 1000))
  
  labvec[c(1,77,282,392,640,803,996)] &lt;- c(&quot;1pcw&quot;,&quot;21pcw&quot;,&quot;Birth&quot;,&quot;1.3years&quot;,&quot;5.4years&quot;,&quot;13.6years&quot;,&quot;40.7years&quot;)
  
  
  toplot&lt;-toplot[,1:998]
  date&lt;-c(1:998)
  dateY&lt;-paste0(round(date/365,2),&quot;_Years&quot;)
  
  names(toplot)&lt;-dateY
  
  par(xpd=FALSE) 
  heatmap.2(as.matrix(toplot), col = cols, 
            main=Module,
            trace = &quot;none&quot;, 
            na.color = &quot;grey&quot;,
            Colv = F, Rowv = F,
            labCol = labvec,
            #breaks = seq(-0.1,0.1, length.out=101),
            symkey = T,
            scale = &quot;row&quot;,
            key.title = &quot;&quot;,
            dendrogram = &quot;none&quot;,
            key.xlab = &quot;eigengene&quot;,
            density.info = &quot;none&quot;,
            #main=paste(&quot;Module&quot;,1),
            srtCol=90,
            tracecol = &quot;none&quot;, 
            cexRow = 1,
            add.expr=eval.parent(abline(v=282),
                                 axis(1,at=c(1,77,282,392,640,803,996),
                                      labels =FALSE)),cexCol = 1)
  
 
}

brainheatmap_gene=function(Genename){
  MEmod=matsel[,Genename]
  toplot=data.frame(matrix(NA, nrow=length(table(PCA_matrix$structure_acronym)), ncol=998))
  rownames(toplot)=unique(PCA_matrix$structure_acronym)
  target &lt;- c(&quot;OFC&quot;, &quot;DFC&quot;, &quot;VFC&quot;, &quot;MFC&quot;,&quot;M1C&quot;,&quot;S1C&quot;,&quot;IPC&quot;,&quot;A1C&quot;,&quot;STC&quot;,&quot;ITC&quot;,&quot;V1C&quot;,&quot;HIP&quot;,&quot;AMY&quot;,&quot;STR&quot;,&quot;MD&quot;,&quot;CBC&quot;)
  toplot&lt;-toplot[c(6,2,8,5,11,12,10,9,7,4,14,3,1,13,16,15),]


  for ( i in unique(PCA_matrix$structure_acronym)){
    index=PCA_matrix$structure_acronym==i
    LOESS=loess(MEmod[index]~PCA_matrix$period[index])
    toplot[i,]=predict(LOESS,newdata = round(exp(seq(log(56),log(15000), length.out=998)),2))
    colnames(toplot)[c(1,77,282,392,640,803,996)]&lt;-
      c(&quot;1pcw&quot;,&quot;21pcw&quot;,&quot;Birth&quot;,&quot;1.3years&quot;,&quot;5.4years&quot;,&quot;13.6years&quot;,&quot;40.7years&quot;)
  }



  cols=viridis(100)
  labvec &lt;- c(rep(NA, 1000))

  labvec[c(1,77,282,392,640,803,996)] &lt;- c(&quot;1pcw&quot;,&quot;21pcw&quot;,&quot;Birth&quot;,&quot;1.3years&quot;,&quot;5.4years&quot;,&quot;13.6years&quot;,&quot;40.7years&quot;)


  toplot&lt;-toplot[,1:998]
  date&lt;-c(1:998)
  dateY&lt;-paste0(round(date/365,2),&quot;_Years&quot;)

  names(toplot)&lt;-dateY

  par(xpd=FALSE)
  heatmap.2(as.matrix(toplot), col = cols,
            main=Genename,
            trace = &quot;none&quot;,
            na.color = &quot;grey&quot;,
            Colv = F, Rowv = F,
            labCol = labvec,
            #breaks = seq(-0.1,0.1, length.out=101),
            symkey = F,
            scale = &quot;none&quot;,
            key.title = &quot;&quot;,
            dendrogram = &quot;none&quot;,
            key.xlab = &quot;eigengene&quot;,
            density.info = &quot;none&quot;,
            #main=paste(&quot;Module&quot;,1),
            #srtCol=90,
            tracecol = &quot;none&quot;,
            cexRow = 1,
            add.expr=eval.parent(abline(v=282),
                                 axis(1,at=c(1,77,282,392,640,803,996),
                                      labels =FALSE))
            ,cexCol = 1)
}


brainheatmap_gene(&quot;SLITRK5&quot;)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/modplots-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for(Module in Modsig2){
  brainheatmap(Module)
}</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/modplots-2.png" width="864" style="display: block; margin: auto;" /><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/modplots-3.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(paste0(Home, &quot;/output/Brain_Module_Heatmap.pdf&quot;))

brainheatmap_gene(&quot;SLITRK5&quot;)
for(Module in Modsig2){
  brainheatmap(Module)
}
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="sem" class="section level2">
<h2>SEM</h2>
<pre class="r"><code>dropfact=c(&quot;site&quot;)


Patdata=as.data.frame(colData(dds_filt))

load(paste0(Home, &quot;/output/envFact.RData&quot;))
load(paste0(Home, &quot;/output/modelFact.RData&quot;))

envFact=envFact[!envFact %in% dropfact] 
modelFact=modelFact[!modelFact %in% dropfact] 

EpiMarker = c()

# TopHit
Patdata$Epi_TopHit=log2_cpm[base::which.min(results_Deseq$pvalue),]

# 1PC of all diff met
tmp=glmpca(log2_cpm[base::which(results_Deseq$pvalue&lt;=thresholdp),], 1)

Patdata$Epi_all= tmp$factors$dim1
  
EpiMarker = c(EpiMarker, &quot;Epi_TopHit&quot;, &quot;Epi_all&quot;)

#Brain Modules

Epitestset=GL_all[Modsig]

for(n in names(Epitestset)){
  index=gettaglistforgenelist(genelist = Epitestset[[n]], dds_filt)
  index = base::intersect(index, base::which(results_Deseq$pvalue&lt;=thresholdp))
  # get eigenvalue
  epiname=paste0(&quot;Epi_&quot;,n)
  tmp=glmpca(log2_cpm[index,], 1)
  Patdata[,epiname]= tmp$factors$dim1
  EpiMarker = c(EpiMarker, epiname)
}

cormat = cor(apply(Patdata[,c(&quot;group&quot;, envFact, modelFact, EpiMarker)] %&gt;% mutate_all(as.numeric), 2, minmax_scaling),
             use = &quot;pairwise.complete.obs&quot;)

par(mfrow=c(1,2))
corrplot(cormat, main=&quot;correlations&quot;)
corrplot(cormat, order = &quot;hclust&quot;, main=&quot;scaled correlations&quot;)</code></pre>
<p><img src="figure/03_01_Posthoc_analyses_Gene_Enrichment.Rmd/SEM,%20-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>fullmodEnv=paste(envFact,modelFact, sep = &quot;+&quot;, collapse = &quot;+&quot;)

Dataset = Patdata[,c(&quot;group&quot;, envFact,modelFact,EpiMarker)]

model = paste0(&quot;group ~ Epi+&quot;, fullmodEnv,&quot;
Epi ~&quot; , fullmodEnv,&quot;
Epi~~Epi
group~~group
&quot;)

Netlist = list()

for (marker in EpiMarker) {
  Dataset$Epi = Dataset[,marker]
  Datasetscaled = Dataset %&gt;% mutate_if(is.numeric, minmax_scaling)
  Datasetscaled = Datasetscaled %&gt;% mutate_if(is.factor, ~ as.numeric(.)-1)
  
  fit&lt;-lavaan(model,data=Datasetscaled)
  
  sink(paste0(Home,&quot;/output/SEM_summary_group&quot;,marker,&quot;.txt&quot;))
  summary(fit)
  print(fitMeasures(fit))
  print(parameterEstimates(fit))
  sink()
  
  #SOURCE FOR PLOT https://stackoverflow.com/questions/51270032/how-can-i-display-only-significant-path-lines-on-a-path-diagram-r-lavaan-sem
  restab=lavaan::standardizedSolution(fit) %&gt;% dplyr::filter(!is.na(pvalue)) %&gt;% 
    arrange(desc(pvalue)) %&gt;% mutate_if(&quot;is.numeric&quot;,&quot;round&quot;,3) %&gt;% 
    dplyr::select(-ci.lower,-ci.upper,-z)
  
  pvalue_cutoff &lt;- 0.05
  
  obj &lt;- semPlot:::semPlotModel(fit)
  original_Pars &lt;- obj@Pars
  check_Pars &lt;- obj@Pars %&gt;% dplyr:::filter(!(edge %in% c(&quot;int&quot;,&quot;&lt;-&gt;&quot;) | lhs == rhs)) # this is the list of parameter to sift thru
  keep_Pars &lt;- obj@Pars %&gt;% dplyr:::filter(edge %in% c(&quot;int&quot;,&quot;&lt;-&gt;&quot;) | lhs == rhs) # this is the list of parameter to keep asis
  
  test_against &lt;- lavaan::standardizedSolution(fit) %&gt;% dplyr::filter(pvalue &lt; pvalue_cutoff, rhs != lhs)
  
  # for some reason, the rhs and lhs are reversed in the standardizedSolution() output, for some of the values
  # I&#39;ll have to reverse it myself, and test against both orders
  
  test_against_rev &lt;- test_against %&gt;% dplyr::rename(rhs2 = lhs, lhs = rhs) %&gt;% dplyr::rename(rhs = rhs2)
  
  checked_Pars &lt;-
    check_Pars %&gt;% semi_join(test_against, by = c(&quot;lhs&quot;, &quot;rhs&quot;)) %&gt;% bind_rows(
      check_Pars %&gt;% semi_join(test_against_rev, by = c(&quot;lhs&quot;, &quot;rhs&quot;))
    )
  
  obj@Pars &lt;- keep_Pars %&gt;% bind_rows(checked_Pars) %&gt;% 
    mutate_if(&quot;is.numeric&quot;,&quot;round&quot;,3) %&gt;% 
    mutate_at(c(&quot;lhs&quot;,&quot;rhs&quot;),~gsub(&quot;Epi&quot;, marker,.))
  
  obj@Vars = obj@Vars %&gt;% mutate_at(c(&quot;name&quot;),~gsub(&quot;Epi&quot;, marker,.))
  
  DF = obj@Pars
  DF = DF[DF$lhs!=DF$rhs,]
  DF = DF[abs(DF$std)&gt;0.1,]
  
  DF = DF[DF$edge == &quot;~&gt;&quot;,] # only include directly modelled effects in figure 
  
  nodes &lt;- data.frame(id=obj@Vars$name, label = obj@Vars$name)
  nodes$color&lt;-Dark8[8]
  nodes$color[nodes$label == &quot;group&quot;] = Dark8[3]
  nodes$color[nodes$label == marker] = Dark8[4]
  nodes$color[nodes$label %in% envFact] = Dark8[5]
  
  edges &lt;- data.frame(from = DF$lhs, 
                      to = DF$rhs, 
                      width=abs(DF$std), 
                      arrows =&quot;to&quot;)
  
  edges$dashes = F
  edges$label =  DF$std
  edges$color=c(&quot;firebrick&quot;, &quot;forestgreen&quot;)[1:2][factor(sign(DF$std), levels=c(-1,0,1),labels=c(1,2,2))]
  edges$width=2
  cexlab = 18
  plotnet&lt;- visNetwork(nodes, edges,  
                       main=list(text=marker,
                                 style=&quot;font-family:arial;font-size:20px;text-align:center&quot;),
                       submain=list(text=&quot;significant paths&quot;, 
                                    style=&quot;font-family:arial;text-align:center&quot;)) %&gt;% 
    visEdges(arrows =list(to = list(enabled = TRUE, scaleFactor = 0.7)),
                       font=list(size=cexlab, style=&quot;font-family:arial;text-align:center&quot;)) %&gt;%
    visNodes(font=list(size=cexlab, style=&quot;font-family:arial;text-align:center&quot;)) %&gt;%
    
    visPhysics(enabled = T, solver = &quot;forceAtlas2Based&quot;)
  
  
  Netlist[[marker]] = plotnet
  htmlfile = paste0(Home,&quot;/output/SEMplot_&quot;,marker,&quot;.html&quot;)
  visSave(plotnet, htmlfile)
  webshot(paste0(Home,&quot;/output/SEMplot_&quot;,marker,&quot;.html&quot;), zoom = 1, 
          file = paste0(Home,&quot;/output/SEMplot_&quot;,marker,&quot;.png&quot;))
}</code></pre>
<pre class="r"><code>rmd_paths &lt;-paste0(tempfile(c(names(Netlist))),&quot;.Rmd&quot;)
names(rmd_paths) &lt;- names(Netlist)

for (n in names(rmd_paths)) {
    sink(file = rmd_paths[n])
    cat(&quot;  \n&quot;,
        &quot;```{r, echo = FALSE}&quot;,
            &quot;Netlist[[n]]&quot;,
        &quot;```&quot;,
        sep = &quot;  \n&quot;)
    sink()
}</code></pre>
</div>
<div id="interactive-results-sem-analysis" class="section level2">
<h2>Interactive results SEM analysis</h2>
<p>only direct effects with a significant standardized effect of p&lt;0.05 are shown.</p>
<pre class="r"><code>    for (n in names(rmd_paths)) {
        cat(knitr::knit_child(rmd_paths[[n]],
                              quiet= TRUE))
        file.remove(rmd_paths[[n]])
    }</code></pre>
<div id="htmlwidget-7965022d8e5ff1ac8673" style="width:864px;height:672px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-7965022d8e5ff1ac8673">{"x":{"nodes":{"id":["group","Epi_TopHit","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"label":["group","Epi_TopHit","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"color":["#7570B3","#E7298A","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666"]},"edges":{"from":["Epi_TopHit","Age","int_dis","medication","EduPar","cigday_1"],"to":["group","group","group","group","group","group"],"width":[2,2,2,2,2,2],"arrows":["to","to","to","to","to","to"],"dashes":[false,false,false,false,false,false],"label":[-0.339,-0.158,0.167,0.214,-0.22,0.334],"color":["firebrick","firebrick","forestgreen","forestgreen","firebrick","forestgreen"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","font":{"size":18,"style":"font-family:arial;text-align:center"}},"manipulation":{"enabled":false},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":0.7}},"font":{"size":18,"style":"font-family:arial;text-align:center"}},"physics":{"solver":"forceAtlas2Based","enabled":true}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":{"text":"Epi_TopHit","style":"font-family:arial;font-size:20px;text-align:center"},"submain":{"text":"significant paths","style":"font-family:arial;text-align:center"},"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-8aa746e8983381d3d120" style="width:864px;height:672px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-8aa746e8983381d3d120">{"x":{"nodes":{"id":["group","Epi_all","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"label":["group","Epi_all","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"color":["#7570B3","#E7298A","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666"]},"edges":{"from":["Epi_all","Age","int_dis","medication","cigday_1","EduPar"],"to":["group","group","group","group","group","Epi_all"],"width":[2,2,2,2,2,2],"arrows":["to","to","to","to","to","to"],"dashes":[false,false,false,false,false,false],"label":[0.599,-0.168,0.16,0.229,0.351,-0.241],"color":["forestgreen","firebrick","forestgreen","forestgreen","forestgreen","firebrick"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","font":{"size":18,"style":"font-family:arial;text-align:center"}},"manipulation":{"enabled":false},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":0.7}},"font":{"size":18,"style":"font-family:arial;text-align:center"}},"physics":{"solver":"forceAtlas2Based","enabled":true}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":{"text":"Epi_all","style":"font-family:arial;font-size:20px;text-align:center"},"submain":{"text":"significant paths","style":"font-family:arial;text-align:center"},"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-39ad7dbd4b5bb3cff80d" style="width:864px;height:672px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-39ad7dbd4b5bb3cff80d">{"x":{"nodes":{"id":["group","Epi_M2","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"label":["group","Epi_M2","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"color":["#7570B3","#E7298A","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666"]},"edges":{"from":["Epi_M2","int_dis","medication","EduPar","cigday_1"],"to":["group","group","group","group","group"],"width":[2,2,2,2,2],"arrows":["to","to","to","to","to"],"dashes":[false,false,false,false,false],"label":[-0.29,0.212,0.263,-0.21,0.352],"color":["firebrick","forestgreen","forestgreen","firebrick","forestgreen"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","font":{"size":18,"style":"font-family:arial;text-align:center"}},"manipulation":{"enabled":false},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":0.7}},"font":{"size":18,"style":"font-family:arial;text-align:center"}},"physics":{"solver":"forceAtlas2Based","enabled":true}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":{"text":"Epi_M2","style":"font-family:arial;font-size:20px;text-align:center"},"submain":{"text":"significant paths","style":"font-family:arial;text-align:center"},"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-2b4f1e25c7ea71270324" style="width:864px;height:672px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-2b4f1e25c7ea71270324">{"x":{"nodes":{"id":["group","Epi_M15","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"label":["group","Epi_M15","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"color":["#7570B3","#E7298A","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666"]},"edges":{"from":["Epi_M15","Age","Matagg","int_dis","medication","cigday_1","EduPar"],"to":["group","group","group","group","group","group","Epi_M15"],"width":[2,2,2,2,2,2,2],"arrows":["to","to","to","to","to","to","to"],"dashes":[false,false,false,false,false,false,false],"label":[-0.545,-0.18,0.198,0.147,0.22,0.351,0.303],"color":["firebrick","firebrick","forestgreen","forestgreen","forestgreen","forestgreen","forestgreen"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","font":{"size":18,"style":"font-family:arial;text-align:center"}},"manipulation":{"enabled":false},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":0.7}},"font":{"size":18,"style":"font-family:arial;text-align:center"}},"physics":{"solver":"forceAtlas2Based","enabled":true}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":{"text":"Epi_M15","style":"font-family:arial;font-size:20px;text-align:center"},"submain":{"text":"significant paths","style":"font-family:arial;text-align:center"},"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
<div id="htmlwidget-9089c5d81d2849e4b34a" style="width:864px;height:672px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-9089c5d81d2849e4b34a">{"x":{"nodes":{"id":["group","Epi_M_all","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"label":["group","Epi_M_all","Matsmk","Age","Matagg","int_dis","FamScore","medication","EduPar","contraceptives","n_trauma","cigday_1"],"color":["#7570B3","#E7298A","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666","#66A61E","#666666"]},"edges":{"from":["Epi_M_all","Age","int_dis","medication","cigday_1","EduPar"],"to":["group","group","group","group","group","Epi_M_all"],"width":[2,2,2,2,2,2],"arrows":["to","to","to","to","to","to"],"dashes":[false,false,false,false,false,false],"label":[0.662,-0.176,0.138,0.203,0.29,-0.227],"color":["forestgreen","firebrick","forestgreen","forestgreen","forestgreen","firebrick"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","font":{"size":18,"style":"font-family:arial;text-align:center"}},"manipulation":{"enabled":false},"edges":{"arrows":{"to":{"enabled":true,"scaleFactor":0.7}},"font":{"size":18,"style":"font-family:arial;text-align:center"}},"physics":{"solver":"forceAtlas2Based","enabled":true}},"groups":null,"width":null,"height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":{"text":"Epi_M_all","style":"font-family:arial;font-size:20px;text-align:center"},"submain":{"text":"significant paths","style":"font-family:arial;text-align:center"},"footer":null,"background":"rgba(0, 0, 0, 0)"},"evals":[],"jsHooks":[]}</script>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] plyr_1.8.6                  scales_1.1.1               
 [3] RCircos_1.2.1               compareGroups_4.4.6        
 [5] lme4_1.1-26                 Matrix_1.2-18              
 [7] webshot_0.5.2               visNetwork_2.0.9           
 [9] org.Hs.eg.db_3.12.0         AnnotationDbi_1.52.0       
[11] xlsx_0.6.5                  gprofiler2_0.2.0           
[13] BiocParallel_1.24.1         kableExtra_1.3.1           
[15] glmpca_0.2.0                knitr_1.30                 
[17] DESeq2_1.30.0               SummarizedExperiment_1.20.0
[19] Biobase_2.50.0              MatrixGenerics_1.2.0       
[21] matrixStats_0.57.0          GenomicRanges_1.42.0       
[23] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[25] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[27] forcats_0.5.0               stringr_1.4.0              
[29] dplyr_1.0.2                 purrr_0.3.4                
[31] readr_1.4.0                 tidyr_1.1.2                
[33] tibble_3.0.4                tidyverse_1.3.0            
[35] semPlot_1.1.2               lavaan_0.6-7               
[37] viridis_0.5.1               viridisLite_0.3.0          
[39] WGCNA_1.69                  fastcluster_1.1.25         
[41] dynamicTreeCut_1.63-1       ggplot2_3.3.3              
[43] gplots_3.1.1                corrplot_0.84              
[45] RColorBrewer_1.1-2          workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] coda_0.19-4            bit64_4.0.5            DelayedArray_0.16.0   
  [4] data.table_1.13.6      rpart_4.1-15           RCurl_1.98-1.2        
  [7] doParallel_1.0.16      generics_0.1.0         preprocessCore_1.52.1 
 [10] callr_3.5.1            RSQLite_2.2.2          mice_3.12.0           
 [13] chron_2.3-56           bit_4.0.4              xml2_1.3.2            
 [16] lubridate_1.7.9.2      httpuv_1.5.5           assertthat_0.2.1      
 [19] d3Network_0.5.2.1      xfun_0.20              hms_1.0.0             
 [22] rJava_0.9-13           evaluate_0.14          promises_1.1.1        
 [25] fansi_0.4.1            caTools_1.18.1         dbplyr_2.0.0          
 [28] readxl_1.3.1           igraph_1.2.6           DBI_1.1.1             
 [31] geneplotter_1.68.0     tmvnsim_1.0-2          Rsolnp_1.16           
 [34] htmlwidgets_1.5.3      ellipsis_0.3.1         crosstalk_1.1.1       
 [37] backports_1.2.0        pbivnorm_0.6.0         annotate_1.68.0       
 [40] vctrs_0.3.6            abind_1.4-5            cachem_1.0.1          
 [43] withr_2.4.1            HardyWeinberg_1.7.1    checkmate_2.0.0       
 [46] fdrtool_1.2.16         mnormt_2.0.2           cluster_2.1.0         
 [49] mi_1.0                 lazyeval_0.2.2         crayon_1.3.4          
 [52] genefilter_1.72.0      pkgconfig_2.0.3        nlme_3.1-151          
 [55] nnet_7.3-15            rlang_0.4.10           lifecycle_0.2.0       
 [58] kutils_1.70            modelr_0.1.8           cellranger_1.1.0      
 [61] rprojroot_2.0.2        flextable_0.6.2        regsem_1.6.2          
 [64] carData_3.0-4          boot_1.3-26            reprex_1.0.0          
 [67] base64enc_0.1-3        processx_3.4.5         whisker_0.4           
 [70] png_0.1-7              rjson_0.2.20           bitops_1.0-6          
 [73] KernSmooth_2.23-18     blob_1.2.1             arm_1.11-2            
 [76] jpeg_0.1-8.1           rockchalk_1.8.144      memoise_2.0.0         
 [79] magrittr_2.0.1         zlibbioc_1.36.0        compiler_4.0.3        
 [82] cli_2.2.0              XVector_0.30.0         pbapply_1.4-3         
 [85] ps_1.5.0               htmlTable_2.1.0        Formula_1.2-4         
 [88] MASS_7.3-53            tidyselect_1.1.0       stringi_1.5.3         
 [91] lisrelToR_0.1.4        sem_3.1-11             yaml_2.2.1            
 [94] OpenMx_2.18.1          locfit_1.5-9.4         latticeExtra_0.6-29   
 [97] grid_4.0.3             tools_4.0.3            matrixcalc_1.0-3      
[100] rstudioapi_0.13        uuid_0.1-4             foreach_1.5.1         
[103] foreign_0.8-81         git2r_0.28.0           gridExtra_2.3         
[106] farver_2.0.3           BDgraph_2.63           digest_0.6.27         
[109] shiny_1.6.0            Rcpp_1.0.5             broom_0.7.3           
[112] later_1.1.0.1          writexl_1.3.1          httr_1.4.2            
[115] gdtools_0.2.3          psych_2.0.12           colorspace_2.0-0      
[118] rvest_0.3.6            XML_3.99-0.5           fs_1.5.0              
[121] truncnorm_1.0-8        splines_4.0.3          statmod_1.4.35        
[124] xlsxjars_0.6.1         plotly_4.9.3           systemfonts_0.3.2     
[127] xtable_1.8-4           jsonlite_1.7.2         nloptr_1.2.2.2        
[130] corpcor_1.6.9          glasso_1.11            R6_2.5.0              
[133] Hmisc_4.4-2            mime_0.9               pillar_1.4.7          
[136] htmltools_0.5.1.1      glue_1.4.2             fastmap_1.1.0         
[139] minqa_1.2.4            codetools_0.2-18       lattice_0.20-41       
[142] huge_1.3.4.1           gtools_3.8.2           officer_0.3.16        
[145] zip_2.1.1              GO.db_3.12.1           openxlsx_4.2.3        
[148] survival_3.2-7         rmarkdown_2.6          qgraph_1.6.5          
[151] munsell_0.5.0          GenomeInfoDbData_1.2.4 iterators_1.0.13      
[154] impute_1.64.0          haven_2.3.1            reshape2_1.4.4        
[157] gtable_0.3.0          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
