<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AG Chiocchetti" />


<title>PlotResults</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">femNATCD_MethSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/achiocch/femNATCD_MethSeq">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">PlotResults</h1>
<h4 class="author">AG Chiocchetti</h4>
<h4 class="date">29 Januar 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>femNATCD_MethSeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree723a1aa6b066fd0af5ab10215e9f45b2b859b783targetblank723a1aaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/723a1aa6b066fd0af5ab10215e9f45b2b859b783" target="_blank">723a1aa</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree723a1aa6b066fd0af5ab10215e9f45b2b859b783targetblank723a1aaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/723a1aa6b066fd0af5ab10215e9f45b2b859b783" target="_blank">723a1aa</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/Epicounts.csv
    Ignored:    data/Epimeta.csv
    Ignored:    data/Epitpm.csv
    Ignored:    data/KangUnivers.txt
    Ignored:    data/Kang_DataPreprocessing.RData
    Ignored:    data/Kang_dataset_genesMod_version2.txt
    Ignored:    data/PatMeta.csv
    Ignored:    data/ProcessedData.RData
    Ignored:    data/RTrawdata/
    Ignored:    data/SNPCommonFilt.csv
    Ignored:    data/femNAT_PC20.txt
    Ignored:    output/BrainMod_Enrichemnt.pdf
    Ignored:    output/Brain_Module_Heatmap.pdf
    Ignored:    output/DMR_Results.csv
    Ignored:    output/GOres.xlsx
    Ignored:    output/LME_GOplot.pdf
    Ignored:    output/LME_Results.csv
    Ignored:    output/LME_Results_Sig.csv
    Ignored:    output/LME_tophit.svg
    Ignored:    output/ProcessedData.RData
    Ignored:    output/RNAvsMETplots.pdf
    Ignored:    output/Regions_GOplot.pdf
    Ignored:    output/ResultsgroupComp.txt
    Ignored:    output/SEM_summary_groupEpi_M15.txt
    Ignored:    output/SEM_summary_groupEpi_M2.txt
    Ignored:    output/SEM_summary_groupEpi_M_all.txt
    Ignored:    output/SEM_summary_groupEpi_TopHit.txt
    Ignored:    output/SEM_summary_groupEpi_all.txt
    Ignored:    output/SEMplot_Epi_M15.html
    Ignored:    output/SEMplot_Epi_M15.png
    Ignored:    output/SEMplot_Epi_M15_files/
    Ignored:    output/SEMplot_Epi_M2.html
    Ignored:    output/SEMplot_Epi_M2.png
    Ignored:    output/SEMplot_Epi_M2_files/
    Ignored:    output/SEMplot_Epi_M_all.html
    Ignored:    output/SEMplot_Epi_M_all.png
    Ignored:    output/SEMplot_Epi_M_all_files/
    Ignored:    output/SEMplot_Epi_TopHit.html
    Ignored:    output/SEMplot_Epi_TopHit.png
    Ignored:    output/SEMplot_Epi_TopHit_files/
    Ignored:    output/SEMplot_Epi_all.html
    Ignored:    output/SEMplot_Epi_all.png
    Ignored:    output/SEMplot_Epi_all_files/
    Ignored:    output/barplots.pdf
    Ignored:    output/circos_DMR_tags.svg
    Ignored:    output/circos_LME_tags.svg
    Ignored:    output/clinFact.RData
    Ignored:    output/dds_filt_analyzed.RData
    Ignored:    output/designh0.RData
    Ignored:    output/designh1.RData
    Ignored:    output/envFact.RData
    Ignored:    output/functional_Enrichemnt.pdf
    Ignored:    output/gostres.pdf
    Ignored:    output/modelFact.RData
    Ignored:    output/resdmr.RData
    Ignored:    output/resultsdmr_table.RData
    Ignored:    output/table1_filtered.Rmd
    Ignored:    output/table1_filtered.docx
    Ignored:    output/table1_unfiltered.Rmd
    Ignored:    output/table1_unfiltered.docx
    Ignored:    setup_built.R

Unstaged changes:
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/04_Plot_Results.Rmd</code>) and HTML (<code>docs/04_Plot_Results.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/70cd649ce41081b3d0bade1bd0cc176931c783f4/docs/04_Plot_Results.html" target="_blank">70cd649</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2a53a8725668ff1a29d6f35437a6c587012809ac/docs/04_Plot_Results.html" target="_blank">2a53a87</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/e4425b8c375b57de693db081af5e6dcdf0d2dfd2/analysis/04_Plot_Results.Rmd" target="_blank">e4425b8</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-06
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/e3a9ae36c28b2ba178d5307fea867742cebed422/docs/04_Plot_Results.html" target="_blank">e3a9ae3</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f6bbdc0c4b54e5c101704cc3c7cb9c80408f92b2/docs/04_Plot_Results.html" target="_blank">f6bbdc0</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/710904a781215272de2c506a1e4c29f7fc802778/docs/04_Plot_Results.html" target="_blank">710904a</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/16112c31a3720918231ac96f98ae578113c75dc9/docs/04_Plot_Results.html" target="_blank">16112c3</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, “docs/”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/cde8384951ac43972b93866ec2512bb1ad4cc315/docs/04_Plot_Results.html" target="_blank">cde8384</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/d3629d5f21b4307bf4b789f9e72542404fe10912/analysis/04_Plot_Results.Rmd" target="_blank">d3629d5</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/d761be40b9985b450720b472375b01adc12b63d6/docs/04_Plot_Results.html" target="_blank">d761be4</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/b452d2f02a4bb8e5c4dd961ce50f42a703696053/docs/04_Plot_Results.html" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2734c4e51aa341b50f59acf0df2900ab91d06654/docs/04_Plot_Results.html" target="_blank">2734c4e</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/a8478239ce3ca497da23250c261c259b1d1f9088/docs/04_Plot_Results.html" target="_blank">a847823</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/9cc52f7bc59be3ada26bac3a9458fb6dbb154395/docs/04_Plot_Results.html" target="_blank">9cc52f7</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/158d0b43dc3337bef41cfd99c32e3322750945f4/docs/04_Plot_Results.html" target="_blank">158d0b4</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/0f262d1b920469dca52216115355a20663827e5e/docs/04_Plot_Results.html" target="_blank">0f262d1</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/5167b90e01239a0adae727c7561e97a2ae16a750/docs/04_Plot_Results.html" target="_blank">5167b90</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/05aac7fc7613833ecedbf0ff9dd4fd18088608fc/docs/04_Plot_Results.html" target="_blank">05aac7f</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f5c52650307ec764d17e5f70981ad2cab2dc514b/docs/04_Plot_Results.html" target="_blank">f5c5265</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/dc9e069fc26e9867005f94c140185f870dca8306/analysis/04_Plot_Results.Rmd" target="_blank">dc9e069</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/17f1eec7fe797b93bcf724547cf86ce90c52e37f/docs/04_Plot_Results.html" target="_blank">17f1eec</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/91de221841561ea6b43fb67c9fd06d813bdb04b4/docs/04_Plot_Results.html" target="_blank">91de221</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b6c6b33c63a1e7fcceeb14961390eedf6a45dc19/analysis/04_Plot_Results.Rmd" target="_blank">b6c6b33</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-05
</td>
<td>
updated GO function, and model def
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/4ea1bba17a570a8d27e65e5b4153833c4d999e77/docs/04_Plot_Results.html" target="_blank">4ea1bba</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/6c21638381f6c8e162516fa9f9278f8e82503ab4/analysis/04_Plot_Results.Rmd" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/6c21638381f6c8e162516fa9f9278f8e82503ab4/docs/04_Plot_Results.html" target="_blank">6c21638</a>
</td>
<td>
achiocch
</td>
<td>
2021-02-25
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"), update = F)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>Home = getwd()</code></pre>
<pre class="r"><code>Dark8 = brewer.pal(8, &quot;Dark2&quot;)

source(paste0(Home,&quot;/code/custom_functions.R&quot;))</code></pre>
<pre><code>Lade nötiges Paket: kableExtra</code></pre>
<pre><code>Lade nötiges Paket: tidyverse</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>v ggplot2 3.3.3     v purrr   0.3.4
v tibble  3.0.4     v dplyr   1.0.2
v tidyr   1.1.2     v stringr 1.4.0
v readr   1.4.0     v forcats 0.5.0</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::collapse()   masks IRanges::collapse()
x dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()
x dplyr::count()      masks matrixStats::count()
x dplyr::desc()       masks IRanges::desc()
x tidyr::expand()     masks S4Vectors::expand()
x dplyr::filter()     masks stats::filter()
x dplyr::first()      masks S4Vectors::first()
x dplyr::group_rows() masks kableExtra::group_rows()
x dplyr::lag()        masks stats::lag()
x ggplot2::Position() masks BiocGenerics::Position(), base::Position()
x purrr::reduce()     masks GenomicRanges::reduce(), IRanges::reduce()
x dplyr::rename()     masks S4Vectors::rename()
x dplyr::slice()      masks IRanges::slice()</code></pre>
<pre><code>Lade nötiges Paket: compareGroups</code></pre>
<pre><code>Lade nötiges Paket: scales</code></pre>
<pre><code>
Attache Paket: &#39;scales&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    discard</code></pre>
<pre><code>The following object is masked from &#39;package:readr&#39;:

    col_factor</code></pre>
<pre class="r"><code>load(paste0(Home,&quot;/output/dds_filt_analyzed.RData&quot;))
load(paste0(Home,&quot;/output/resultsdmr_table.RData&quot;))


Patdata=colData(dds_filt)
cpm=counts(dds_filt, normalized=T)
log2_cpm=log2(cpm+1)
res=results(dds_filt)</code></pre>
<div id="plot-significant-lme-hits" class="section level2">
<h2>Plot significant LME hits</h2>
<pre class="r"><code>GRresultslme_table = rowRanges(dds_filt)

DFtoplotall=as.data.frame(GRresultslme_table)
colnames(DFtoplotall)[1:3] = c(&quot;Chromosome&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;)
DFtoplotall = DFtoplotall[order(DFtoplotall$Chromosome, DFtoplotall$chromStart),] 
DFtoplotall$Chromosome = as.factor(DFtoplotall$Chromosome)

countspergroup &lt;- data.frame(cases=rowSums(cpm[,dds_filt$group ==&quot;CD&quot;])/sum(dds_filt$group ==&quot;CD&quot;),
                             controls=rowSums(cpm[,dds_filt$group ==&quot;CTRL&quot;])/sum(dds_filt$group ==&quot;CTRL&quot;))

DFtoplotall$MeanCD = log2(countspergroup[,&quot;cases&quot;]+1)
DFtoplotall$MeanCTRL = log2(countspergroup[,&quot;controls&quot;]+1)

DFtoplot = DFtoplotall[DFtoplotall$WaldPvalue_groupCD &lt;= cutoff,]

colnames(DFtoplot) = gsub(&quot;WaldPvalue&quot;, &quot;-log10_P&quot;, colnames(DFtoplot))

targetlist=list(inner=c(&quot;-log10_P_contraceptivesyes&quot;,
                        &quot;-log10_P_cigday_1&quot;,&quot;-log10_P_Age&quot;), 
                outer=c(&quot;MeanCD&quot;, &quot;MeanCTRL&quot;,&quot;-log10_P_groupCD&quot;))

svgfile = paste0(Home,&quot;/output/circos_LME_tags.svg&quot;)

topindex=rownames(DFtoplotall)[which(res$pvalue %in% sort(res$pvalue, decreasing = F)[1:10])]

plotcircos(plotdata =DFtoplot, targets = targetlist, 
           labcol=&quot;gene&quot;, 
           title=&quot;Differentially Methylated Tag&quot;,
           pvalident = &quot;log10_P&quot;, #collabel for pval identification
           pvallog=FALSE, # is log transformed
           cutoffpval = 8, #after this cutoff will be black
           labelsidx=topindex,
           filename = svgfile)</code></pre>
<pre><code>
RCircos.Core.Components initialized.
Type ?RCircos.Reset.Plot.Parameters to see how to modify the core components.</code></pre>
<pre><code>pdf 
  2 </code></pre>
<p><img src="C:/Users/chiocchetti/Projects/femNATCD_MethSeq/output/circos_LME_tags.svg"/></p>
<pre class="r"><code>#plot best hit 


# settings Genomic annotation 

scheme &lt;- getScheme()
scheme$GdObject$background.panel = &quot;white&quot;
scheme$GdObject$background.title = &quot;white&quot;
scheme$GdObject$fontcolor.title = &quot;black&quot;

scheme$AnnotationTrack$featureAnnotation = NULL
scheme$AnnotationTrack$fill = &quot;gray&quot;
scheme$AnnotationTrack$fontcolor.item = &quot;black&quot;
scheme$AnnotationTrack$cex = 0.5

scheme$DataTrack$col = Dark8
scheme$GeneRegionTrack$fill &lt;- &quot;dodgerblue3&quot;
scheme$GeneRegionTrack$col &lt;- NULL
scheme$GeneRegionTrack$transcriptAnnotation &lt;- NULL

window=2000


index = which(res$pvalue == min(res$pvalue))
subtitel=paste0(DFtoplotall$Chromosome[index],&quot;: &quot;, DFtoplotall$chromStart[index])


chr=DFtoplotall$Chromosome[index]
from = DFtoplotall$chromStart[index]-window
to = DFtoplotall$chromEnd[index]+window

targetrange = GRanges(seqnames = chr, 
                      IRanges(from,
                              to))

tbl.gene = NULL
attempt = 0
while(is.null(tbl.gene) &amp; attempt&lt;10){
  print(paste(&quot;tbl.gene attempt:&quot;, attempt+1))
  if (attempt != 0){
    Sys.sleep(60)}
  
  try(tbl.gene &lt;- getTable(ucscTableQuery(mySession, 
                                          track=&quot;refSeqComposite&quot;, 
                                          range=targetrange, 
                                          table=&quot;ncbiRefSeq&quot;)))
  attempt = attempt+1
}</code></pre>
<pre><code>[1] &quot;tbl.gene attempt: 1&quot;</code></pre>
<pre class="r"><code>tbl.cpg = NULL
attempt = 0
while(is.null(tbl.cpg) &amp; attempt&lt;10){
  print(paste(&quot;tbl.cpg attempt:&quot;, attempt+1))
  if (attempt != 0){
    Sys.sleep(60)}
  try(tbl.cpg &lt;- getTable(ucscTableQuery(mySession, 
                                         track=&quot;cpgIslandExt&quot;,
                                         range=targetrange, 
                                         table=&quot;cpgIslandExt&quot;)))
  attempt = attempt+1
}</code></pre>
<pre><code>[1] &quot;tbl.cpg attempt: 1&quot;</code></pre>
<pre class="r"><code>tbl.TFBdg = NULL
attempt = 0
while(is.null(tbl.TFBdg) &amp; attempt&lt;10){
  print(paste(&quot;tbl.TFBdg attempt:&quot;, attempt+1))
  if (attempt != 0){
    Sys.sleep(60)}
  
  try(tbl.TFBdg &lt;- getTable(ucscTableQuery(mySession, 
                                           track=&quot;tfbsConsSites&quot;,
                                           range=targetrange, 
                                           table=&quot;tfbsConsSites&quot;)))
  attempt = attempt+1
}</code></pre>
<pre><code>[1] &quot;tbl.TFBdg attempt: 1&quot;</code></pre>
<pre class="r"><code>tbl.gene_GR &lt;- convertUCSCtoGR(tbl.gene, col.start = &quot;txStart&quot;,col.end = &quot;txEnd&quot;, col.strand = &quot;strand&quot;)
tbl.cpg_GR &lt;- convertUCSCtoGR(tbl.cpg)
tbl.TFBdg_GR &lt;- convertUCSCtoGR(tbl.TFBdg)

itrack &lt;- IdeogramTrack(genome = &quot;hg19&quot;, chromosome = as.character(DFtoplotall$Chromosome[index]))
atrack &lt;- GenomeAxisTrack()


dataplot=subsetByOverlaps(GRresultslme_table, targetrange)

indexreads=findOverlaps(GRresultslme_table, targetrange)

if(length(indexreads)&gt;1){
  targetcpm=log2_cpm[indexreads@from,]
  tmp = dataplot
  values(tmp)=targetcpm} else {
    targetcpm=t(as.dataframe(log2_cpm[indexreads@from,]))
    tmp = dataplot
    values(tmp)=targetcpm
  }



addScheme(scheme, &quot;myScheme&quot;)
options(Gviz.scheme = &quot;myScheme&quot;)

ptrack &lt;- DataTrack(dataplot, data = -log10(dataplot$WaldPvalue_groupCD), baseline=0,
                    name = &quot;P_group&quot;, type=c(&quot;histogram&quot;), fill=&quot;black&quot;, col = &quot;black&quot;, 
                    col.baseline = &quot;grey&quot;)

dtrack &lt;- DataTrack(tmp, groups=Patdata$group,
                    name = &quot;mean log2(cpm) [SD])&quot;, c(&quot;heatmap&quot;))
displayPars(dtrack) &lt;- list(type=c(&quot;a&quot;,&quot;confint&quot;))


tbl.gene_GR$symbol = tbl.gene_GR$name2
genotrack &lt;- GeneRegionTrack(tbl.gene_GR, name = &quot;genes&quot;, transcriptAnnotation=&quot;symbol&quot;)

tbl.cpg_GR$symbol = tbl.cpg_GR$name
cpgtrack &lt;- GeneRegionTrack(tbl.cpg_GR, name = &quot;GpG&quot;, transcriptAnnotation = &quot;symbol&quot;)
displayPars(cpgtrack)&lt;- list(col=&quot;white&quot;, fill =Dark8[5])

values(tbl.TFBdg_GR)$symbol = tbl.TFBdg_GR$name
tfbtrack &lt;- GeneRegionTrack(tbl.TFBdg_GR, name = &quot;TF-Sites&quot;, transcriptAnnotation = &quot;symbol&quot;)
displayPars(tfbtrack)&lt;- list(col=&quot;white&quot;, fill =Dark8[6])


ncols &lt;- 2
nrows &lt;- 1

grid.newpage()
pushViewport(viewport(layout = grid.layout(nrows, ncols, widths=c(1,2))))
pushViewport(viewport(layout.pos.col = 2,layout.pos.row = 1))
p2 = plotTracks(list(itrack,atrack, genotrack, dtrack, ptrack, cpgtrack, tfbtrack), from = from, to = to, sizes=c(1,1,1,4,2,1,4), add=TRUE)
upViewport()
pushViewport(viewport(layout.pos.col = 1,layout.pos.row = 1))


tmp = data.frame(log2_cpm = log2_cpm[index, ], group=Patdata$group)

p1 = ggplot(tmp, aes(x=group, y=log2_cpm, fill=group)) + geom_boxplot(show.legend=T, aes(fill=group)) +
  geom_point(position=position_jitterdodge(jitter.width=0.5, dodge.width = 0.3))+scale_fill_manual(values = c(&quot;CTRL&quot; = Dark8[1],&quot;CD&quot;=Dark8[2]))
grid.draw(as.grob(p1))
upViewport()

grid.export(paste0(Home,&quot;/output/LME_tophit.svg&quot;))</code></pre>
<pre><code>Warning in grabDL(warn, wrap, wrap.grobs, ...): one or more grobs overwritten
(grab WILL not be faithful; try &#39;wrap.grobs = TRUE&#39;)</code></pre>
<pre><code>Warning in svgStyleAttributes(style, svgdev): Removing non-SVG style attribute
name(s): subscripts, group.number, group.value, span, degree, family, evaluation

Warning in svgStyleAttributes(style, svgdev): Removing non-SVG style attribute
name(s): subscripts, group.number, group.value, span, degree, family, evaluation</code></pre>
<p><img src="figure/04_Plot_Results.Rmd/plot_region-1.png" width="7500" style="display: block; margin: auto;" /><img src="figure/04_Plot_Results.Rmd/plot_region-2.png" width="7500" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-dmr-results" class="section level2">
<h2>Plot DMR results</h2>
<pre class="r"><code>DFtoplotall = resultsdmr_table
colnames(DFtoplotall)[1:3] = c(&quot;Chromosome&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;)
DFtoplotall = DFtoplotall[order(DFtoplotall$Chromosome, DFtoplotall$chromStart),] 
DFtoplotall$Chromosome = as.factor(DFtoplotall$Chromosome)


DFtoplot = DFtoplotall[(DFtoplotall$p.value&lt;=cutoff | DFtoplotall$p.valueArea&lt;cutoff), ]

targetlist=list(inner=c(&quot;p.value&quot;,
                        &quot;p.valueArea&quot;), 
                outer=c(&quot;clusterL&quot;))

svgfile = paste0(Home,&quot;/output/circos_DMR_tags.svg&quot;)

topindex=rownames(DFtoplot)[which(DFtoplot$p.value %in% sort(DFtoplot$p.value, decreasing = F)[1:10])]
plotcircos(plotdata =DFtoplot, targets = targetlist, 
           labcol=&quot;name&quot;, 
           title=&quot;Differentially Methylated Regions&quot;,
           pvalident = &quot;p.value&quot;, #collabel for pval identification
           pvallog=FALSE, # is log transfromed
           cutoffpval = 8, #after this cutoff will be black
           labelsidx=topindex,
           filename = svgfile)</code></pre>
<pre><code>
RCircos.Core.Components initialized.
Type ?RCircos.Reset.Plot.Parameters to see how to modify the core components.</code></pre>
<pre><code>pdf 
  2 </code></pre>
<pre class="r"><code>window=2000
index = which.min(resultsdmr_table$p.value)

chr=resultsdmr_table$chr[index]
from = resultsdmr_table$start[index]-window
to = resultsdmr_table$end[index]+window

targetrange = GRanges(seqnames = chr, 
                      IRanges(from,
                              to))

tbl.gene &lt;- getTable(ucscTableQuery(mySession, track=&quot;refSeqComposite&quot;,range=targetrange, table=&quot;ncbiRefSeq&quot;))
tbl.cpg &lt;- getTable(ucscTableQuery(mySession, track=&quot;cpgIslandExt&quot;,range=targetrange, table=&quot;cpgIslandExt&quot;))
tbl.TFBdg &lt;- getTable(ucscTableQuery(mySession, track=&quot;tfbsConsSites&quot;,range=targetrange, table=&quot;tfbsConsSites&quot;))

tbl.gene_GR &lt;- convertUCSCtoGR(tbl.gene, col.start = &quot;txStart&quot;,col.end = &quot;txEnd&quot;, col.strand = &quot;strand&quot;)
tbl.cpg_GR &lt;- convertUCSCtoGR(tbl.cpg)
tbl.TFBdg_GR &lt;- convertUCSCtoGR(tbl.TFBdg)

itrack &lt;- IdeogramTrack(genome = &quot;hg19&quot;, chromosome = as.character(chr))
atrack &lt;- GenomeAxisTrack()


dataplot=subsetByOverlaps(GRresultslme_table, targetrange)
indexreads=findOverlaps(GRresultslme_table, targetrange)
selected = which.min(as.data.frame(GRresultslme_table)[indexreads@from, &quot;WaldPvalue_groupCD&quot;])


if(length(indexreads)&gt;1){
  targetcpm=log2_cpm[indexreads@from,]
  tmp = dataplot
  values(tmp)=targetcpm} else {
    targetcpm=t(as.dataframe(log2_cpm[indexreads@from,]))
    tmp = dataplot
    values(tmp)=targetcpm
  }


ptrack &lt;- DataTrack(dataplot, data = -log10(dataplot$WaldPvalue_groupCD), baseline=0,
                    name = &quot;P_group&quot;, type=c(&quot;histogram&quot;), fill=&quot;black&quot;, col = &quot;black&quot;, 
                    col.baseline = &quot;grey&quot;)

dtrack &lt;- DataTrack(tmp, groups=Patdata$group,
                    name = &quot;mean log2(cpm) [SD])&quot;, c(&quot;heatmap&quot;))
displayPars(dtrack) &lt;- list(type=c(&quot;a&quot;,&quot;confint&quot;))


tbl.gene_GR$symbol = tbl.gene_GR$name2
genotrack &lt;- GeneRegionTrack(tbl.gene_GR, name = &quot;genes&quot;, transcriptAnnotation=&quot;symbol&quot;)

tbl.cpg_GR$symbol = tbl.cpg_GR$name
cpgtrack &lt;- GeneRegionTrack(tbl.cpg_GR, name = &quot;GpG&quot;, transcriptAnnotation = &quot;symbol&quot;)
displayPars(cpgtrack)&lt;- list(col=&quot;white&quot;, fill =Dark8[5])

values(tbl.TFBdg_GR)$symbol = tbl.TFBdg_GR$name
tfbtrack &lt;- GeneRegionTrack(tbl.TFBdg_GR, name = &quot;TF-Sites&quot;, transcriptAnnotation = &quot;symbol&quot;)
displayPars(tfbtrack)&lt;- list(col=&quot;white&quot;, fill =Dark8[6])


ncols &lt;- 2
nrows &lt;- 1

grid.newpage()
pushViewport(viewport(layout = grid.layout(nrows, ncols, widths=c(1,2))))
pushViewport(viewport(layout.pos.col = 2,layout.pos.row = 1))
p2 = plotTracks(list(itrack,atrack, genotrack, dtrack, ptrack, cpgtrack, tfbtrack), 
                from = from, to = to, sizes=c(1,2,2,4,2,1,4), add=T)
upViewport()
pushViewport(viewport(layout.pos.col = 1,layout.pos.row = 1))

tmp = data.frame(log2_cpm = log2_cpm[indexreads@from[selected],], group=Patdata$group)

p1 = ggplot(tmp, aes(x=group, y=log2_cpm, fill=group)) + geom_boxplot(show.legend=T, aes(fill=group)) +
  geom_point(position=position_jitterdodge(jitter.width=0.5, dodge.width = 0.3))+scale_fill_manual(values = c(&quot;CTRL&quot; = Dark8[1],&quot;CD&quot;=Dark8[2]))
grid.draw(as.grob(p1))
upViewport()</code></pre>
<p><img src="figure/04_Plot_Results.Rmd/plot_results_DMR-1.png" width="5000" style="display: block; margin: auto;" /></p>
<p><img src="C:/Users/chiocchetti/Projects/femNATCD_MethSeq/output/circos_DMR_tags.svg"/></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] scales_1.1.1                compareGroups_4.4.6        
 [3] forcats_0.5.0               stringr_1.4.0              
 [5] dplyr_1.0.2                 purrr_0.3.4                
 [7] readr_1.4.0                 tidyr_1.1.2                
 [9] tibble_3.0.4                ggplot2_3.3.3              
[11] tidyverse_1.3.0             kableExtra_1.3.1           
[13] gridSVG_1.7-2               ggplotify_0.0.5            
[15] rtracklayer_1.49.5          Gviz_1.34.0                
[17] RColorBrewer_1.1-2          beeswarm_0.2.3             
[19] RCircos_1.2.1               DESeq2_1.30.0              
[21] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[23] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[25] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[27] IRanges_2.24.1              S4Vectors_0.28.1           
[29] BiocGenerics_0.36.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] uuid_0.1-4               readxl_1.3.1             backports_1.2.0         
  [4] Hmisc_4.4-2              systemfonts_0.3.2        BiocFileCache_1.14.0    
  [7] lazyeval_0.2.2           splines_4.0.3            BiocParallel_1.24.1     
 [10] digest_0.6.27            ensembldb_2.14.0         htmltools_0.5.1.1       
 [13] fansi_0.4.1              Rsolnp_1.16              magrittr_2.0.1          
 [16] checkmate_2.0.0          memoise_2.0.0            BSgenome_1.58.0         
 [19] cluster_2.1.0            Biostrings_2.58.0        annotate_1.68.0         
 [22] modelr_0.1.8             officer_0.3.16           askpass_1.1             
 [25] prettyunits_1.1.1        jpeg_0.1-8.1             colorspace_2.0-0        
 [28] blob_1.2.1               rvest_0.3.6              rappdirs_0.3.1          
 [31] haven_2.3.1              xfun_0.20                crayon_1.3.4            
 [34] RCurl_1.98-1.2           jsonlite_1.7.2           genefilter_1.72.0       
 [37] survival_3.2-7           VariantAnnotation_1.36.0 glue_1.4.2              
 [40] gtable_0.3.0             zlibbioc_1.36.0          XVector_0.30.0          
 [43] webshot_0.5.2            DelayedArray_0.16.0      DBI_1.1.1               
 [46] Rcpp_1.0.5               viridisLite_0.3.0        xtable_1.8-4            
 [49] progress_1.2.2           htmlTable_2.1.0          gridGraphics_0.5-1      
 [52] foreign_0.8-81           bit_4.0.4                Formula_1.2-4           
 [55] truncnorm_1.0-8          htmlwidgets_1.5.3        httr_1.4.2              
 [58] ellipsis_0.3.1           mice_3.12.0              farver_2.0.3            
 [61] pkgconfig_2.0.3          XML_3.99-0.5             nnet_7.3-15             
 [64] dbplyr_2.0.0             locfit_1.5-9.4           labeling_0.4.2          
 [67] tidyselect_1.1.0         rlang_0.4.10             later_1.1.0.1           
 [70] AnnotationDbi_1.52.0     cellranger_1.1.0         munsell_0.5.0           
 [73] tools_4.0.3              cachem_1.0.1             cli_2.2.0               
 [76] generics_0.1.0           RSQLite_2.2.2            broom_0.7.3             
 [79] evaluate_0.14            fastmap_1.1.0            yaml_2.2.1              
 [82] knitr_1.30               bit64_4.0.5              fs_1.5.0                
 [85] zip_2.1.1                AnnotationFilter_1.14.0  whisker_0.4             
 [88] xml2_1.3.2               biomaRt_2.46.2           compiler_4.0.3          
 [91] rstudioapi_0.13          curl_4.3                 png_0.1-7               
 [94] reprex_1.0.0             geneplotter_1.68.0       HardyWeinberg_1.7.1     
 [97] stringi_1.5.3            ps_1.5.0                 GenomicFeatures_1.42.1  
[100] gdtools_0.2.3            lattice_0.20-41          ProtGenerics_1.22.0     
[103] Matrix_1.2-18            vctrs_0.3.6              pillar_1.4.7            
[106] lifecycle_0.2.0          BiocManager_1.30.10      flextable_0.6.2         
[109] data.table_1.13.6        bitops_1.0-6             httpuv_1.5.5            
[112] R6_2.5.0                 latticeExtra_0.6-29      promises_1.1.1          
[115] gridExtra_2.3            writexl_1.3.1            dichromat_2.0-0         
[118] assertthat_0.2.1         chron_2.3-56             openssl_1.4.3           
[121] rprojroot_2.0.2          withr_2.4.1              GenomicAlignments_1.26.0
[124] Rsamtools_2.6.0          GenomeInfoDbData_1.2.4   hms_1.0.0               
[127] rpart_4.1-15             rmarkdown_2.6            rvcheck_0.1.8           
[130] git2r_0.28.0             biovizBase_1.38.0        lubridate_1.7.9.2       
[133] base64enc_0.1-3         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
