<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="AG Chiocchetti" />


<title>The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">femNATCD_MethSeq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/achiocch/femNATCD_MethSeq">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">The Methylome in Female Adolescent Conduct Disorder: Neural Pathomechanisms and Environmental Risk Factors</h1>
<h3 class="subtitle">Differential Methylation Analysis</h3>
<h4 class="author">AG Chiocchetti</h4>
<h4 class="date">29 Dezember 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>femNATCD_MethSeq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210128code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210128)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210128code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210128)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree1a30b735ba10dd9975520489c931126184fd6967targetblank1a30b73a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/1a30b735ba10dd9975520489c931126184fd6967" target="_blank">1a30b73</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomachiocchfemNATCDMethSeqtree1a30b735ba10dd9975520489c931126184fd6967targetblank1a30b73a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/achiocch/femNATCD_MethSeq/tree/1a30b735ba10dd9975520489c931126184fd6967" target="_blank">1a30b73</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    code/.Rhistory
    Ignored:    data/Epicounts.csv
    Ignored:    data/Epimeta.csv
    Ignored:    data/Epitpm.csv
    Ignored:    data/KangUnivers.txt
    Ignored:    data/Kang_DataPreprocessing.RData
    Ignored:    data/Kang_dataset_genesMod_version2.txt
    Ignored:    data/PatMeta.csv
    Ignored:    data/ProcessedData.RData
    Ignored:    data/RTrawdata/
    Ignored:    data/SNPCommonFilt.csv
    Ignored:    data/femNAT_PC20.txt
    Ignored:    output/4A76FA10
    Ignored:    output/BrainMod_Enrichemnt.pdf
    Ignored:    output/Brain_Module_Heatmap.pdf
    Ignored:    output/DMR_Results.csv
    Ignored:    output/GOres.xlsx
    Ignored:    output/LME_GOplot.pdf
    Ignored:    output/LME_GOplot.svg
    Ignored:    output/LME_Results.csv
    Ignored:    output/LME_Results_Sig.csv
    Ignored:    output/LME_Results_Sig_mod.csv
    Ignored:    output/LME_tophit.svg
    Ignored:    output/ProcessedData.RData
    Ignored:    output/RNAvsMETplots.pdf
    Ignored:    output/Regions_GOplot.pdf
    Ignored:    output/Regions_GOplot.svg
    Ignored:    output/ResultsgroupComp.txt
    Ignored:    output/SEM_summary_groupEpi_M15.txt
    Ignored:    output/SEM_summary_groupEpi_M2.txt
    Ignored:    output/SEM_summary_groupEpi_M_all.txt
    Ignored:    output/SEM_summary_groupEpi_TopHit.txt
    Ignored:    output/SEM_summary_groupEpi_all.txt
    Ignored:    output/SEMplot_Epi_M15.html
    Ignored:    output/SEMplot_Epi_M15.png
    Ignored:    output/SEMplot_Epi_M15_files/
    Ignored:    output/SEMplot_Epi_M2.html
    Ignored:    output/SEMplot_Epi_M2.png
    Ignored:    output/SEMplot_Epi_M2_files/
    Ignored:    output/SEMplot_Epi_M_all.html
    Ignored:    output/SEMplot_Epi_M_all.png
    Ignored:    output/SEMplot_Epi_M_all_files/
    Ignored:    output/SEMplot_Epi_TopHit.html
    Ignored:    output/SEMplot_Epi_TopHit.png
    Ignored:    output/SEMplot_Epi_TopHit_files/
    Ignored:    output/SEMplot_Epi_all.html
    Ignored:    output/SEMplot_Epi_all.png
    Ignored:    output/SEMplot_Epi_all_files/
    Ignored:    output/barplots.pdf
    Ignored:    output/circos_DMR_tags.svg
    Ignored:    output/circos_LME_tags.svg
    Ignored:    output/clinFact.RData
    Ignored:    output/dds_filt_analyzed.RData
    Ignored:    output/designh0.RData
    Ignored:    output/designh1.RData
    Ignored:    output/envFact.RData
    Ignored:    output/functional_Enrichemnt.pdf
    Ignored:    output/gostres.pdf
    Ignored:    output/modelFact.RData
    Ignored:    output/resdmr.RData
    Ignored:    output/resultsdmr_table.RData
    Ignored:    output/table1_filtered.Rmd
    Ignored:    output/table1_filtered.docx
    Ignored:    output/table1_unfiltered.Rmd
    Ignored:    output/table1_unfiltered.docx
    Ignored:    setup_built.R

Unstaged changes:
    Deleted:    analysis/RNAvsMETplots.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/99_helperforPaper.Rmd</code>) and HTML (<code>docs/99_helperforPaper.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/710904a781215272de2c506a1e4c29f7fc802778/docs/99_helperforPaper.html" target="_blank">710904a</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/16112c31a3720918231ac96f98ae578113c75dc9/docs/99_helperforPaper.html" target="_blank">16112c3</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, “docs/”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/cde8384951ac43972b93866ec2512bb1ad4cc315/docs/99_helperforPaper.html" target="_blank">cde8384</a>
</td>
<td>
achiocch
</td>
<td>
2021-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/d761be40b9985b450720b472375b01adc12b63d6/docs/99_helperforPaper.html" target="_blank">d761be4</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/b452d2f02a4bb8e5c4dd961ce50f42a703696053/analysis/99_helperforPaper.Rmd" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/b452d2f02a4bb8e5c4dd961ce50f42a703696053/docs/99_helperforPaper.html" target="_blank">b452d2f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/1a9f36fde268d7bd9b5615ff46b35309442db585/analysis/99_helperforPaper.Rmd" target="_blank">1a9f36f</a>
</td>
<td>
achiocch
</td>
<td>
2021-07-30
</td>
<td>
reviewed analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/2734c4e51aa341b50f59acf0df2900ab91d06654/docs/99_helperforPaper.html" target="_blank">2734c4e</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/a8478239ce3ca497da23250c261c259b1d1f9088/docs/99_helperforPaper.html" target="_blank">a847823</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/9cc52f7bc59be3ada26bac3a9458fb6dbb154395/docs/99_helperforPaper.html" target="_blank">9cc52f7</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/158d0b43dc3337bef41cfd99c32e3322750945f4/docs/99_helperforPaper.html" target="_blank">158d0b4</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-08
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/0f262d1b920469dca52216115355a20663827e5e/docs/99_helperforPaper.html" target="_blank">0f262d1</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/5167b90e01239a0adae727c7561e97a2ae16a750/docs/99_helperforPaper.html" target="_blank">5167b90</a>
</td>
<td>
achiocch
</td>
<td>
2021-05-07
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/05aac7fc7613833ecedbf0ff9dd4fd18088608fc/docs/99_helperforPaper.html" target="_blank">05aac7f</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/achiocch/femNATCD_MethSeq/f5c52650307ec764d17e5f70981ad2cab2dc514b/docs/99_helperforPaper.html" target="_blank">f5c5265</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/achiocch/femNATCD_MethSeq/blob/dc9e069fc26e9867005f94c140185f870dca8306/analysis/99_helperforPaper.Rmd" target="_blank">dc9e069</a>
</td>
<td>
achiocch
</td>
<td>
2021-04-19
</td>
<td>
wflow_publish(c(“analysis/<em>", "code/</em>”, "docs/*"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>## load Data  ####
load(paste0(Home,&quot;/output/dds_filt_analyzed.RData&quot;))</code></pre>
<pre class="r"><code>restab=results(dds_filt)
selEpitpm = counts(dds_filt, normalized=T)
selEpiMeta = mcols(dds_filt)

alldiff = sum(restab$pvalue&lt;=0.001)
neg_LFC = sum(restab$pvalue&lt;=0.001 &amp; restab$log2FoldChange&lt;0)
pos_LFC = sum(restab$pvalue&lt;=0.001&amp; restab$log2FoldChange&gt;0)

head(restab[order(restab$pvalue, decreasing = F), ])</code></pre>
<pre><code>log2 fold change (MLE): group CD vs CTRL 
Wald test p-value: groupCD 
DataFrame with 6 rows and 6 columns
   baseMean log2FoldChange     lfcSE      stat      pvalue      padj
  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;
1  13.29401      -1.128424  0.201492  -5.60034 2.13936e-08 0.0046232
2  14.79801       0.791735  0.173707   4.55787 5.16740e-06 0.3075086
3   5.31577      -1.542914  0.338922  -4.55242 5.30329e-06 0.3075086
4  36.54142      -0.467965  0.103132  -4.53752 5.69192e-06 0.3075086
5  18.59065      -0.685889  0.154236  -4.44703 8.70673e-06 0.3212312
6   9.82539       1.042628  0.234728   4.44185 8.91888e-06 0.3212312</code></pre>
<pre class="r"><code>GOtab = readxl::read_xlsx(paste0(Home, &quot;/output/GOres.xlsx&quot;))</code></pre>
<pre><code>New names:
* `` -&gt; ...1</code></pre>
<pre class="r"><code>GOtab_LME=GOtab[GOtab$query==&quot;02_dmtag&quot;,]
GOtab_Reg=GOtab[GOtab$query==&quot;01_dmregions&quot;,]

GOtable = GOtab_LME
N = 15 

GOplot = function(GOtable, N){
  if(nrow(GOtable)&lt;N){N=nrow(GOtable)}
  GOtable = GOtable[GOtable$parents!=&quot;character(0)&quot;,]
  Tabtoplot=GOtable[order(GOtable$p_value, decreasing = F)[1:N],]
  Tabtoplot$&quot;-log10(pvalue)&quot;=-log10(Tabtoplot$p_value)
  par(mar=c(5,12,5,2))
  genperc=Tabtoplot$intersection_size/Tabtoplot$effective_domain_size
  plot(x=Tabtoplot$&quot;-log10(pvalue)&quot;, y=N:1, 
       xlim=c(0, max(Tabtoplot$&quot;-log10(pvalue)&quot;)), 
       ylab=&quot;&quot;, yaxt=&quot;n&quot;, xlab=&quot;-log10(corr. p-value)&quot;, 
       type=&quot;n&quot;)
    for(i in 1:N){
  lines(y = c(i,i), x=c(-1,rev(Tabtoplot$&quot;-log10(pvalue)&quot;)[i]), lty=2, 
        col=&quot;grey&quot;)
  }
  
  points(, x=Tabtoplot$&quot;-log10(pvalue)&quot;, y=N:1,
         cex=10*Tabtoplot$precision,
         pch=16, 
         col=jetcolors(100)[genperc/max(genperc)*100],)
  axis(2, at = N:1, labels = Tabtoplot$term_name, las=2, cex.axis=0.6)
  par(mar=c(5,0.5,5,0.5))
  
  lgd_ = rep(NA, 11)
  labset=c(0,0.5*max(genperc),max(genperc))
  lgd_[c(1,6,11)] = round(labset*100,2)
  legend(&quot;bottomright&quot;,
       legend = lgd_,
       title = &quot;genomic covarage&quot;,
       fill = jetcolors(11),
       border = NA,
       bty=&quot;none&quot;,
       y.intersp = 0.4,
       cex = 1, 
       text.font = 1)
  
}

pdf(paste0(Home, &quot;/output/LME_GOplot.pdf&quot;))
GOplot(GOtab_LME[GOtab_LME$source==&quot;GO:BP&quot;,], 20)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>pdf(paste0(Home, &quot;/output/Regions_GOplot.pdf&quot;))
GOplot(GOtab_Reg[GOtab_Reg$source==&quot;GO:BP&quot;,], 20)
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>RefGenes = c(&quot;GUSB&quot;)
Targets_of_Int = c(&quot;SLITRK5&quot;, &quot;MIR4500HG&quot;)
nreplicates = 3
flagscore=0.05 #replication quality error

SamplesMeta=read_xlsx(paste0(Home,&quot;/data/RTrawdata/ZelllinienRNA_femNAT.xlsx&quot;))
as.data.frame(SamplesMeta) -&gt; SamplesMeta
                      
SamplesMeta$Pou=paste(&quot;POU&quot;, SamplesMeta$Pou)
rownames(SamplesMeta)=SamplesMeta$Pou

SamplesMeta$Group = dds_filt$group[match(SamplesMeta$femNATID, dds_filt$ID_femNAT)]


Files=list.files(paste0(Home,&quot;/data/RTrawdata/&quot;), full.names = T)

Files=Files[grepl(&quot;_data&quot;,Files)]

Sets=unique(substr(basename(Files), 1,8))

Targets_all=vector()
Samples_all=vector()


geoMean=function(x){
  x=x[!is.na(x)]
  if(length(x)==0)
    return(NA)
  else
    return((prod(x))^(1/length(x)))}

for (Set in Sets){

    Setfiles=Files[grep(Set, Files)]

    for( i in 1:length(Setfiles)){
      tmp=read.table(Setfiles[i], skip=8, header=T, sep=&quot;\t&quot;, comment.char = &quot;&quot;, fill=T)[1:96,]
      tmp=tmp[,c(&quot;Sample.Name&quot;, &quot;Target.Name&quot;,&quot;CÑ.&quot;)]
      colnames(tmp)=c(&quot;Sample.Name&quot;, &quot;Target.Name&quot;, &quot;CT&quot;)
      tmp$Target.Name=gsub(&quot;SLITRK5_L&quot;, &quot;SLITRK5_&quot;, tmp$Target.Name)
      tmp$Target.Name=gsub(&quot;VD_&quot;, &quot;&quot;, tmp$Target.Name)
      tmp$Target.Name=gsub(&quot;_&quot;, &quot;&quot;, tmp$Target.Name)
      tmp$Target.Name=substr(tmp$Target.Name,1, regexpr(&quot;#&quot;, tmp$Target.Name)-1)
      tmp$CT=as.numeric(tmp$CT)
      
      # set bad replicates to NA 
      tmpmu = tapply(tmp$CT, paste0(tmp$Sample.Name,&quot;_&quot;,tmp$Target.Name), mean, na.rm=T)
      tmpsd = tapply(tmp$CT, paste0(tmp$Sample.Name,&quot;_&quot;,tmp$Target.Name), sd, na.rm=T)
      for (corr in which(tmpsd/tmpmu&gt;flagscore)){
        index=unlist(strsplit(names(tmpmu)[corr], &quot;_&quot;))
        tmp[which(tmp$Sample.Name==index[1] &amp; tmp$Target.Name==index[2]),&quot;CT&quot;] = NA
      }
      assign(paste0(&quot;tmp_&quot;,Set,&quot;_&quot;,i),tmp)
    }
    
    tmp=do.call(&quot;rbind&quot;, mget(apropos(paste0(&quot;tmp_&quot;,Set))))
    tmp=tmp[which(!(tmp$Sample.Name==&quot;&quot;|is.na(tmp$Sample.Name))), ]
    tmp=tmp[!tmp$Sample.Name==&quot;NTC&quot;,]
    
    Samples=unique(tmp$Sample.Name)
    Targets=unique(tmp$Target.Name)
    Samples_all=unique(c(Samples_all, Samples))
    Targets_all=unique(c(Targets_all, Targets))
    
    Reform=data.frame(matrix(NA, nrow=length(Samples), ncol=length(Targets)*nreplicates))
    
    colnames(Reform)=paste0(rep(Targets, each=3), letters[1:nreplicates])
    rownames(Reform)=Samples
    
    for (i in Samples) {
      #print(i)
      for (j in Targets){
        Reform[i,grep(j, colnames(Reform))]=tmp[tmp$Sample.Name==i &amp; tmp$Target.Name==j,&quot;CT&quot;]
      }
    }
    
    HK=colnames(Reform)[grep(paste0(RefGenes, collapse=&quot;|&quot;),colnames(Reform))]
    
    GMHK=apply(Reform[,HK], 1, geoMean)
    tmp2=Reform-GMHK
    
    assign(paste0(Set,&quot;_dCT&quot;), tmp2)
    rm(list=c(apropos(&quot;tmp&quot;), &quot;Reform&quot;, &quot;GMHK&quot;))
    
}</code></pre>
<pre><code>Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt

Warning: NAs durch Umwandlung erzeugt</code></pre>
<pre class="r"><code>Samples_all=unique(Samples_all)
Targets_all = unique(Targets_all)</code></pre>
<pre class="r"><code>mergedCTtable=data.frame(matrix(NA,ncol=length(Targets_all)*nreplicates, nrow=length(Samples_all)))
colnames(mergedCTtable)=paste0(rep(unique(Targets_all), each=nreplicates), letters[1:nreplicates])
rownames(mergedCTtable)=Samples_all

CTobj=apropos(&quot;_dCT&quot;)

for( obj in CTobj){
  DF=get(obj)
  for(k in colnames(DF)){
    for(l in rownames(DF)){
      mergedCTtable[l,k]=DF[l,k]
    }
  }
}

CTmeans=colMeans(mergedCTtable, na.rm = T)

meanvec=tapply(CTmeans,gsub(paste0(letters[1:nreplicates],collapse=&quot;|&quot;),&quot;&quot;,names(CTmeans)), mean, na.rm=T)
meanvec = rep(meanvec, each=nreplicates)
names(meanvec) = paste0(names(meanvec), letters[1:nreplicates])
meanvec=meanvec[colnames(mergedCTtable)]
ddCT=apply(mergedCTtable,1, function(x){x-meanvec}) 

FC=2^-ddCT


SamplesMeta$inset=F
SamplesMeta$inset[SamplesMeta$Pou %in% colnames(FC)]=T

SamplesMeta=SamplesMeta[SamplesMeta$inset,]

CTRLCASEsorter=c(which(SamplesMeta$Group==&quot;CTRL&quot;),which(SamplesMeta$Group==&quot;CD&quot;))
SamplesMeta = SamplesMeta[CTRLCASEsorter, ]

searcher=paste0(Targets_of_Int, collapse = &quot;|&quot;)

FC = FC[grepl(searcher, rownames(FC)),SamplesMeta$Pou]

MuFC=apply(FC, 2, function(x){tapply(log2(x), gsub(&quot;a|b|c&quot;,&quot;&quot;,rownames(FC)), mean, na.rm=T)})
SDFC=apply(FC, 2, function(x){tapply(log2(x), gsub(&quot;a|b|c&quot;,&quot;&quot;,rownames(FC)), sd, na.rm=T)})

plot(unlist(MuFC[2, ]), unlist(SDFC[2, ]))</code></pre>
<p><img src="figure/99_helperforPaper.Rmd/RTPCR_step2-1.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(paste0(Home, &quot;/output/barplots.pdf&quot;))

for(i in Targets_of_Int){
  if(any(!is.na(MuFC[i,]))){
  a=barplot(unlist(MuFC[i,]), col=as.numeric(SamplesMeta[colnames(MuFC),&quot;Group&quot;])+1, main=i, las=3, 
            ylim=c(-max(abs(MuFC[i,])*1.2, na.rm=T), (max(abs(MuFC[i,])*1.2, na.rm=T))))
  arrows(a, MuFC[i,], a, MuFC[i,]+SDFC[i,], angle = 90, length = 0.1)
  arrows(a, MuFC[i,], a, MuFC[i,]-SDFC[i,], angle = 90, length = 0.1)
  legend(&quot;topleft&quot;, c(&quot;case&quot;, &quot;control&quot;), col=c(1,2), pch=15, bty=&quot;n&quot;)
  } else {
    plot(0,0, type=&quot;n&quot;, main=paste(i, &quot;not detected&quot;))
  }
}

dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>for(i in Targets_of_Int){
  if(any(!is.na(MuFC[i,]))){
  a=barplot(unlist(MuFC[i,]), col=as.numeric(SamplesMeta[colnames(MuFC),&quot;Group&quot;])+1, main=i, las=3, 
            ylim=c(-max(abs(MuFC[i,])*1.2, na.rm=T), (max(abs(MuFC[i,])*1.2, na.rm=T))))
  arrows(a, MuFC[i,], a, MuFC[i,]+SDFC[i,], angle = 90, length = 0.1)
  arrows(a, MuFC[i,], a, MuFC[i,]-SDFC[i,], angle = 90, length = 0.1)
  legend(&quot;topleft&quot;, c(&quot;case&quot;, &quot;control&quot;), col=c(1,2), pch=15, bty=&quot;n&quot;)
  } else {
    plot(0,0, type=&quot;n&quot;, main=paste(i, &quot;not detected&quot;))
  }
}</code></pre>
<p><img src="figure/99_helperforPaper.Rmd/RTPCR_step3plot-1.png" width="4500" style="display: block; margin: auto;" /><img src="figure/99_helperforPaper.Rmd/RTPCR_step3plot-2.png" width="4500" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sink(paste0(Home, &quot;/output/ResultsgroupComp.txt&quot;))

Group=SamplesMeta$Group
for(i in Targets_of_Int){
  print(i)
  print(summary(try(lm(unlist(MuFC[i,])~Group))))
  print(t.test(unlist(MuFC[i,])~Group))
}</code></pre>
<pre><code>[1] &quot;SLITRK5&quot;

Call:
lm(formula = unlist(MuFC[i, ]) ~ Group)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.3710 -0.4489  0.1142  0.4382  0.9983 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)  -0.1703     0.1718  -0.991    0.334
GroupCD       0.5109     0.2975   1.717    0.102

Residual standard error: 0.6427 on 19 degrees of freedom
Multiple R-squared:  0.1343,    Adjusted R-squared:  0.08878 
F-statistic: 2.949 on 1 and 19 DF,  p-value: 0.1022


    Welch Two Sample t-test

data:  unlist(MuFC[i, ]) by Group
t = -2.0316, df = 18.197, p-value = 0.05706
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.03871726  0.01701686
sample estimates:
mean in group CTRL   mean in group CD 
        -0.1702834          0.3405668 

[1] &quot;MIR4500HG&quot;

Call:
lm(formula = unlist(MuFC[i, ]) ~ Group)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.7172 -1.2116 -0.2660  0.5749  5.4474 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)  -0.6053     0.5683  -1.065    0.301
GroupCD       0.6614     0.9607   0.688    0.500

Residual standard error: 2.049 on 18 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.02566,   Adjusted R-squared:  -0.02847 
F-statistic: 0.474 on 1 and 18 DF,  p-value: 0.4999


    Welch Two Sample t-test

data:  unlist(MuFC[i, ]) by Group
t = -0.65047, df = 10.602, p-value = 0.5292
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -2.909646  1.586853
sample estimates:
mean in group CTRL   mean in group CD 
       -0.60525003         0.05614653 </code></pre>
<pre class="r"><code>sink()

for(i in Targets_of_Int){
  print(i)
  print(summary(try(lm(unlist(MuFC[i,])~Group))))
  print(t.test(unlist(MuFC[i,])~Group, var.qual=F, mu=0))
}</code></pre>
<pre><code>[1] &quot;SLITRK5&quot;

Call:
lm(formula = unlist(MuFC[i, ]) ~ Group)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.3710 -0.4489  0.1142  0.4382  0.9983 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)  -0.1703     0.1718  -0.991    0.334
GroupCD       0.5109     0.2975   1.717    0.102

Residual standard error: 0.6427 on 19 degrees of freedom
Multiple R-squared:  0.1343,    Adjusted R-squared:  0.08878 
F-statistic: 2.949 on 1 and 19 DF,  p-value: 0.1022


    Welch Two Sample t-test

data:  unlist(MuFC[i, ]) by Group
t = -2.0316, df = 18.197, p-value = 0.05706
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -1.03871726  0.01701686
sample estimates:
mean in group CTRL   mean in group CD 
        -0.1702834          0.3405668 

[1] &quot;MIR4500HG&quot;

Call:
lm(formula = unlist(MuFC[i, ]) ~ Group)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.7172 -1.2116 -0.2660  0.5749  5.4474 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)  -0.6053     0.5683  -1.065    0.301
GroupCD       0.6614     0.9607   0.688    0.500

Residual standard error: 2.049 on 18 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.02566,   Adjusted R-squared:  -0.02847 
F-statistic: 0.474 on 1 and 18 DF,  p-value: 0.4999


    Welch Two Sample t-test

data:  unlist(MuFC[i, ]) by Group
t = -0.65047, df = 10.602, p-value = 0.5292
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -2.909646  1.586853
sample estimates:
mean in group CTRL   mean in group CD 
       -0.60525003         0.05614653 </code></pre>
<pre class="r"><code>SamplesMeta$femNATID2=paste0(&quot;ID_&quot;,gsub(&quot;-&quot;,&quot;_&quot;,SamplesMeta$femNATID))

SamplesMeta=SamplesMeta[SamplesMeta$Pou %in% colnames(MuFC),]
MuFC=MuFC[,SamplesMeta$Pou]

TPM4RNA=selEpitpm[,SamplesMeta$femNATID2]
colnames(TPM4RNA)=SamplesMeta$Pou

tags=list()


Targets=Targets_of_Int


sigtags=which(restab$padj&lt;=0.05)

selEpiMeta[sigtags,]</code></pre>
<pre><code>DataFrame with 1 row and 55 columns
           raw_gene              gene    distance     feature         cpg
        &lt;character&gt;       &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;
1 MIR4500HG,SLITRK5 MIR4500HG,SLITRK5           0    upstream         cpg
   tf_binding samples_per_target      mean       std  baseMean   baseVar
  &lt;character&gt;          &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
1    Pol2//GR                100   15.7059   12.6308    13.294   61.7602
    allZero dispGeneEst dispGeneIter   dispFit dispersion  dispIter dispOutlier
  &lt;logical&gt;   &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;integer&gt;   &lt;logical&gt;
1     FALSE    0.176742           11   0.17053   0.175669        11       FALSE
    dispMAP        Age int_disno int_disyes medicationyes contraceptivesyes
  &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;     &lt;numeric&gt;         &lt;numeric&gt;
1  0.175669 -0.0452923   4.74064     5.0314      0.166961         -0.126415
   cigday_1        V8   groupCD    SE_Age SE_int_disno SE_int_disyes
  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;     &lt;numeric&gt;
1 0.0305576  0.706369  -1.12842 0.0539633     0.854072      0.884392
  SE_medicationyes SE_contraceptivesyes SE_cigday_1     SE_V8 SE_groupCD
         &lt;numeric&gt;            &lt;numeric&gt;   &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
1         0.213896             0.191566   0.0173334  0.879379   0.201492
  WaldStatistic_Age WaldStatistic_int_disno WaldStatistic_int_disyes
          &lt;numeric&gt;               &lt;numeric&gt;                &lt;numeric&gt;
1         -0.839318                 5.55064                  5.68911
  WaldStatistic_medicationyes WaldStatistic_contraceptivesyes
                    &lt;numeric&gt;                       &lt;numeric&gt;
1                    0.780573                       -0.659903
  WaldStatistic_cigday_1 WaldStatistic_V8 WaldStatistic_groupCD WaldPvalue_Age
               &lt;numeric&gt;        &lt;numeric&gt;             &lt;numeric&gt;      &lt;numeric&gt;
1                1.76293         0.803259              -5.60034       0.401291
  WaldPvalue_int_disno WaldPvalue_int_disyes WaldPvalue_medicationyes
             &lt;numeric&gt;             &lt;numeric&gt;                &lt;numeric&gt;
1          2.84626e-08           1.27704e-08                 0.435054
  WaldPvalue_contraceptivesyes WaldPvalue_cigday_1 WaldPvalue_V8
                     &lt;numeric&gt;           &lt;numeric&gt;     &lt;numeric&gt;
1                     0.509316           0.0779121      0.421825
  WaldPvalue_groupCD  betaConv  betaIter  deviance  maxCooks
           &lt;numeric&gt; &lt;logical&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;logical&gt;
1        2.13936e-08      TRUE         6   645.105        NA</code></pre>
<pre class="r"><code>tagsOI=grep(paste0(Targets, collapse = &quot;|&quot;),selEpiMeta$gene)

sigtagsOI = tagsOI[tagsOI %in% sigtags]

fintagsOI=data.frame(tags=sigtagsOI, gene=selEpiMeta[sigtagsOI,&quot;gene&quot;])

#Targ=Targets[1]
#tag=tags[1]</code></pre>
<pre class="r"><code>pdf(paste0(Home,&quot;/output/RNAvsMETplots.pdf&quot;), width = 15, height = 8)

MuFC=as.data.frame(MuFC)
MuFCsel=MuFC[Targets,]
par(mar=c(5,5,5,3), mfrow=c(1,2))
for (Targ in Targets){
  tags=fintagsOI$tags[grep(Targ, fintagsOI$gene)]
  for (tag in tags){
    data=data.frame(tpm=unlist(TPM4RNA[tag,SamplesMeta$Pou]) , 
                    RT=unlist(MuFCsel[grep(Targ, rownames(MuFCsel)),SamplesMeta$Pou]))
    plot(data$tpm,data$RT, 
         xlab=&quot;methylation tpm&quot;, 
         ylab = &quot;mRNA log2FC vs mean&quot;, 
         ylim=c(-3,3),
         col=4-as.numeric(SamplesMeta$Group), 
         pch=as.numeric(SamplesMeta$Group)+14,
         main=paste(tag, &quot;Meth vs mRNA Expr&quot;, Targ))
    legend(&quot;topleft&quot;, c(&quot;control&quot;, &quot;case&quot;), pch=c(15,16), col=c(3,2), bty=&quot;n&quot;)
    a=lm(RT~tpm, data)
    b=summary(a)
    abline(a, col=&quot;blue&quot;)

    SperCor=cor(data$RT,data$tpm,use = &quot;c&quot;, method = &quot;spearman&quot;)
    mtext(3, text = paste(&quot;beta = &quot;, round(coefficients(a)[2],2), 
                          &quot;; se =&quot;, round(b$coefficients[2,2],2), 
                          &quot;; pvalue = &quot;, round(b$coefficients[2,4],3), 
                          &quot;; sperman cor = &quot;, round(SperCor,3)))
  }
}

dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>MuFC=as.data.frame(MuFC)
MuFCsel=MuFC[Targets,]
par(mar=c(5,5,5,3), mfrow=c(1,2))
for (Targ in Targets){
  tags=fintagsOI$tags[grep(Targ, fintagsOI$gene)]
  for (tag in tags){
    data=data.frame(tpm=unlist(TPM4RNA[tag,SamplesMeta$Pou]) , 
                    RT=unlist(MuFCsel[grep(Targ, rownames(MuFCsel)),SamplesMeta$Pou]))
    plot(data$tpm,data$RT, 
         xlab=&quot;methylation tpm&quot;, 
         ylab = &quot;mRNA log2FC vs mean&quot;, 
         ylim=c(-3,3),
         col=4-as.numeric(SamplesMeta$Group), 
         pch=as.numeric(SamplesMeta$Group)+14,
         main=paste(tag, &quot;Meth vs mRNA Expr&quot;, Targ))
    legend(&quot;topleft&quot;, c(&quot;control&quot;, &quot;case&quot;), pch=c(15,16), col=c(3,2), bty=&quot;n&quot;)
    a=lm(RT~tpm, data)
    b=summary(a)
    print(b)
    abline(a, col=&quot;blue&quot;)
    SperCor=cor(data$RT,data$tpm,use = &quot;c&quot;, method = &quot;spearman&quot;)
    mtext(3, text = paste(&quot;beta = &quot;, round(coefficients(a)[2],2), 
                          &quot;; se =&quot;, round(b$coefficients[2,2],2), 
                          &quot;; pvalue = &quot;, round(b$coefficients[2,4],3), 
                          &quot;; sperman cor = &quot;, round(SperCor,3)))
  }
}</code></pre>
<pre><code>
Call:
lm(formula = RT ~ tpm, data = data)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.04611 -0.38191  0.09584  0.38038  1.06231 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)  0.66910    0.26738   2.502   0.0216 *
tpm         -0.05479    0.01930  -2.839   0.0105 *
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.5788 on 19 degrees of freedom
Multiple R-squared:  0.2979,    Adjusted R-squared:  0.2609 
F-statistic: 8.061 on 1 and 19 DF,  p-value: 0.01049</code></pre>
<p><img src="figure/99_helperforPaper.Rmd/RTPCR_step4%20test_plot-1.png" width="7500" style="display: block; margin: auto;" /></p>
<pre><code>
Call:
lm(formula = RT ~ tpm, data = data)

Residuals:
    Min      1Q  Median      3Q     Max 
-2.9602 -0.9674 -0.3599  0.7415  5.1303 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)
(Intercept)  0.77882    0.93125   0.836    0.414
tpm         -0.09302    0.06621  -1.405    0.177

Residual standard error: 1.971 on 18 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.09882,   Adjusted R-squared:  0.04876 
F-statistic: 1.974 on 1 and 18 DF,  p-value: 0.1771</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scales_1.1.1                RCircos_1.2.1              
 [3] compareGroups_4.4.6         kableExtra_1.3.1           
 [5] readxl_1.3.1                RColorBrewer_1.1-2         
 [7] data.table_1.13.6           DESeq2_1.30.0              
 [9] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[11] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[13] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[15] IRanges_2.24.1              S4Vectors_0.28.1           
[17] BiocGenerics_0.36.0         forcats_0.5.0              
[19] stringr_1.4.0               dplyr_1.0.2                
[21] purrr_0.3.4                 readr_1.4.0                
[23] tidyr_1.1.2                 tibble_3.0.4               
[25] ggplot2_3.3.3               tidyverse_1.3.0            
[27] knitr_1.30                  workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0       ellipsis_0.3.1         rprojroot_2.0.2       
 [4] flextable_0.6.2        XVector_0.30.0         base64enc_0.1-3       
 [7] fs_1.5.0               mice_3.12.0            rstudioapi_0.13       
[10] bit64_4.0.5            AnnotationDbi_1.52.0   fansi_0.4.1           
[13] lubridate_1.7.9.2      xml2_1.3.2             splines_4.0.3         
[16] cachem_1.0.1           geneplotter_1.68.0     jsonlite_1.7.2        
[19] broom_0.7.3            annotate_1.68.0        dbplyr_2.0.0          
[22] compiler_4.0.3         httr_1.4.2             backports_1.2.0       
[25] assertthat_0.2.1       Matrix_1.2-18          fastmap_1.1.0         
[28] cli_2.2.0              later_1.1.0.1          htmltools_0.5.1.1     
[31] tools_4.0.3            gtable_0.3.0           glue_1.4.2            
[34] GenomeInfoDbData_1.2.4 Rcpp_1.0.5             cellranger_1.1.0      
[37] vctrs_0.3.6            writexl_1.3.1          xfun_0.20             
[40] ps_1.5.0               rvest_0.3.6            lifecycle_0.2.0       
[43] XML_3.99-0.5           zlibbioc_1.36.0        hms_1.0.0             
[46] promises_1.1.1         HardyWeinberg_1.7.1    yaml_2.2.1            
[49] memoise_2.0.0          gdtools_0.2.3          stringi_1.5.3         
[52] RSQLite_2.2.2          genefilter_1.72.0      zip_2.1.1             
[55] BiocParallel_1.24.1    truncnorm_1.0-8        chron_2.3-56          
[58] systemfonts_0.3.2      rlang_0.4.10           pkgconfig_2.0.3       
[61] bitops_1.0-6           Rsolnp_1.16            evaluate_0.14         
[64] lattice_0.20-41        bit_4.0.4              tidyselect_1.1.0      
[67] magrittr_2.0.1         R6_2.5.0               generics_0.1.0        
[70] DelayedArray_0.16.0    DBI_1.1.1              pillar_1.4.7          
[73] haven_2.3.1            whisker_0.4            withr_2.4.1           
[76] survival_3.2-7         RCurl_1.98-1.2         modelr_0.1.8          
[79] crayon_1.3.4           uuid_0.1-4             officer_0.3.16        
[82] rmarkdown_2.6          locfit_1.5-9.4         grid_4.0.3            
[85] blob_1.2.1             git2r_0.28.0           reprex_1.0.0          
[88] digest_0.6.27          webshot_0.5.2          xtable_1.8-4          
[91] httpuv_1.5.5           munsell_0.5.0          viridisLite_0.3.0     </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
